<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en"><head>

<meta charset="utf-8">
<meta name="generator" content="quarto-1.3.433">

<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">


<title>06 ANALYSIS 2</title>
<style>
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
div.columns{display: flex; gap: min(4vw, 1.5em);}
div.column{flex: auto; overflow-x: auto;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
ul.task-list li input[type="checkbox"] {
  width: 0.8em;
  margin: 0 0.8em 0.2em -1em; /* quarto-specific, see https://github.com/quarto-dev/quarto-cli/issues/4556 */ 
  vertical-align: middle;
}
/* CSS for syntax highlighting */
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
  }
pre.numberSource { margin-left: 3em;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
</style>


<script src="00_all_files/libs/clipboard/clipboard.min.js"></script>
<script src="00_all_files/libs/quarto-html/quarto.js"></script>
<script src="00_all_files/libs/quarto-html/popper.min.js"></script>
<script src="00_all_files/libs/quarto-html/tippy.umd.min.js"></script>
<script src="00_all_files/libs/quarto-html/anchor.min.js"></script>
<link href="00_all_files/libs/quarto-html/tippy.css" rel="stylesheet">
<link href="00_all_files/libs/quarto-html/quarto-syntax-highlighting.css" rel="stylesheet" id="quarto-text-highlighting-styles">
<script src="00_all_files/libs/bootstrap/bootstrap.min.js"></script>
<link href="00_all_files/libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="00_all_files/libs/bootstrap/bootstrap.min.css" rel="stylesheet" id="quarto-bootstrap" data-mode="light">


</head>

<body class="fullcontent">

<div id="quarto-content" class="page-columns page-rows-contents page-layout-article">

<main class="content" id="quarto-document-content">

<header id="title-block-header" class="quarto-title-block default">
<div class="quarto-title">
<h1 class="title">06 ANALYSIS 2</h1>
</div>



<div class="quarto-title-meta">

    
  
    
  </div>
  

</header>

<div class="cell">
<div class="sourceCode cell-code" id="cb1"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb1-1"><a href="#cb1-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(quarto)</span>
<span id="cb1-2"><a href="#cb1-2" aria-hidden="true" tabindex="-1"></a><span class="co">#| message: false</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<section id="render-individual-documents" class="level2">
<h2 class="anchored" data-anchor-id="render-individual-documents">Render individual documents</h2>
<p>A folderis created for each render document..?</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb2"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb2-1"><a href="#cb2-1" aria-hidden="true" tabindex="-1"></a><span class="fu">quarto_render</span>(<span class="at">input =</span> <span class="st">"01_load.qmd"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[31m

processing file: 01_load.qmd
[39m
  |                                                       
  |                                                 |   0%
  |                                                       
  |....                                             |   9%                     
  |                                                       
  |.........                                        |  18% [01_Load packages]  
  |                                                       
  |.............                                    |  27%                     
  |                                                       
  |..................                               |  36% [01_Set up]         
  |                                                       
  |......................                           |  45%                     
  |                                                       
  |...........................                      |  55% [01_Load expr-data] 
  |                                                       
  |...............................                  |  64%                     
  |                                                       
  |....................................             |  73% [01_Load meta-data] 
  |                                                       
  |........................................         |  82%                     
  |                                                       
  |.............................................    |  91% [01_Write tsv-files]
  |                                                       
  |.................................................| 100%                     
                                                                                                               
[31moutput file: 01_load.knit.md

[39m[1mpandoc [22m
  to: html
  output-file: 01_load.html
  standalone: true
  section-divs: true
  html-math-method: mathjax
  wrap: none
  default-image-extension: png
  
[1mmetadata[22m
  document-css: false
  link-citations: true
  date-format: long
  lang: en
  title: 01 LOAD
  editor: visual
  
Output created: 01_load.html</code></pre>
</div>
<div class="sourceCode cell-code" id="cb4"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb4-1"><a href="#cb4-1" aria-hidden="true" tabindex="-1"></a><span class="fu">file.rename</span>(<span class="at">from =</span> <span class="st">"01_load.html"</span>,</span>
<span id="cb4-2"><a href="#cb4-2" aria-hidden="true" tabindex="-1"></a>            <span class="at">to =</span> <span class="st">"../results/01_load.html"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] TRUE</code></pre>
</div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb6"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb6-1"><a href="#cb6-1" aria-hidden="true" tabindex="-1"></a><span class="fu">quarto_render</span>(<span class="at">input =</span> <span class="st">"02_clean.qmd"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[31m

processing file: 02_clean.qmd
[39m
  |                  
  |            |   0%
  |                  
  |            |   2%                                                          
  |                  
  |.           |   5% [02_Load packages]                                       
  |                  
  |.           |   7%                                                          
  |                  
  |.           |  10% [02_Load data]                                           
  |                  
  |.           |  12%                                                          
  |                  
  |..          |  15% [unnamed-chunk-3]                                        
  |                  
  |..          |  17%                                                          
  |                  
  |..          |  20% [02_Clean dat_expr]                                      
  |                  
  |...         |  22%                                                          
  |                  
  |...         |  24% [02_Simplify dat_expr]                                   
  |                  
  |...         |  27%                                                          
  |                  
  |....        |  29% [unnamed-chunk-6]                                        
  |                  
  |....        |  32%                                                          
  |                  
  |....        |  34% [02_Extract relevant dat_meta]                           
  |                  
  |....        |  37%                                                          
  |                  
  |.....       |  39% [02_Simplify dat_meta]                                   
  |                  
  |.....       |  41%                                                          
  |                  
  |.....       |  44% [unnamed-chunk-9]                                        
  |                  
  |......      |  46%                                                          
  |                  
  |......      |  49% [02_Streamline 'identifier']                             
  |                  
  |......      |  51%                                                          
  |                  
  |......      |  54% [02_Compare 'identifier' with dat_expr column name]      
  |                  
  |.......     |  56%                                                          
  |                  
  |.......     |  59% [02_Pivot dat_expr]                                      
  |                  
  |.......     |  61%                                                          
  |                  
  |........    |  63% [02_Join dat_expr with dat_meta]                         
  |                  
  |........    |  66%                                                          
  |                  
  |........    |  68% [02_Extract replicates]                                  
  |                  
  |........    |  71%                                                          
  |                  
  |.........   |  73% [02_Mean expression for replicates]                      
  |                  
  |.........   |  76%                                                          
  |                  
  |.........   |  78% [02_Extraxt and collapse meta data for replicates]       
  |                  
  |..........  |  80%                                                          
  |                  
  |..........  |  83% [02_Add column modified]                                 
  |                  
  |..........  |  85%                                                          
  |                  
  |........... |  88% [02_Combine mean expression and meta data for replicates]
  |                  
  |........... |  90%                                                          
  |                  
  |........... |  93% [02_Add rep data with non-rep data]                      
  |                  
  |........... |  95%                                                          
  |                  
  |............|  98% [02_Write tsv-files]                                     
  |                  
  |............| 100%                                                          
                                                                                                                                                    
[31moutput file: 02_clean.knit.md

[39m[1mpandoc [22m
  to: html
  output-file: 02_clean.html
  standalone: true
  section-divs: true
  html-math-method: mathjax
  wrap: none
  default-image-extension: png
  
[1mmetadata[22m
  document-css: false
  link-citations: true
  date-format: long
  lang: en
  title: 02 CLEAN
  editor: visual
  
Output created: 02_clean.html</code></pre>
</div>
<div class="sourceCode cell-code" id="cb8"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb8-1"><a href="#cb8-1" aria-hidden="true" tabindex="-1"></a><span class="fu">file.rename</span>(<span class="at">from =</span> <span class="st">"02_clean.html"</span>,</span>
<span id="cb8-2"><a href="#cb8-2" aria-hidden="true" tabindex="-1"></a>            <span class="at">to =</span> <span class="st">"../results/02_clean.html"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] TRUE</code></pre>
</div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb10"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb10-1"><a href="#cb10-1" aria-hidden="true" tabindex="-1"></a><span class="fu">quarto_render</span>(<span class="at">input =</span> <span class="st">"03_describe.qmd"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[31m

processing file: 03_describe.qmd
[39m
  |                                              
  |                                        |   0%
  |                                              
  |....                                    |   9%                              
  |                                              
  |.......                                 |  18% [03_Load packages]           
  |                                              
  |...........                             |  27%                              
  |                                              
  |...............                         |  36% [03_Load data]               
  |                                              
  |..................                      |  45%                              
  |                                              
  |......................                  |  55% [03_Define settings for plot]
  |                                              
  |.........................               |  64%                              
  |                                              
  |.............................           |  73% [03_Bar plot]                
  |                                              
  |.................................       |  82%                              
  |                                              
  |....................................    |  91% [03_Summary table]           
  |                                              
  |........................................| 100%                              
                                                                                                                        
[31moutput file: 03_describe.knit.md

[39m[1mpandoc [22m
  to: html
  output-file: 03_describe.html
  standalone: true
  section-divs: true
  html-math-method: mathjax
  wrap: none
  default-image-extension: png
  
[1mmetadata[22m
  document-css: false
  link-citations: true
  date-format: long
  lang: en
  title: 03 DESCRIBE
  editor: visual
  
Output created: 03_describe.html</code></pre>
</div>
<div class="sourceCode cell-code" id="cb12"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb12-1"><a href="#cb12-1" aria-hidden="true" tabindex="-1"></a><span class="fu">file.rename</span>(<span class="at">from =</span> <span class="st">"03_describe.html"</span>,</span>
<span id="cb12-2"><a href="#cb12-2" aria-hidden="true" tabindex="-1"></a>            <span class="at">to =</span> <span class="st">"../results/03_describe.html"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] TRUE</code></pre>
</div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb14"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb14-1"><a href="#cb14-1" aria-hidden="true" tabindex="-1"></a><span class="fu">quarto_render</span>(<span class="at">input =</span> <span class="st">"04_augment.qmd"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[31m

processing file: 04_augment.qmd
[39m
  |                                                    
  |                                              |   0%
  |                                                    
  |..                                            |   5%                        
  |                                                    
  |.....                                         |  11% [04_Load packages]     
  |                                                    
  |.......                                       |  16%                        
  |                                                    
  |..........                                    |  21% [04_Load data]         
  |                                                    
  |............                                  |  26%                        
  |                                                    
  |...............                               |  32% [04_Split data]        
  |                                                    
  |.................                             |  37%                        
  |                                                    
  |...................                           |  42% [04_Normalise dat_expr]
  |                                                    
  |......................                        |  47%                        
  |                                                    
  |........................                      |  53% [04_Log-transformation]
  |                                                    
  |...........................                   |  58%                        
  |                                                    
  |.............................                 |  63% [unnamed-chunk-6]      
  |                                                    
  |...............................               |  68%                        
  |                                                    
  |..................................            |  74% [04_Write tsv-files]   
  |                                                    
  |....................................          |  79%                        
  |                                                    
  |.......................................       |  84% [unnamed-chunk-8]      
  |                                                    
  |.........................................     |  89%                        
  |                                                    
  |............................................  |  95% [unnamed-chunk-9]      
  |                                                    
  |..............................................| 100%                        
                                                                                                                  
[31moutput file: 04_augment.knit.md

[39m[1mpandoc [22m
  to: html
  output-file: 04_augment.html
  standalone: true
  section-divs: true
  html-math-method: mathjax
  wrap: none
  default-image-extension: png
  
[1mmetadata[22m
  document-css: false
  link-citations: true
  date-format: long
  lang: en
  title: 04 AUGMENT
  editor: visual
  
Output created: 04_augment.html</code></pre>
</div>
<div class="sourceCode cell-code" id="cb16"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb16-1"><a href="#cb16-1" aria-hidden="true" tabindex="-1"></a><span class="fu">file.rename</span>(<span class="at">from =</span> <span class="st">"04_augment.html"</span>,</span>
<span id="cb16-2"><a href="#cb16-2" aria-hidden="true" tabindex="-1"></a>            <span class="at">to =</span> <span class="st">"../results/04_augment.html"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] TRUE</code></pre>
</div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb18"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb18-1"><a href="#cb18-1" aria-hidden="true" tabindex="-1"></a><span class="fu">quarto_render</span>(<span class="at">input =</span> <span class="st">"05_analysis_1.qmd"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[31m

processing file: 05_analysis_1.qmd
[39m
  |                                               
  |                                         |   0%
  |                                               
  |....                                     |   9%                             
  |                                               
  |.......                                  |  18% [05_Load packages]          
  |                                               
  |...........                              |  27%                             
  |                                               
  |...............                          |  36% [05_Load data]              
  |                                               
  |...................                      |  45%                             
  |                                               
  |......................                   |  55% [05_PCA]                    
  |                                               
  |..........................               |  64%                             
  |                                               
  |..............................           |  73% [05_Plot PC1 vs PC2]        
  |                                               
  |..................................       |  82%                             
  |                                               
  |.....................................    |  91% [05_Plot variance explained]
  |                                               
  |.........................................| 100%                             
                                                                                                                       
[31moutput file: 05_analysis_1.knit.md

[39m[1mpandoc [22m
  to: html
  output-file: 05_analysis_1.html
  standalone: true
  section-divs: true
  html-math-method: mathjax
  wrap: none
  default-image-extension: png
  
[1mmetadata[22m
  document-css: false
  link-citations: true
  date-format: long
  lang: en
  title: 05 ANALYSIS 1
  editor: visual
  
Output created: 05_analysis_1.html</code></pre>
</div>
<div class="sourceCode cell-code" id="cb20"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb20-1"><a href="#cb20-1" aria-hidden="true" tabindex="-1"></a><span class="fu">file.rename</span>(<span class="at">from =</span> <span class="st">"05_analysis_1.html"</span>,</span>
<span id="cb20-2"><a href="#cb20-2" aria-hidden="true" tabindex="-1"></a>            <span class="at">to =</span> <span class="st">"../results/05_analysis_1.html"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] TRUE</code></pre>
</div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb22"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb22-1"><a href="#cb22-1" aria-hidden="true" tabindex="-1"></a><span class="fu">quarto_render</span>(<span class="at">input =</span> <span class="st">"06_analysis_2.qmd"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[31m

processing file: 06_analysis_2.qmd
[39m
  |                          
  |                    |   0%
  |                          
  |.                   |   4%                                                  
  |                          
  |..                  |   8% [06_Load packages]                               
  |                          
  |..                  |  12%                                                  
  |                          
  |...                 |  16% [06_Load data]                                   
  |                          
  |....                |  20%                                                  
  |                          
  |.....               |  24% [06_Pivot and nest data]                         
  |                          
  |......              |  28%                                                  
  |                          
  |......              |  32% [06_Linear regression model]                     
  |                          
  |.......             |  36%                                                  
  |                          
  |........            |  40% [06_Apply tidy to model objects]                 
  |                          
  |.........           |  44%                                                  
  |                          
  |..........          |  48% [06_Unnest and ungroup data]                     
  |                          
  |..........          |  52%                                                  
  |                          
  |...........         |  56% [06_Adjusted p-value and significant label]      
  |                          
  |............        |  60%                                                  
  |                          
  |.............       |  64% [unnamed-chunk-8]                                
  |                          
  |..............      |  68%                                                  
  |                          
  |..............      |  72% [06_extract original gene IDs]                   
  |                          
  |...............     |  76%                                                  
  |                          
  |................    |  80% [06_compare significant genes with 251 signature]
  |                          
  |.................   |  84%                                                  
  |                          
  |..................  |  88% [06_Forrest plot]                                
  |                          
  |..................  |  92%                                                  
  |                          
  |................... |  96% [06_volcano plot]                                
  |                          
  |....................| 100%                                                  
                                                                                                                                            
[31moutput file: 06_analysis_2.knit.md

[39m[1mpandoc [22m
  to: html
  output-file: 06_analysis_2.html
  standalone: true
  section-divs: true
  html-math-method: mathjax
  wrap: none
  default-image-extension: png
  
[1mmetadata[22m
  document-css: false
  link-citations: true
  date-format: long
  lang: en
  title: 06 ANALYSIS 2
  editor: visual
  
Output created: 06_analysis_2.html</code></pre>
</div>
<div class="sourceCode cell-code" id="cb24"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb24-1"><a href="#cb24-1" aria-hidden="true" tabindex="-1"></a><span class="fu">file.rename</span>(<span class="at">from =</span> <span class="st">"06_analysis_2.html"</span>,</span>
<span id="cb24-2"><a href="#cb24-2" aria-hidden="true" tabindex="-1"></a>            <span class="at">to =</span> <span class="st">"../results/06_analysis_2.html"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] TRUE</code></pre>
</div>
</div>
</section>
<section id="run-full-analysis-to-create-one-html-file" class="level2">
<h2 class="anchored" data-anchor-id="run-full-analysis-to-create-one-html-file">Run full analysis to create one html file</h2>
</section>
<section id="data-origin" class="level2">
<h2 class="anchored" data-anchor-id="data-origin">Data Origin</h2>
<p>The gene expression data used in this project is available in the Gene Expression Omnibus (GEO) under the accession number GSE50834. The data originates from the following paper:</p>
<p>Dawany N, Showe LC, Kossenkov AV, Chang C et al.&nbsp;Identification of a 251 gene expression signature that can accurately detect M. tuberculosis in patients with and without HIV co-infection. PLoS One 2014;9(2):e89925. PMID: 24587128</p>
</section>
<section id="summary" class="level2">
<h2 class="anchored" data-anchor-id="summary">Summary</h2>
<p>The data comes from total RNA isolated from peripheral blood mononuclear cell (PBMC) from a South African cohort. Microarry analysis was performed to compare gene expression in patients either infected with HIV or co-infected with HIV and tuberculosis (TB). Diagnosis of TB, especially in the presence of an HIV co-infection, can be challenging when using conventional diagnostic methods. Therefore, the data was used to identify a TB-specific gene signature.</p>
</section>
<section id="setup" class="level2">
<h2 class="anchored" data-anchor-id="setup">Setup</h2>
<p>Loading required packages.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb26"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb26-1"><a href="#cb26-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(<span class="st">"tidyverse"</span>)</span>
<span id="cb26-2"><a href="#cb26-2" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(<span class="st">"stringr"</span>)</span>
<span id="cb26-3"><a href="#cb26-3" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(<span class="st">"GEOquery"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>Note: Instruction of how to install the package GEOquery can be found here <em>https://bioconductor.org/packages/release/bioc/html/GEOquery.html</em></p>
</section>
<section id="load-data" class="level2">
<h2 class="anchored" data-anchor-id="load-data">Load Data</h2>
<p>This document will load the data from Gene Expression Omnibus. Furthermore, the file will export the data as a tsv-file. In the following, the data directory is defined along with the location of the original data.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb27"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb27-1"><a href="#cb27-1" aria-hidden="true" tabindex="-1"></a>dest_dir <span class="ot">&lt;-</span> <span class="st">"../data/"</span></span>
<span id="cb27-2"><a href="#cb27-2" aria-hidden="true" tabindex="-1"></a>raw_dir <span class="ot">&lt;-</span> <span class="fu">str_c</span>(dest_dir, <span class="st">"_raw/"</span>)</span>
<span id="cb27-3"><a href="#cb27-3" aria-hidden="true" tabindex="-1"></a>expr_file <span class="ot">&lt;-</span> <span class="st">"GSE50834_non-normalized.txt.gz"</span></span>
<span id="cb27-4"><a href="#cb27-4" aria-hidden="true" tabindex="-1"></a>meta_file <span class="ot">&lt;-</span> <span class="st">"GSE50834_series_matrix.txt.gz"</span></span>
<span id="cb27-5"><a href="#cb27-5" aria-hidden="true" tabindex="-1"></a>geo <span class="ot">&lt;-</span> <span class="st">"GSE50834"</span></span>
<span id="cb27-6"><a href="#cb27-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-7"><a href="#cb27-7" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span>( <span class="sc">!</span><span class="fu">dir.exists</span>(raw_dir) ){</span>
<span id="cb27-8"><a href="#cb27-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">dir.create</span>(<span class="at">path =</span> raw_dir,</span>
<span id="cb27-9"><a href="#cb27-9" aria-hidden="true" tabindex="-1"></a>             <span class="at">recursive =</span> <span class="cn">TRUE</span>)</span>
<span id="cb27-10"><a href="#cb27-10" aria-hidden="true" tabindex="-1"></a>}</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>The gene expression data, containing patient ID’s and gene expression, is downloaded and stored in the raw data directory using the GEOquery library. GEOquery can download data sets from the NCBI Gene Expression Omnibus (GEO), using a unique accession number.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb28"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb28-1"><a href="#cb28-1" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span>( <span class="sc">!</span><span class="fu">file.exists</span>(<span class="fu">str_c</span>(raw_dir, expr_file)) ){</span>
<span id="cb28-2"><a href="#cb28-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">getGEOSuppFiles</span>(<span class="at">GEO =</span> geo,</span>
<span id="cb28-3"><a href="#cb28-3" aria-hidden="true" tabindex="-1"></a>                  <span class="at">makeDirectory =</span> <span class="cn">FALSE</span>,</span>
<span id="cb28-4"><a href="#cb28-4" aria-hidden="true" tabindex="-1"></a>                  <span class="at">baseDir =</span> raw_dir,</span>
<span id="cb28-5"><a href="#cb28-5" aria-hidden="true" tabindex="-1"></a>                  <span class="at">fetch_files =</span> <span class="cn">TRUE</span>,</span>
<span id="cb28-6"><a href="#cb28-6" aria-hidden="true" tabindex="-1"></a>                  <span class="at">filter_regex =</span> <span class="st">"txt"</span>)</span>
<span id="cb28-7"><a href="#cb28-7" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb28-8"><a href="#cb28-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb28-9"><a href="#cb28-9" aria-hidden="true" tabindex="-1"></a>dat_expr <span class="ot">&lt;-</span> <span class="fu">read.table</span>(<span class="at">file =</span> <span class="fu">str_c</span>(raw_dir, expr_file),</span>
<span id="cb28-10"><a href="#cb28-10" aria-hidden="true" tabindex="-1"></a>                         <span class="at">sep =</span> <span class="st">"</span><span class="sc">\t</span><span class="st">"</span>,</span>
<span id="cb28-11"><a href="#cb28-11" aria-hidden="true" tabindex="-1"></a>                         <span class="at">header =</span> <span class="cn">TRUE</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>The meta data, containing patient information, is downloaded and stored in the raw data directory.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb29"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb29-1"><a href="#cb29-1" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span> (<span class="sc">!</span><span class="fu">file.exists</span>(<span class="fu">str_c</span>(raw_dir, meta_file))){</span>
<span id="cb29-2"><a href="#cb29-2" aria-hidden="true" tabindex="-1"></a>  dat_meta <span class="ot">&lt;-</span> <span class="fu">getGEO</span>(<span class="at">GEO =</span> geo)</span>
<span id="cb29-3"><a href="#cb29-3" aria-hidden="true" tabindex="-1"></a>  dat_meta <span class="ot">&lt;-</span> <span class="fu">pData</span>(<span class="fu">phenoData</span>(dat_meta[[<span class="dv">1</span>]]))</span>
<span id="cb29-4"><a href="#cb29-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">write.table</span>(dat_meta, <span class="at">file =</span> <span class="fu">str_c</span>(raw_dir, meta_file), </span>
<span id="cb29-5"><a href="#cb29-5" aria-hidden="true" tabindex="-1"></a>             <span class="at">sep =</span> <span class="st">"</span><span class="sc">\t</span><span class="st">"</span>)</span>
<span id="cb29-6"><a href="#cb29-6" aria-hidden="true" tabindex="-1"></a>  dat_meta <span class="ot">&lt;-</span> <span class="fu">read.table</span>(<span class="fu">str_c</span>(raw_dir, meta_file), </span>
<span id="cb29-7"><a href="#cb29-7" aria-hidden="true" tabindex="-1"></a>                        <span class="at">sep =</span> <span class="st">"</span><span class="sc">\t</span><span class="st">"</span>,</span>
<span id="cb29-8"><a href="#cb29-8" aria-hidden="true" tabindex="-1"></a>                        <span class="at">header =</span> <span class="cn">TRUE</span>)</span>
<span id="cb29-9"><a href="#cb29-9" aria-hidden="true" tabindex="-1"></a>} <span class="cf">else</span> {</span>
<span id="cb29-10"><a href="#cb29-10" aria-hidden="true" tabindex="-1"></a>  dat_meta <span class="ot">&lt;-</span> <span class="fu">read.table</span>(<span class="fu">str_c</span>(raw_dir, meta_file), </span>
<span id="cb29-11"><a href="#cb29-11" aria-hidden="true" tabindex="-1"></a>                          <span class="at">sep =</span> <span class="st">"</span><span class="sc">\t</span><span class="st">"</span>,</span>
<span id="cb29-12"><a href="#cb29-12" aria-hidden="true" tabindex="-1"></a>                          <span class="at">header =</span> <span class="cn">TRUE</span>)</span>
<span id="cb29-13"><a href="#cb29-13" aria-hidden="true" tabindex="-1"></a>}</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="write-data-to-files" class="level2">
<h2 class="anchored" data-anchor-id="write-data-to-files">Write Data to Files</h2>
<p>The gene expression data and the meta data are saved as .tsv-files</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb30"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb30-1"><a href="#cb30-1" aria-hidden="true" tabindex="-1"></a>dat_expr <span class="sc">|&gt;</span> </span>
<span id="cb30-2"><a href="#cb30-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">write_tsv</span>(<span class="at">file =</span> <span class="fu">str_c</span>(dest_dir, <span class="st">"01_dat_load_expr.tsv"</span>))</span>
<span id="cb30-3"><a href="#cb30-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb30-4"><a href="#cb30-4" aria-hidden="true" tabindex="-1"></a>dat_meta <span class="sc">|&gt;</span> </span>
<span id="cb30-5"><a href="#cb30-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">write_tsv</span>(<span class="at">file =</span> <span class="fu">str_c</span>(dest_dir, <span class="st">"01_dat_load_meta.tsv"</span>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="setup-1" class="level2">
<h2 class="anchored" data-anchor-id="setup-1">Setup</h2>
<p>Load required packages.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb31"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb31-1"><a href="#cb31-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(<span class="st">"tidyverse"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>Load gene expression and meta data.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb32"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb32-1"><a href="#cb32-1" aria-hidden="true" tabindex="-1"></a>dat_expr <span class="ot">&lt;-</span> <span class="fu">read_tsv</span>(<span class="st">"../data/01_dat_load_expr.tsv"</span>)</span>
<span id="cb32-2"><a href="#cb32-2" aria-hidden="true" tabindex="-1"></a>dat_meta <span class="ot">&lt;-</span> <span class="fu">read_tsv</span>(<span class="st">"../data/01_dat_load_meta.tsv"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="data-wrangling-i" class="level2">
<h2 class="anchored" data-anchor-id="data-wrangling-i">Data Wrangling I</h2>
<p>In this section, variable of gene expression data and meta data are variables are cleaned separately.</p>
<section id="gene-expression-data" class="level3">
<h3 class="anchored" data-anchor-id="gene-expression-data">Gene Expression Data</h3>
<p>In the following the gene expression data is inspected, cleaned and simplified.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb33"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb33-1"><a href="#cb33-1" aria-hidden="true" tabindex="-1"></a>dat_expr <span class="sc">|&gt;</span> </span>
<span id="cb33-2"><a href="#cb33-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">sample_n</span>(<span class="dv">10</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code># A tibble: 10 × 89
   ID_REF       AAA0VKQN.12_18.Mar.2009 Detection.Pval AAA0W22R.11_16.Apr.2009
   &lt;chr&gt;                          &lt;dbl&gt;          &lt;dbl&gt;                   &lt;dbl&gt;
 1 ILMN_1892599                    323.         0.0883                   145. 
 2 ILMN_3310690                    204.         0.714                    128. 
 3 ILMN_3274775                    258.         0.527                    144. 
 4 ILMN_1707372                    337.         0.0494                   131. 
 5 ILMN_1814583                    159.         0.821                     77.6
 6 ILMN_3243583                    229.         0.668                    112. 
 7 ILMN_2129545                    519.         0                        316. 
 8 ILMN_3237085                    233.         0.660                    118. 
 9 ILMN_1774259                    416.         0                        196. 
10 ILMN_1653578                    151.         0.865                     76.2
# ℹ 85 more variables: Detection.Pval.1 &lt;dbl&gt;, AAA0W22R.14_16.Apr.2009 &lt;dbl&gt;,
#   Detection.Pval.2 &lt;dbl&gt;, BAA0V9SR.15_25.Feb.2009 &lt;dbl&gt;,
#   Detection.Pval.3 &lt;dbl&gt;, BAA0W3BM.11_20.Apr.2009 &lt;dbl&gt;,
#   Detection.Pval.4 &lt;dbl&gt;, BAA0W69B.10_29.Apr.2009 &lt;dbl&gt;,
#   Detection.Pval.5 &lt;dbl&gt;, BAA0WVPG.10_04.Jun.2009 &lt;dbl&gt;,
#   Detection.Pval.6 &lt;dbl&gt;, CAA0Q3CK.13_24.Apr.2008 &lt;dbl&gt;,
#   Detection.Pval.7 &lt;dbl&gt;, CAA0W4LJ.15_23.Apr.2009 &lt;dbl&gt;, …</code></pre>
</div>
</div>
<p>From the inspection, it is seen that the data contain both gene expressions and p values. The detected p values are removed such that the gene expression data only consists of the gene expressions.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb35"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb35-1"><a href="#cb35-1" aria-hidden="true" tabindex="-1"></a>dat_expr <span class="ot">&lt;-</span> dat_expr <span class="sc">|&gt;</span> </span>
<span id="cb35-2"><a href="#cb35-2" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">select</span>(<span class="sc">!</span><span class="fu">matches</span>(<span class="at">match =</span> <span class="st">"Detection.Pval"</span>))</span>
<span id="cb35-3"><a href="#cb35-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb35-4"><a href="#cb35-4" aria-hidden="true" tabindex="-1"></a>dat_expr <span class="sc">|&gt;</span> </span>
<span id="cb35-5"><a href="#cb35-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">sample_n</span>(<span class="dv">10</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code># A tibble: 10 × 45
   ID_REF   AAA0VKQN.12_18.Mar.2…¹ AAA0W22R.11_16.Apr.2…² AAA0W22R.14_16.Apr.2…³
   &lt;chr&gt;                     &lt;dbl&gt;                  &lt;dbl&gt;                  &lt;dbl&gt;
 1 ILMN_16…                   595.                  277.                    373.
 2 ILMN_21…                   795.                  367.                    489.
 3 ILMN_18…                   304.                  153.                    256.
 4 ILMN_17…                   340.                  189.                    296.
 5 ILMN_32…                   214.                  112.                    227.
 6 ILMN_32…                   209.                  115.                    239.
 7 ILMN_16…                   355.                  176.                    263.
 8 ILMN_18…                   315.                  170.                    281.
 9 ILMN_32…                   316.                  168.                    263.
10 ILMN_17…                   145.                   66.6                   127.
# ℹ abbreviated names: ¹​AAA0VKQN.12_18.Mar.2009, ²​AAA0W22R.11_16.Apr.2009,
#   ³​AAA0W22R.14_16.Apr.2009
# ℹ 41 more variables: BAA0V9SR.15_25.Feb.2009 &lt;dbl&gt;,
#   BAA0W3BM.11_20.Apr.2009 &lt;dbl&gt;, BAA0W69B.10_29.Apr.2009 &lt;dbl&gt;,
#   BAA0WVPG.10_04.Jun.2009 &lt;dbl&gt;, CAA0Q3CK.13_24.Apr.2008 &lt;dbl&gt;,
#   CAA0W4LJ.15_23.Apr.2009 &lt;dbl&gt;, CAA0W4LT.13_23.Apr.2009 &lt;dbl&gt;,
#   DAA0PFCT.11_17.Jan.2008 &lt;dbl&gt;, DAA0V9RP.16_25.Feb.2009 &lt;dbl&gt;, …</code></pre>
</div>
</div>
<p>Column names are simplified as the date is saved in the meta data.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb37"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb37-1"><a href="#cb37-1" aria-hidden="true" tabindex="-1"></a>dat_expr <span class="sc">|&gt;</span> </span>
<span id="cb37-2"><a href="#cb37-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">colnames</span>() <span class="ot">=</span> <span class="fu">str_extract</span>(<span class="at">string =</span> <span class="fu">colnames</span>(dat_expr),</span>
<span id="cb37-3"><a href="#cb37-3" aria-hidden="true" tabindex="-1"></a>                           <span class="at">pattern =</span> <span class="st">"^[A-Z0-9]+</span><span class="sc">\\</span><span class="st">.?[0-9]*"</span>)</span>
<span id="cb37-4"><a href="#cb37-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb37-5"><a href="#cb37-5" aria-hidden="true" tabindex="-1"></a>dat_expr <span class="sc">|&gt;</span> </span>
<span id="cb37-6"><a href="#cb37-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">sample_n</span>(<span class="dv">10</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code># A tibble: 10 × 45
   ID    AAA0VKQN.12 AAA0W22R.11 AAA0W22R.14 BAA0V9SR.15 BAA0W3BM.11 BAA0W69B.10
   &lt;chr&gt;       &lt;dbl&gt;       &lt;dbl&gt;       &lt;dbl&gt;       &lt;dbl&gt;       &lt;dbl&gt;       &lt;dbl&gt;
 1 ILMN…        255.       130.         232.        254.        262.       131. 
 2 ILMN…        252.        91.5        184.        217.        233.       142. 
 3 ILMN…        312.       182.         258.        314.        301.       140. 
 4 ILMN…        278.       136.         228.        242.        237.       117. 
 5 ILMN…        171.        96.0        150.        196.        160.       107. 
 6 ILMN…        251.       131.         222.        282.        241.       131. 
 7 ILMN…        331.       151.         273.        258.        306.       135. 
 8 ILMN…        186.       101.         214.        241.        232.       126. 
 9 ILMN…        181.        99.8        140.        168.        156.        63.1
10 ILMN…        144.        94.1        136.        212.        145.        85.7
# ℹ 38 more variables: BAA0WVPG.10 &lt;dbl&gt;, CAA0Q3CK.13 &lt;dbl&gt;, CAA0W4LJ.15 &lt;dbl&gt;,
#   CAA0W4LT.13 &lt;dbl&gt;, DAA0PFCT.11 &lt;dbl&gt;, DAA0V9RP.16 &lt;dbl&gt;, DAA0VD27.11 &lt;dbl&gt;,
#   FAA0W22D.13 &lt;dbl&gt;, HAA0W3B8.14 &lt;dbl&gt;, HAA0W3R9.13 &lt;dbl&gt;, JAA0P0DQ.11 &lt;dbl&gt;,
#   JAA0W223.10 &lt;dbl&gt;, KAA0PBTZ.17 &lt;dbl&gt;, KAA0Q3QH.14 &lt;dbl&gt;, AAA0WVKX.11 &lt;dbl&gt;,
#   EAA0VKQY.17 &lt;dbl&gt;, JAA0WVMP.13 &lt;dbl&gt;, AAA0KHPN.10 &lt;dbl&gt;, AAA0RWJ0.14 &lt;dbl&gt;,
#   AAA0Z4D1.15 &lt;dbl&gt;, AAA0ZSJX.10 &lt;dbl&gt;, BAA0YXVB.12 &lt;dbl&gt;, BAA0ZC4J.13 &lt;dbl&gt;,
#   BAA0ZXTF.11 &lt;dbl&gt;, CAA0ZN6L.12 &lt;dbl&gt;, DAA0Z4CR.10 &lt;dbl&gt;, …</code></pre>
</div>
</div>
</section>
<section id="meta-data" class="level3">
<h3 class="anchored" data-anchor-id="meta-data">Meta Data</h3>
<p>In the following the meta data is inspected, relevant columns are extracted, renamed and simplified</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb39"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb39-1"><a href="#cb39-1" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(dat_meta)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code> [1] "title"                   "geo_accession"          
 [3] "status"                  "submission_date"        
 [5] "last_update_date"        "type"                   
 [7] "channel_count"           "source_name_ch1"        
 [9] "organism_ch1"            "characteristics_ch1"    
[11] "characteristics_ch1.1"   "characteristics_ch1.2"  
[13] "molecule_ch1"            "extract_protocol_ch1"   
[15] "label_ch1"               "label_protocol_ch1"     
[17] "taxid_ch1"               "hyb_protocol"           
[19] "scan_protocol"           "description"            
[21] "description.1"           "data_processing"        
[23] "platform_id"             "contact_name"           
[25] "contact_email"           "contact_phone"          
[27] "contact_laboratory"      "contact_institute"      
[29] "contact_address"         "contact_city"           
[31] "contact_state"           "contact_zip.postal_code"
[33] "contact_country"         "supplementary_file"     
[35] "data_row_count"          "disease.state.ch1"      
[37] "gender.ch1"              "sample.type.ch1"        </code></pre>
</div>
</div>
<p>We are interested in keeping ‘title’, ‘geo_accession’, and ‘description.1’. The last mentioned column is the identifier refering to each sample found in the gene expression data. Furthermore, we want to extract information about gender and disease state described by ‘characteristics_ch1’ and ‘characteristics_ch1.2’. This information is also found in the variables ‘gender.ch1’ and ‘disease.state.ch1’ in a nicer format. We chose to work with the first mentioned columns for educational purposes.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb41"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb41-1"><a href="#cb41-1" aria-hidden="true" tabindex="-1"></a>dat_meta <span class="ot">&lt;-</span> dat_meta <span class="sc">|&gt;</span></span>
<span id="cb41-2"><a href="#cb41-2" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">select</span>(title,</span>
<span id="cb41-3"><a href="#cb41-3" aria-hidden="true" tabindex="-1"></a>                geo_accession,</span>
<span id="cb41-4"><a href="#cb41-4" aria-hidden="true" tabindex="-1"></a>                characteristics_ch1,</span>
<span id="cb41-5"><a href="#cb41-5" aria-hidden="true" tabindex="-1"></a>                characteristics_ch1<span class="fl">.2</span>,</span>
<span id="cb41-6"><a href="#cb41-6" aria-hidden="true" tabindex="-1"></a>                description<span class="fl">.1</span>) <span class="sc">|&gt;</span> </span>
<span id="cb41-7"><a href="#cb41-7" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">rename</span>(<span class="st">"patient_n"</span> <span class="ot">=</span> title,</span>
<span id="cb41-8"><a href="#cb41-8" aria-hidden="true" tabindex="-1"></a>                <span class="st">"gender"</span> <span class="ot">=</span> characteristics_ch1,</span>
<span id="cb41-9"><a href="#cb41-9" aria-hidden="true" tabindex="-1"></a>                <span class="st">"disease_state"</span> <span class="ot">=</span> characteristics_ch1<span class="fl">.2</span>,</span>
<span id="cb41-10"><a href="#cb41-10" aria-hidden="true" tabindex="-1"></a>                <span class="st">"identifier"</span> <span class="ot">=</span> description<span class="fl">.1</span>)</span>
<span id="cb41-11"><a href="#cb41-11" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb41-12"><a href="#cb41-12" aria-hidden="true" tabindex="-1"></a>dat_meta <span class="sc">|&gt;</span> </span>
<span id="cb41-13"><a href="#cb41-13" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">select</span>(patient_n, </span>
<span id="cb41-14"><a href="#cb41-14" aria-hidden="true" tabindex="-1"></a>         gender, </span>
<span id="cb41-15"><a href="#cb41-15" aria-hidden="true" tabindex="-1"></a>         disease_state,</span>
<span id="cb41-16"><a href="#cb41-16" aria-hidden="true" tabindex="-1"></a>         identifier) <span class="sc">|&gt;</span></span>
<span id="cb41-17"><a href="#cb41-17" aria-hidden="true" tabindex="-1"></a>  <span class="fu">sample_n</span>(<span class="dv">10</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code># A tibble: 10 × 4
   patient_n                     gender         disease_state         identifier
   &lt;chr&gt;                         &lt;chr&gt;          &lt;chr&gt;                 &lt;chr&gt;     
 1 HIV/TB co-infected patient 7  gender: Female disease state: HIV/TB BAA0ZXTF-…
 2 HIV/TB co-infected patient 15 gender: Female disease state: HIV/TB JAA0Q3QZ-…
 3 HIV infected patient 14       gender: Female disease state: HIV o… HAA0W3B8-…
 4 HIV/TB co-infected patient 9  gender: Male   disease state: HIV/TB DAA0Z4CR-…
 5 HIV infected patient 4        gender: Female disease state: HIV o… BAA0W3BM-…
 6 HIV infected patient 22       gender: Female disease state: HIV o… JAA0WVMP-…
 7 HIV/TB co-infected patient 3  gender: Male   disease state: HIV/TB AAA0Z4D1-…
 8 HIV infected patient 6        gender: Female disease state: HIV o… BAA0WVPG-…
 9 HIV infected patient 20       gender: Female disease state: HIV o… AAA0WVKX-…
10 HIV/TB co-infected patient 1  gender: Female disease state: HIV/TB AAA0KHPN-…</code></pre>
</div>
</div>
<p>Simplify the columns ‘patient’, ‘gender’ and ‘disease state’.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb43"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb43-1"><a href="#cb43-1" aria-hidden="true" tabindex="-1"></a>dat_meta <span class="ot">&lt;-</span> dat_meta <span class="sc">|&gt;</span> </span>
<span id="cb43-2"><a href="#cb43-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">patient_n =</span> <span class="fu">str_extract</span>(<span class="at">string =</span> patient_n,</span>
<span id="cb43-3"><a href="#cb43-3" aria-hidden="true" tabindex="-1"></a>                                 <span class="at">pattern =</span> <span class="st">"[0-9]+_?[0-9]?"</span>),</span>
<span id="cb43-4"><a href="#cb43-4" aria-hidden="true" tabindex="-1"></a>         <span class="at">gender =</span> <span class="fu">str_extract</span>(<span class="at">string =</span> gender,</span>
<span id="cb43-5"><a href="#cb43-5" aria-hidden="true" tabindex="-1"></a>                              <span class="at">pattern =</span> <span class="st">"Female|Male"</span>),</span>
<span id="cb43-6"><a href="#cb43-6" aria-hidden="true" tabindex="-1"></a>         <span class="at">disease_state =</span> <span class="fu">str_extract</span>(<span class="at">string =</span> disease_state,</span>
<span id="cb43-7"><a href="#cb43-7" aria-hidden="true" tabindex="-1"></a>                              <span class="at">pattern =</span> <span class="st">"HIV/TB|HIV"</span>))</span>
<span id="cb43-8"><a href="#cb43-8" aria-hidden="true" tabindex="-1"></a>dat_meta <span class="sc">|&gt;</span> </span>
<span id="cb43-9"><a href="#cb43-9" aria-hidden="true" tabindex="-1"></a>  <span class="fu">sample_n</span>(<span class="dv">10</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code># A tibble: 10 × 5
   patient_n geo_accession gender disease_state identifier             
   &lt;chr&gt;     &lt;chr&gt;         &lt;chr&gt;  &lt;chr&gt;         &lt;chr&gt;                  
 1 21        GSM1230923    Male   HIV           EAA0VKQY-17_18-Mar-2009
 2 14        GSM1230938    Female HIV/TB        HAA0Z4Z0-19_26-Nov-2009
 3 18        GSM1230942    Female HIV/TB        KAA0Z0GV-10_18-Nov-2009
 4 2_1       GSM1230903    Female HIV           AAA0W22R-11_16-Apr-2009
 5 4         GSM1230928    Female HIV/TB        AAA0ZSJX-10_26-Jan-2010
 6 17        GSM1230919    Male   HIV           JAA0W223-10_16-Apr-2009
 7 13        GSM1230915    Male   HIV           FAA0W22D-13_16-Apr-2009
 8 20        GSM1230922    Female HIV           AAA0WVKX-11_04-Jun-2009
 9 5         GSM1230907    Female HIV           BAA0W69B-10_29-Apr-2009
10 3         GSM1230927    Male   HIV/TB        AAA0Z4D1-15_25-Nov-2009</code></pre>
</div>
</div>
<p>The column ‘identifier’ in the meta data contain the names of the genes described in the expression data. In the following, the ‘identifier’ is corrected to mimic the genes of the expression data in order to join the two data sets later on.</p>
<p>The ‘identifier’ in the meta data is split into two columns to simplify the identifier but still keep the information about the dates.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb45"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb45-1"><a href="#cb45-1" aria-hidden="true" tabindex="-1"></a>dat_meta <span class="ot">&lt;-</span> dat_meta <span class="sc">|&gt;</span> </span>
<span id="cb45-2"><a href="#cb45-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">separate</span>(<span class="at">col =</span> identifier,</span>
<span id="cb45-3"><a href="#cb45-3" aria-hidden="true" tabindex="-1"></a>           <span class="at">into =</span> <span class="fu">c</span>(<span class="st">"identifier"</span>, <span class="st">"date"</span>),</span>
<span id="cb45-4"><a href="#cb45-4" aria-hidden="true" tabindex="-1"></a>           <span class="at">sep =</span> <span class="st">"_"</span>)</span>
<span id="cb45-5"><a href="#cb45-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb45-6"><a href="#cb45-6" aria-hidden="true" tabindex="-1"></a>dat_meta <span class="sc">|&gt;</span> </span>
<span id="cb45-7"><a href="#cb45-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">sample_n</span>(<span class="dv">10</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code># A tibble: 10 × 6
   patient_n geo_accession gender disease_state identifier  date       
   &lt;chr&gt;     &lt;chr&gt;         &lt;chr&gt;  &lt;chr&gt;         &lt;chr&gt;       &lt;chr&gt;      
 1 13        GSM1230937    Female HIV/TB        HAA0QL3W-13 29-May-2008
 2 2         GSM1230926    Female HIV/TB        AAA0RWJ0-14 02-Sep-2008
 3 22        GSM1230924    Female HIV           JAA0WVMP-13 04-Jun-2009
 4 11        GSM1230935    Male   HIV/TB        FAA0ZVHT-14 02-Feb-2010
 5 12        GSM1230914    Female HIV           DAA0VD27-11 04-Mar-2009
 6 21        GSM1230945    Male   HIV/TB        EAA0ZNH3-13 13-Jan-2010
 7 9         GSM1230933    Male   HIV/TB        DAA0Z4CR-10 25-Nov-2009
 8 5         GSM1230907    Female HIV           BAA0W69B-10 29-Apr-2009
 9 14        GSM1230938    Female HIV/TB        HAA0Z4Z0-19 26-Nov-2009
10 7         GSM1230909    Female HIV           CAA0Q3CK-13 24-Apr-2008</code></pre>
</div>
</div>
<p>In the gene expression data the gene names contain “-” while the ‘identifier’ in the meta data contain “.”. Replace the “-” in the meta data to streamline the two data sets.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb47"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb47-1"><a href="#cb47-1" aria-hidden="true" tabindex="-1"></a>dat_meta <span class="ot">&lt;-</span> dat_meta <span class="sc">|&gt;</span> </span>
<span id="cb47-2"><a href="#cb47-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">identifier =</span> <span class="fu">str_replace_all</span>(<span class="at">string =</span> identifier,</span>
<span id="cb47-3"><a href="#cb47-3" aria-hidden="true" tabindex="-1"></a>                                      <span class="at">pattern =</span> <span class="st">"-"</span>,</span>
<span id="cb47-4"><a href="#cb47-4" aria-hidden="true" tabindex="-1"></a>                                      <span class="at">replacement =</span> <span class="st">"."</span>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>Check if there is consensus between the ‘identifiers’ in the meta data and the column names (gene names) of the gene expression data.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb48"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb48-1"><a href="#cb48-1" aria-hidden="true" tabindex="-1"></a>meta_identifier <span class="ot">&lt;-</span> dat_meta <span class="sc">|&gt;</span></span>
<span id="cb48-2"><a href="#cb48-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">pull</span>(identifier)</span>
<span id="cb48-3"><a href="#cb48-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb48-4"><a href="#cb48-4" aria-hidden="true" tabindex="-1"></a><span class="fu">setdiff</span>(<span class="fu">colnames</span>(dat_expr), meta_identifier)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] "ID"</code></pre>
</div>
</div>
<p>The only difference between ‘identifier’ in meta data and column names of the gene expression data is the ‘ID’ column for the genes as expected.</p>
</section>
</section>
<section id="data-wrangling-ii" class="level2">
<h2 class="anchored" data-anchor-id="data-wrangling-ii">Data Wrangling II</h2>
<p>In the following steps, the format of the data will be changed. Furthermore, we wish to join the gene expression data and the meta data.</p>
<section id="data-pivoting" class="level3">
<h3 class="anchored" data-anchor-id="data-pivoting">Data Pivoting</h3>
<p>The gene expression data is transposed by pivoting sample ‘identifier’ to long format and gene ‘ID’ to wide format. The data should be tidy, which requires each observation, that is, each patient, must be a row, with the variables as columns.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb50"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb50-1"><a href="#cb50-1" aria-hidden="true" tabindex="-1"></a>dat_expr <span class="ot">&lt;-</span> dat_expr <span class="sc">|&gt;</span> </span>
<span id="cb50-2"><a href="#cb50-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">pivot_longer</span>(<span class="at">cols =</span> <span class="sc">!</span>ID,</span>
<span id="cb50-3"><a href="#cb50-3" aria-hidden="true" tabindex="-1"></a>               <span class="at">names_to =</span> <span class="st">"identifier"</span>,</span>
<span id="cb50-4"><a href="#cb50-4" aria-hidden="true" tabindex="-1"></a>               <span class="at">values_to =</span> <span class="st">"count"</span>) <span class="sc">|&gt;</span> </span>
<span id="cb50-5"><a href="#cb50-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">pivot_wider</span>(<span class="at">names_from =</span> ID,</span>
<span id="cb50-6"><a href="#cb50-6" aria-hidden="true" tabindex="-1"></a>              <span class="at">values_from =</span> count)</span>
<span id="cb50-7"><a href="#cb50-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb50-8"><a href="#cb50-8" aria-hidden="true" tabindex="-1"></a>dat_expr <span class="sc">|&gt;</span> </span>
<span id="cb50-9"><a href="#cb50-9" aria-hidden="true" tabindex="-1"></a>  <span class="fu">sample_n</span>(<span class="dv">10</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code># A tibble: 10 × 47,232
   identifier  ILMN_1762337 ILMN_2055271 ILMN_1736007 ILMN_2383229 ILMN_1806310
   &lt;chr&gt;              &lt;dbl&gt;        &lt;dbl&gt;        &lt;dbl&gt;        &lt;dbl&gt;        &lt;dbl&gt;
 1 AAA0W22R.11         158.         149.         149.         156.         142.
 2 KAA0ZQ5C.11         279.         257.         293.         224.         233.
 3 DAA0Z4CR.10         332.         264.         274.         341.         318.
 4 CAA0W4LJ.15         305.         335.         343.         293.         282.
 5 KAA0PBTZ.17         216.         260.         218.         224.         180.
 6 KAA0ZMKX.10         338.         305.         266.         342.         272.
 7 BAA0W69B.10         158.         141.         167.         159.         137.
 8 BAA0W3BM.11         289.         333.         294.         259.         258.
 9 HAA0W3B8.14         141.         157.         175.         159.         124.
10 DAA0PFCT.11         287.         308.         275.         276.         244.
# ℹ 47,226 more variables: ILMN_1779670 &lt;dbl&gt;, ILMN_1653355 &lt;dbl&gt;,
#   ILMN_1717783 &lt;dbl&gt;, ILMN_1705025 &lt;dbl&gt;, ILMN_1814316 &lt;dbl&gt;,
#   ILMN_2359168 &lt;dbl&gt;, ILMN_1731507 &lt;dbl&gt;, ILMN_1787689 &lt;dbl&gt;,
#   ILMN_3241953 &lt;dbl&gt;, ILMN_1745607 &lt;dbl&gt;, ILMN_2136495 &lt;dbl&gt;,
#   ILMN_1668111 &lt;dbl&gt;, ILMN_2295559 &lt;dbl&gt;, ILMN_1735045 &lt;dbl&gt;,
#   ILMN_1680754 &lt;dbl&gt;, ILMN_2375184 &lt;dbl&gt;, ILMN_1659452 &lt;dbl&gt;,
#   ILMN_1767388 &lt;dbl&gt;, ILMN_1675204 &lt;dbl&gt;, ILMN_1673870 &lt;dbl&gt;, …</code></pre>
</div>
</div>
</section>
<section id="data-joining" class="level3">
<h3 class="anchored" data-anchor-id="data-joining">Data Joining</h3>
<p>The meta data and the count data should be merge such that all data about a patient is contained in one tibble. Meta data should be in the first columns followed by gene variables.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb52"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb52-1"><a href="#cb52-1" aria-hidden="true" tabindex="-1"></a>dat_clean <span class="ot">&lt;-</span> <span class="fu">inner_join</span>(<span class="at">x =</span> dat_expr,</span>
<span id="cb52-2"><a href="#cb52-2" aria-hidden="true" tabindex="-1"></a>                        <span class="at">y =</span> dat_meta,</span>
<span id="cb52-3"><a href="#cb52-3" aria-hidden="true" tabindex="-1"></a>                        <span class="at">by =</span> <span class="st">"identifier"</span>) <span class="sc">|&gt;</span> </span>
<span id="cb52-4"><a href="#cb52-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">relocate</span>(<span class="fu">colnames</span>(dat_meta))</span>
<span id="cb52-5"><a href="#cb52-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb52-6"><a href="#cb52-6" aria-hidden="true" tabindex="-1"></a>dat_clean <span class="sc">|&gt;</span> </span>
<span id="cb52-7"><a href="#cb52-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">sample_n</span>(<span class="dv">10</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code># A tibble: 10 × 47,237
   patient_n geo_accession gender disease_state identifier  date    ILMN_1762337
   &lt;chr&gt;     &lt;chr&gt;         &lt;chr&gt;  &lt;chr&gt;         &lt;chr&gt;       &lt;chr&gt;          &lt;dbl&gt;
 1 9         GSM1230933    Male   HIV/TB        DAA0Z4CR.10 25-Nov…         332.
 2 17        GSM1230919    Male   HIV           JAA0W223.10 16-Apr…         166.
 3 2_1       GSM1230903    Female HIV           AAA0W22R.11 16-Apr…         158.
 4 14        GSM1230916    Female HIV           HAA0W3B8.14 20-Apr…         141.
 5 22        GSM1230924    Female HIV           JAA0WVMP.13 04-Jun…         260.
 6 5         GSM1230907    Female HIV           BAA0W69B.10 29-Apr…         158.
 7 18        GSM1230920    Female HIV           KAA0PBTZ.17 06-Mar…         216.
 8 13        GSM1230937    Female HIV/TB        HAA0QL3W.13 29-May…         222.
 9 21        GSM1230923    Male   HIV           EAA0VKQY.17 18-Mar…         318.
10 4         GSM1230906    Female HIV           BAA0W3BM.11 20-Apr…         289.
# ℹ 47,230 more variables: ILMN_2055271 &lt;dbl&gt;, ILMN_1736007 &lt;dbl&gt;,
#   ILMN_2383229 &lt;dbl&gt;, ILMN_1806310 &lt;dbl&gt;, ILMN_1779670 &lt;dbl&gt;,
#   ILMN_1653355 &lt;dbl&gt;, ILMN_1717783 &lt;dbl&gt;, ILMN_1705025 &lt;dbl&gt;,
#   ILMN_1814316 &lt;dbl&gt;, ILMN_2359168 &lt;dbl&gt;, ILMN_1731507 &lt;dbl&gt;,
#   ILMN_1787689 &lt;dbl&gt;, ILMN_3241953 &lt;dbl&gt;, ILMN_1745607 &lt;dbl&gt;,
#   ILMN_2136495 &lt;dbl&gt;, ILMN_1668111 &lt;dbl&gt;, ILMN_2295559 &lt;dbl&gt;,
#   ILMN_1735045 &lt;dbl&gt;, ILMN_1680754 &lt;dbl&gt;, ILMN_2375184 &lt;dbl&gt;, …</code></pre>
</div>
</div>
</section>
<section id="handling-replicates" class="level3">
<h3 class="anchored" data-anchor-id="handling-replicates">Handling Replicates</h3>
<p>There are two technical replicates available for one patient, namely patient 2_1 and 2_2 indicated by the ‘patient_n’ column. We wish to averaged the measurements for this patient for further analysis.</p>
<p>First, the gene expression values for the replicates are extracted.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb54"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb54-1"><a href="#cb54-1" aria-hidden="true" tabindex="-1"></a>dat_rep <span class="ot">&lt;-</span> dat_clean <span class="sc">|&gt;</span></span>
<span id="cb54-2"><a href="#cb54-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(<span class="fu">str_detect</span>(patient_n, <span class="st">"_"</span>)) <span class="sc">|&gt;</span></span>
<span id="cb54-3"><a href="#cb54-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">patient_n =</span> <span class="fu">str_extract</span>(<span class="at">string =</span> patient_n, </span>
<span id="cb54-4"><a href="#cb54-4" aria-hidden="true" tabindex="-1"></a>                                 <span class="at">pattern =</span> <span class="st">"[0-9]+"</span>)) </span>
<span id="cb54-5"><a href="#cb54-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb54-6"><a href="#cb54-6" aria-hidden="true" tabindex="-1"></a>dat_rep </span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code># A tibble: 2 × 47,237
  patient_n geo_accession gender disease_state identifier  date     ILMN_1762337
  &lt;chr&gt;     &lt;chr&gt;         &lt;chr&gt;  &lt;chr&gt;         &lt;chr&gt;       &lt;chr&gt;           &lt;dbl&gt;
1 2         GSM1230903    Female HIV           AAA0W22R.11 16-Apr-…         158.
2 2         GSM1230904    Female HIV           AAA0W22R.14 16-Apr-…         300.
# ℹ 47,230 more variables: ILMN_2055271 &lt;dbl&gt;, ILMN_1736007 &lt;dbl&gt;,
#   ILMN_2383229 &lt;dbl&gt;, ILMN_1806310 &lt;dbl&gt;, ILMN_1779670 &lt;dbl&gt;,
#   ILMN_1653355 &lt;dbl&gt;, ILMN_1717783 &lt;dbl&gt;, ILMN_1705025 &lt;dbl&gt;,
#   ILMN_1814316 &lt;dbl&gt;, ILMN_2359168 &lt;dbl&gt;, ILMN_1731507 &lt;dbl&gt;,
#   ILMN_1787689 &lt;dbl&gt;, ILMN_3241953 &lt;dbl&gt;, ILMN_1745607 &lt;dbl&gt;,
#   ILMN_2136495 &lt;dbl&gt;, ILMN_1668111 &lt;dbl&gt;, ILMN_2295559 &lt;dbl&gt;,
#   ILMN_1735045 &lt;dbl&gt;, ILMN_1680754 &lt;dbl&gt;, ILMN_2375184 &lt;dbl&gt;, …</code></pre>
</div>
</div>
<p>The mean gene expression is calculated for each replicated patient.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb56"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb56-1"><a href="#cb56-1" aria-hidden="true" tabindex="-1"></a>dat_rep_mean <span class="ot">&lt;-</span> dat_rep <span class="sc">|&gt;</span> </span>
<span id="cb56-2"><a href="#cb56-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(patient_n) <span class="sc">|&gt;</span></span>
<span id="cb56-3"><a href="#cb56-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarise</span>(<span class="fu">across</span>(<span class="at">.cols =</span> <span class="fu">starts_with</span>(<span class="st">"ILMN"</span>),</span>
<span id="cb56-4"><a href="#cb56-4" aria-hidden="true" tabindex="-1"></a>                   <span class="at">.fns =</span> mean,</span>
<span id="cb56-5"><a href="#cb56-5" aria-hidden="true" tabindex="-1"></a>                   <span class="at">na.rm =</span> <span class="cn">TRUE</span>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: There was 1 warning in `summarise()`.
ℹ In argument: `across(.cols = starts_with("ILMN"), .fns = mean, na.rm =
  TRUE)`.
ℹ In group 1: `patient_n = "2"`.
Caused by warning:
! The `...` argument of `across()` is deprecated as of dplyr 1.1.0.
Supply arguments directly to `.fns` through an anonymous function instead.

  # Previously
  across(a:b, mean, na.rm = TRUE)

  # Now
  across(a:b, \(x) mean(x, na.rm = TRUE))</code></pre>
</div>
<div class="sourceCode cell-code" id="cb58"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb58-1"><a href="#cb58-1" aria-hidden="true" tabindex="-1"></a>dat_rep_mean</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code># A tibble: 1 × 47,232
  patient_n ILMN_1762337 ILMN_2055271 ILMN_1736007 ILMN_2383229 ILMN_1806310
  &lt;chr&gt;            &lt;dbl&gt;        &lt;dbl&gt;        &lt;dbl&gt;        &lt;dbl&gt;        &lt;dbl&gt;
1 2                 229.         209.         200.         192.         189.
# ℹ 47,226 more variables: ILMN_1779670 &lt;dbl&gt;, ILMN_1653355 &lt;dbl&gt;,
#   ILMN_1717783 &lt;dbl&gt;, ILMN_1705025 &lt;dbl&gt;, ILMN_1814316 &lt;dbl&gt;,
#   ILMN_2359168 &lt;dbl&gt;, ILMN_1731507 &lt;dbl&gt;, ILMN_1787689 &lt;dbl&gt;,
#   ILMN_3241953 &lt;dbl&gt;, ILMN_1745607 &lt;dbl&gt;, ILMN_2136495 &lt;dbl&gt;,
#   ILMN_1668111 &lt;dbl&gt;, ILMN_2295559 &lt;dbl&gt;, ILMN_1735045 &lt;dbl&gt;,
#   ILMN_1680754 &lt;dbl&gt;, ILMN_2375184 &lt;dbl&gt;, ILMN_1659452 &lt;dbl&gt;,
#   ILMN_1767388 &lt;dbl&gt;, ILMN_1675204 &lt;dbl&gt;, ILMN_1673870 &lt;dbl&gt;, …</code></pre>
</div>
</div>
<p>The metadata for each replicated patient is extracted. The ‘geo accession’ and ‘identifer’ is collapsed, meaning that the new ‘geo accession’ and ‘identifer’ for the replicated patient is a combination of the originals.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb60"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb60-1"><a href="#cb60-1" aria-hidden="true" tabindex="-1"></a>cols_rep_meta_collapesed <span class="ot">&lt;-</span> dat_rep <span class="sc">|&gt;</span></span>
<span id="cb60-2"><a href="#cb60-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(patient_n) <span class="sc">|&gt;</span> </span>
<span id="cb60-3"><a href="#cb60-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">geo_accession =</span> <span class="fu">str_c</span>(geo_accession, <span class="at">collapse =</span> <span class="st">"_"</span>),</span>
<span id="cb60-4"><a href="#cb60-4" aria-hidden="true" tabindex="-1"></a>            <span class="at">identifier =</span> <span class="fu">str_c</span>(identifier, <span class="at">collapse =</span> <span class="st">"_"</span>))</span>
<span id="cb60-5"><a href="#cb60-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb60-6"><a href="#cb60-6" aria-hidden="true" tabindex="-1"></a>cols_rep_meta_collapesed</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code># A tibble: 2 × 47,237
# Groups:   patient_n [1]
  patient_n geo_accession     gender disease_state identifier date  ILMN_1762337
  &lt;chr&gt;     &lt;chr&gt;             &lt;chr&gt;  &lt;chr&gt;         &lt;chr&gt;      &lt;chr&gt;        &lt;dbl&gt;
1 2         GSM1230903_GSM12… Female HIV           AAA0W22R.… 16-A…         158.
2 2         GSM1230903_GSM12… Female HIV           AAA0W22R.… 16-A…         300.
# ℹ 47,230 more variables: ILMN_2055271 &lt;dbl&gt;, ILMN_1736007 &lt;dbl&gt;,
#   ILMN_2383229 &lt;dbl&gt;, ILMN_1806310 &lt;dbl&gt;, ILMN_1779670 &lt;dbl&gt;,
#   ILMN_1653355 &lt;dbl&gt;, ILMN_1717783 &lt;dbl&gt;, ILMN_1705025 &lt;dbl&gt;,
#   ILMN_1814316 &lt;dbl&gt;, ILMN_2359168 &lt;dbl&gt;, ILMN_1731507 &lt;dbl&gt;,
#   ILMN_1787689 &lt;dbl&gt;, ILMN_3241953 &lt;dbl&gt;, ILMN_1745607 &lt;dbl&gt;,
#   ILMN_2136495 &lt;dbl&gt;, ILMN_1668111 &lt;dbl&gt;, ILMN_2295559 &lt;dbl&gt;,
#   ILMN_1735045 &lt;dbl&gt;, ILMN_1680754 &lt;dbl&gt;, ILMN_2375184 &lt;dbl&gt;, …</code></pre>
</div>
</div>
<p>A new column called ‘modified’ is added to the meta data. The column describes if the patient has been modified (eg. the gene expression represent a mean of multiple gene expressions for one patient).</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb62"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb62-1"><a href="#cb62-1" aria-hidden="true" tabindex="-1"></a>cols_rep_meta <span class="ot">&lt;-</span> cols_rep_meta_collapesed <span class="sc">|&gt;</span></span>
<span id="cb62-2"><a href="#cb62-2" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">select</span>(<span class="sc">!</span><span class="fu">starts_with</span>(<span class="st">"ILMN"</span>)) <span class="sc">|&gt;</span> </span>
<span id="cb62-3"><a href="#cb62-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">distinct</span>() <span class="sc">|&gt;</span> </span>
<span id="cb62-4"><a href="#cb62-4" aria-hidden="true" tabindex="-1"></a> <span class="fu">mutate</span>(<span class="at">modified =</span> <span class="dv">1</span>)</span>
<span id="cb62-5"><a href="#cb62-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb62-6"><a href="#cb62-6" aria-hidden="true" tabindex="-1"></a>cols_rep_meta</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code># A tibble: 1 × 7
# Groups:   patient_n [1]
  patient_n geo_accession         gender disease_state identifier date  modified
  &lt;chr&gt;     &lt;chr&gt;                 &lt;chr&gt;  &lt;chr&gt;         &lt;chr&gt;      &lt;chr&gt;    &lt;dbl&gt;
1 2         GSM1230903_GSM1230904 Female HIV           AAA0W22R.… 16-A…        1</code></pre>
</div>
<div class="sourceCode cell-code" id="cb64"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb64-1"><a href="#cb64-1" aria-hidden="true" tabindex="-1"></a>dat_clean <span class="ot">&lt;-</span> dat_clean <span class="sc">|&gt;</span></span>
<span id="cb64-2"><a href="#cb64-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">modified =</span> <span class="dv">0</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>Combine the new meta data with the mean gene expression for the each replicated patient.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb65"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb65-1"><a href="#cb65-1" aria-hidden="true" tabindex="-1"></a>dat_rep_clean <span class="ot">&lt;-</span> <span class="fu">inner_join</span>(<span class="at">x =</span> cols_rep_meta, </span>
<span id="cb65-2"><a href="#cb65-2" aria-hidden="true" tabindex="-1"></a>                            <span class="at">y =</span> dat_rep_mean,</span>
<span id="cb65-3"><a href="#cb65-3" aria-hidden="true" tabindex="-1"></a>                            <span class="at">by =</span> <span class="st">"patient_n"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>Finally, the new data for th replicated patients are combines with the data for the non-replicated patients to form a clean data set without replicates.</p>
<p>As the patient_n variable is no longer informative, this column is removed for all patients.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb66"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb66-1"><a href="#cb66-1" aria-hidden="true" tabindex="-1"></a>dat_clean <span class="ot">&lt;-</span> <span class="fu">full_join</span>(dat_clean, dat_rep_clean) <span class="sc">|&gt;</span></span>
<span id="cb66-2"><a href="#cb66-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">relocate</span>(modified, <span class="at">.after =</span> date) <span class="sc">|&gt;</span></span>
<span id="cb66-3"><a href="#cb66-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(<span class="sc">!</span><span class="fu">str_detect</span>(patient_n, <span class="st">"_"</span>)) <span class="sc">|&gt;</span>  </span>
<span id="cb66-4"><a href="#cb66-4" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">select</span>(<span class="sc">!</span>patient_n) <span class="sc">|&gt;</span> </span>
<span id="cb66-5"><a href="#cb66-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">arrange</span>(disease_state) </span>
<span id="cb66-6"><a href="#cb66-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb66-7"><a href="#cb66-7" aria-hidden="true" tabindex="-1"></a>dat_clean <span class="sc">|&gt;</span> </span>
<span id="cb66-8"><a href="#cb66-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">sample_n</span>(<span class="dv">10</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code># A tibble: 10 × 47,237
   geo_accession gender disease_state identifier  date     modified ILMN_1762337
   &lt;chr&gt;         &lt;chr&gt;  &lt;chr&gt;         &lt;chr&gt;       &lt;chr&gt;       &lt;dbl&gt;        &lt;dbl&gt;
 1 GSM1230927    Male   HIV/TB        AAA0Z4D1.15 25-Nov-…        0         318.
 2 GSM1230908    Female HIV           BAA0WVPG.10 04-Jun-…        0         157.
 3 GSM1230912    Female HIV           DAA0PFCT.11 17-Jan-…        0         287.
 4 GSM1230925    Female HIV/TB        AAA0KHPN.10 14-Jun-…        0         328.
 5 GSM1230936    Male   HIV/TB        GAA0S8NL.11 30-Sep-…        0         212.
 6 GSM1230916    Female HIV           HAA0W3B8.14 20-Apr-…        0         141.
 7 GSM1230924    Female HIV           JAA0WVMP.13 04-Jun-…        0         260.
 8 GSM1230932    Male   HIV/TB        CAA0ZN6L.12 13-Jan-…        0         342.
 9 GSM1230926    Female HIV/TB        AAA0RWJ0.14 02-Sep-…        0         283.
10 GSM1230931    Female HIV/TB        BAA0ZXTF.11 08-Feb-…        0         121.
# ℹ 47,230 more variables: ILMN_2055271 &lt;dbl&gt;, ILMN_1736007 &lt;dbl&gt;,
#   ILMN_2383229 &lt;dbl&gt;, ILMN_1806310 &lt;dbl&gt;, ILMN_1779670 &lt;dbl&gt;,
#   ILMN_1653355 &lt;dbl&gt;, ILMN_1717783 &lt;dbl&gt;, ILMN_1705025 &lt;dbl&gt;,
#   ILMN_1814316 &lt;dbl&gt;, ILMN_2359168 &lt;dbl&gt;, ILMN_1731507 &lt;dbl&gt;,
#   ILMN_1787689 &lt;dbl&gt;, ILMN_3241953 &lt;dbl&gt;, ILMN_1745607 &lt;dbl&gt;,
#   ILMN_2136495 &lt;dbl&gt;, ILMN_1668111 &lt;dbl&gt;, ILMN_2295559 &lt;dbl&gt;,
#   ILMN_1735045 &lt;dbl&gt;, ILMN_1680754 &lt;dbl&gt;, ILMN_2375184 &lt;dbl&gt;, …</code></pre>
</div>
</div>
</section>
</section>
<section id="write-data-to-file" class="level2">
<h2 class="anchored" data-anchor-id="write-data-to-file">Write Data to File</h2>
<p>The cleaned data is written as a tsv file.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb68"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb68-1"><a href="#cb68-1" aria-hidden="true" tabindex="-1"></a>dat_clean <span class="sc">|&gt;</span> </span>
<span id="cb68-2"><a href="#cb68-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">write_tsv</span>(<span class="at">file =</span> <span class="st">"../data/02_dat_clean.tsv"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="setup-2" class="level2">
<h2 class="anchored" data-anchor-id="setup-2">Setup</h2>
<p>Loading required packages</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb69"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb69-1"><a href="#cb69-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(<span class="st">"tidyverse"</span>)</span>
<span id="cb69-2"><a href="#cb69-2" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(<span class="st">"stringr"</span>)</span>
<span id="cb69-3"><a href="#cb69-3" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(<span class="st">"table1"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>Loading the cleaned data.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb70"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb70-1"><a href="#cb70-1" aria-hidden="true" tabindex="-1"></a>dat_clean <span class="ot">&lt;-</span> <span class="fu">read_tsv</span>(<span class="st">"../data/02_dat_clean.tsv"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="data-description" class="level2">
<h2 class="anchored" data-anchor-id="data-description">Data Description</h2>
<p>Define setting for bar plot to illustrate the number of males and females for each patient group.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb71"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb71-1"><a href="#cb71-1" aria-hidden="true" tabindex="-1"></a>female_sign <span class="ot">=</span> <span class="fu">intToUtf8</span>(<span class="dv">9792</span>)</span>
<span id="cb71-2"><a href="#cb71-2" aria-hidden="true" tabindex="-1"></a>male_sign <span class="ot">=</span> <span class="fu">intToUtf8</span>(<span class="dv">9794</span>)</span>
<span id="cb71-3"><a href="#cb71-3" aria-hidden="true" tabindex="-1"></a>sign_size <span class="ot">=</span> <span class="dv">10</span></span>
<span id="cb71-4"><a href="#cb71-4" aria-hidden="true" tabindex="-1"></a>bar_width <span class="ot">=</span> <span class="fl">0.75</span></span>
<span id="cb71-5"><a href="#cb71-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb71-6"><a href="#cb71-6" aria-hidden="true" tabindex="-1"></a>groups <span class="ot">&lt;-</span> dat_clean <span class="sc">|&gt;</span> </span>
<span id="cb71-7"><a href="#cb71-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(gender,</span>
<span id="cb71-8"><a href="#cb71-8" aria-hidden="true" tabindex="-1"></a>           disease_state) <span class="sc">|&gt;</span> </span>
<span id="cb71-9"><a href="#cb71-9" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarize</span>(<span class="at">n =</span> <span class="fu">n</span>())</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>`summarise()` has grouped output by 'gender'. You can override using the
`.groups` argument.</code></pre>
</div>
</div>
<p>Create plot.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb73"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb73-1"><a href="#cb73-1" aria-hidden="true" tabindex="-1"></a>barplot_patient_group <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(<span class="at">data =</span> dat_clean,</span>
<span id="cb73-2"><a href="#cb73-2" aria-hidden="true" tabindex="-1"></a>                                <span class="at">mapping =</span> <span class="fu">aes</span>(<span class="at">x =</span> disease_state,</span>
<span id="cb73-3"><a href="#cb73-3" aria-hidden="true" tabindex="-1"></a>                                              <span class="at">fill =</span> disease_state,</span>
<span id="cb73-4"><a href="#cb73-4" aria-hidden="true" tabindex="-1"></a>                                              <span class="at">alpha =</span> gender)) <span class="sc">+</span></span>
<span id="cb73-5"><a href="#cb73-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_bar</span>(<span class="at">width =</span> bar_width,</span>
<span id="cb73-6"><a href="#cb73-6" aria-hidden="true" tabindex="-1"></a>           <span class="at">position =</span> <span class="st">"dodge"</span>) <span class="sc">+</span></span>
<span id="cb73-7"><a href="#cb73-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_fill_manual</span>(</span>
<span id="cb73-8"><a href="#cb73-8" aria-hidden="true" tabindex="-1"></a>    <span class="at">values =</span> <span class="fu">c</span>(<span class="st">"HIV"</span> <span class="ot">=</span> <span class="st">"royalblue"</span>, </span>
<span id="cb73-9"><a href="#cb73-9" aria-hidden="true" tabindex="-1"></a>               <span class="st">"HIV/TB"</span> <span class="ot">=</span> <span class="st">"tomato"</span>) ) <span class="sc">+</span></span>
<span id="cb73-10"><a href="#cb73-10" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_alpha_manual</span>(<span class="at">values =</span> <span class="fu">c</span>(<span class="st">"Male"</span> <span class="ot">=</span> <span class="fl">0.6</span>,</span>
<span id="cb73-11"><a href="#cb73-11" aria-hidden="true" tabindex="-1"></a>                                <span class="st">"Female"</span> <span class="ot">=</span> <span class="fl">0.9</span>)) <span class="sc">+</span></span>
<span id="cb73-12"><a href="#cb73-12" aria-hidden="true" tabindex="-1"></a>  <span class="fu">annotate</span>(<span class="st">"text"</span>, </span>
<span id="cb73-13"><a href="#cb73-13" aria-hidden="true" tabindex="-1"></a>           <span class="at">x =</span> <span class="dv">1</span><span class="sc">:</span><span class="dv">2</span> <span class="sc">-</span> bar_width<span class="sc">/</span><span class="dv">4</span>, </span>
<span id="cb73-14"><a href="#cb73-14" aria-hidden="true" tabindex="-1"></a>           <span class="at">y =</span> groups<span class="sc">$</span>n[<span class="dv">1</span><span class="sc">:</span><span class="dv">2</span>] <span class="sc">+</span> <span class="dv">2</span>, </span>
<span id="cb73-15"><a href="#cb73-15" aria-hidden="true" tabindex="-1"></a>           <span class="at">label =</span> female_sign, </span>
<span id="cb73-16"><a href="#cb73-16" aria-hidden="true" tabindex="-1"></a>           <span class="at">size =</span> sign_size) <span class="sc">+</span></span>
<span id="cb73-17"><a href="#cb73-17" aria-hidden="true" tabindex="-1"></a>  <span class="fu">annotate</span>(<span class="st">"text"</span>, </span>
<span id="cb73-18"><a href="#cb73-18" aria-hidden="true" tabindex="-1"></a>           <span class="at">x =</span> <span class="dv">1</span><span class="sc">:</span><span class="dv">2</span> <span class="sc">+</span> bar_width<span class="sc">/</span><span class="dv">4</span>,</span>
<span id="cb73-19"><a href="#cb73-19" aria-hidden="true" tabindex="-1"></a>           <span class="at">y =</span> groups<span class="sc">$</span>n[<span class="dv">3</span><span class="sc">:</span><span class="dv">4</span>] <span class="sc">+</span> <span class="dv">2</span>, </span>
<span id="cb73-20"><a href="#cb73-20" aria-hidden="true" tabindex="-1"></a>           <span class="at">label =</span> male_sign, </span>
<span id="cb73-21"><a href="#cb73-21" aria-hidden="true" tabindex="-1"></a>           <span class="at">size =</span> sign_size) <span class="sc">+</span></span>
<span id="cb73-22"><a href="#cb73-22" aria-hidden="true" tabindex="-1"></a>  <span class="fu">coord_cartesian</span>(<span class="at">ylim =</span>  <span class="fu">c</span>(<span class="dv">0</span>, <span class="dv">20</span>)) <span class="sc">+</span></span>
<span id="cb73-23"><a href="#cb73-23" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_minimal</span>(<span class="at">base_size =</span> <span class="dv">16</span>) <span class="sc">+</span> </span>
<span id="cb73-24"><a href="#cb73-24" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(</span>
<span id="cb73-25"><a href="#cb73-25" aria-hidden="true" tabindex="-1"></a>    <span class="at">x =</span> <span class="st">"Disease state"</span>,</span>
<span id="cb73-26"><a href="#cb73-26" aria-hidden="true" tabindex="-1"></a>    <span class="at">y =</span> <span class="st">"Cases"</span>,</span>
<span id="cb73-27"><a href="#cb73-27" aria-hidden="true" tabindex="-1"></a>    <span class="at">fill =</span> <span class="st">"Gender"</span>,</span>
<span id="cb73-28"><a href="#cb73-28" aria-hidden="true" tabindex="-1"></a>    <span class="at">title =</span> <span class="st">"Patient Summary"</span>,</span>
<span id="cb73-29"><a href="#cb73-29" aria-hidden="true" tabindex="-1"></a>    <span class="at">subtitle =</span> <span class="fu">str_c</span>(<span class="st">"Distribution of males and females for HIV mono-infected and"</span>,</span>
<span id="cb73-30"><a href="#cb73-30" aria-hidden="true" tabindex="-1"></a>                     <span class="st">"</span><span class="sc">\n</span><span class="st">"</span>,</span>
<span id="cb73-31"><a href="#cb73-31" aria-hidden="true" tabindex="-1"></a>                     <span class="st">"HIV/TB co-infected patients"</span>),</span>
<span id="cb73-32"><a href="#cb73-32" aria-hidden="true" tabindex="-1"></a>    <span class="at">caption =</span> <span class="st">"Data from Dawany et al. (2014)"</span></span>
<span id="cb73-33"><a href="#cb73-33" aria-hidden="true" tabindex="-1"></a>  ) <span class="sc">+</span></span>
<span id="cb73-34"><a href="#cb73-34" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">legend.position =</span> <span class="st">"none"</span>,</span>
<span id="cb73-35"><a href="#cb73-35" aria-hidden="true" tabindex="-1"></a>        <span class="at">axis.text.x =</span> <span class="fu">element_text</span>(<span class="at">color =</span> <span class="st">"black"</span>, </span>
<span id="cb73-36"><a href="#cb73-36" aria-hidden="true" tabindex="-1"></a>                                   <span class="at">size =</span> <span class="dv">14</span>),</span>
<span id="cb73-37"><a href="#cb73-37" aria-hidden="true" tabindex="-1"></a>        <span class="at">axis.text.y =</span> <span class="fu">element_text</span>(<span class="at">color =</span> <span class="st">"black"</span>, </span>
<span id="cb73-38"><a href="#cb73-38" aria-hidden="true" tabindex="-1"></a>                                   <span class="at">size =</span> <span class="dv">14</span>),</span>
<span id="cb73-39"><a href="#cb73-39" aria-hidden="true" tabindex="-1"></a>        <span class="at">axis.title =</span> <span class="fu">element_text</span>(<span class="at">face=</span><span class="st">"bold"</span>),</span>
<span id="cb73-40"><a href="#cb73-40" aria-hidden="true" tabindex="-1"></a>        <span class="at">plot.caption =</span> <span class="fu">element_text</span>(<span class="at">face =</span> <span class="st">"italic"</span>))</span>
<span id="cb73-41"><a href="#cb73-41" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb73-42"><a href="#cb73-42" aria-hidden="true" tabindex="-1"></a><span class="fu">ggsave</span>(<span class="st">"../results/barplot_patient_group.png"</span>, <span class="at">plot =</span> barplot_patient_group)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Saving 7 x 5 in image</code></pre>
</div>
<div class="sourceCode cell-code" id="cb75"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb75-1"><a href="#cb75-1" aria-hidden="true" tabindex="-1"></a><span class="fu">show</span>(barplot_patient_group)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<p><img src="00_all_files/figure-html/03_Bar%20plot-1.png" class="img-fluid" width="672"></p>
</div>
</div>
<p>The distribution of males and females for each patient group can also be summarized in a table.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb76"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb76-1"><a href="#cb76-1" aria-hidden="true" tabindex="-1"></a>summary_table <span class="ot">&lt;-</span> dat_clean <span class="sc">|&gt;</span></span>
<span id="cb76-2"><a href="#cb76-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">Gender =</span> <span class="fu">factor</span>(gender),</span>
<span id="cb76-3"><a href="#cb76-3" aria-hidden="true" tabindex="-1"></a>         <span class="at">Disease =</span> <span class="fu">factor</span>(disease_state)) <span class="sc">|&gt;</span> </span>
<span id="cb76-4"><a href="#cb76-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">table1</span>(<span class="at">x =</span> <span class="fu">formula</span>(<span class="sc">~</span> Gender <span class="sc">|</span> Disease))</span>
<span id="cb76-5"><a href="#cb76-5" aria-hidden="true" tabindex="-1"></a><span class="fu">show</span>(summary_table)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>&lt;table class="Rtable1"&gt;
&lt;thead&gt;
&lt;tr&gt;
&lt;th class='rowlabel firstrow lastrow'&gt;&lt;/th&gt;
&lt;th class='firstrow lastrow'&gt;&lt;span class='stratlabel'&gt;HIV&lt;br&gt;&lt;span class='stratn'&gt;(N=22)&lt;/span&gt;&lt;/span&gt;&lt;/th&gt;
&lt;th class='firstrow lastrow'&gt;&lt;span class='stratlabel'&gt;HIV/TB&lt;br&gt;&lt;span class='stratn'&gt;(N=21)&lt;/span&gt;&lt;/span&gt;&lt;/th&gt;
&lt;th class='firstrow lastrow'&gt;&lt;span class='stratlabel'&gt;Overall&lt;br&gt;&lt;span class='stratn'&gt;(N=43)&lt;/span&gt;&lt;/span&gt;&lt;/th&gt;
&lt;/tr&gt;
&lt;/thead&gt;
&lt;tbody&gt;
&lt;tr&gt;
&lt;td class='rowlabel firstrow'&gt;Gender&lt;/td&gt;
&lt;td class='firstrow'&gt;&lt;/td&gt;
&lt;td class='firstrow'&gt;&lt;/td&gt;
&lt;td class='firstrow'&gt;&lt;/td&gt;
&lt;/tr&gt;
&lt;tr&gt;
&lt;td class='rowlabel'&gt;Female&lt;/td&gt;
&lt;td&gt;16 (72.7%)&lt;/td&gt;
&lt;td&gt;11 (52.4%)&lt;/td&gt;
&lt;td&gt;27 (62.8%)&lt;/td&gt;
&lt;/tr&gt;
&lt;tr&gt;
&lt;td class='rowlabel lastrow'&gt;Male&lt;/td&gt;
&lt;td class='lastrow'&gt;6 (27.3%)&lt;/td&gt;
&lt;td class='lastrow'&gt;10 (47.6%)&lt;/td&gt;
&lt;td class='lastrow'&gt;16 (37.2%)&lt;/td&gt;
&lt;/tr&gt;
&lt;/tbody&gt;
&lt;/table&gt;</code></pre>
</div>
</div>
<p>From the above it is seen, that the gender of the HIV/TB co-infected patients is close to equal distributed between male and female, and that the gender of the HIV mono-infected is unequal distributed with 72% females and only 27% males.</p>
</section>
<section id="setup-3" class="level2">
<h2 class="anchored" data-anchor-id="setup-3">Setup</h2>
<p>Loading required packages.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb78"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb78-1"><a href="#cb78-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(<span class="st">"tidyverse"</span>)</span>
<span id="cb78-2"><a href="#cb78-2" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(<span class="st">"preprocessCore"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>Load the clean data.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb79"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb79-1"><a href="#cb79-1" aria-hidden="true" tabindex="-1"></a>dat_clean <span class="ot">&lt;-</span> <span class="fu">read_tsv</span>(<span class="at">file =</span> <span class="st">"../data/02_dat_clean.tsv"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>In the following the gene expression data will be normalized and log2 transformed. In order to do so the clean data is split into gene expression data and meta data.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb80"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb80-1"><a href="#cb80-1" aria-hidden="true" tabindex="-1"></a>dat_expr <span class="ot">&lt;-</span> dat_clean <span class="sc">|&gt;</span> </span>
<span id="cb80-2"><a href="#cb80-2" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">select</span>(<span class="fu">starts_with</span>(<span class="st">"ILMN"</span>)) </span>
<span id="cb80-3"><a href="#cb80-3" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb80-4"><a href="#cb80-4" aria-hidden="true" tabindex="-1"></a>dat_meta <span class="ot">&lt;-</span> dat_clean <span class="sc">|&gt;</span> </span>
<span id="cb80-5"><a href="#cb80-5" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">select</span>(<span class="sc">!</span><span class="fu">starts_with</span>(<span class="st">"ILMN"</span>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="normalization" class="level2">
<h2 class="anchored" data-anchor-id="normalization">Normalization</h2>
<p>The purpose of normalization is to minimize technical variability such that the remaining variation can be explained by biological differences. Technical variation can occur because of preparation differences or differences in the production of the microarrays and the processing of the arrays (for instance scanner differences). A general strategy is to re-distribute signal intensities across all samples such that they now all have the same distribution in order to compare the data from multiple samples.</p>
<p>The gene expression levels in the cleaned data are normalized by quantile normalization implemented in the function “r normalize.quantiles()”. This ensures that expression levels can be compared between patients in the data set. The algorithm is as follows:</p>
<ol type="1">
<li><p>Sort the the expression levels for each patient.</p></li>
<li><p>Calculate the mean expression level of the genes within the same rank.</p></li>
<li><p>Assign this mean to each gene within this rank.</p></li>
<li><p>Rearranging the genes for each patient to obtain the original order.</p></li>
</ol>
<div class="cell">
<div class="sourceCode cell-code" id="cb81"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb81-1"><a href="#cb81-1" aria-hidden="true" tabindex="-1"></a>dat_expr_matrix <span class="ot">&lt;-</span> <span class="fu">as.matrix</span>(<span class="fu">t</span>(dat_expr))</span>
<span id="cb81-2"><a href="#cb81-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb81-3"><a href="#cb81-3" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(dat_expr_matrix) <span class="ot">&lt;-</span> dat_meta <span class="sc">|&gt;</span> </span>
<span id="cb81-4"><a href="#cb81-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">pull</span>(identifier) </span>
<span id="cb81-5"><a href="#cb81-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb81-6"><a href="#cb81-6" aria-hidden="true" tabindex="-1"></a>dat_expr <span class="ot">&lt;-</span> <span class="fu">normalize.quantiles</span>(<span class="at">x =</span> dat_expr_matrix,</span>
<span id="cb81-7"><a href="#cb81-7" aria-hidden="true" tabindex="-1"></a>                                <span class="at">keep.names =</span> <span class="cn">TRUE</span>) <span class="sc">|&gt;</span></span>
<span id="cb81-8"><a href="#cb81-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">t</span>() <span class="sc">|&gt;</span> </span>
<span id="cb81-9"><a href="#cb81-9" aria-hidden="true" tabindex="-1"></a>  <span class="fu">as_tibble</span>(<span class="at">rownames =</span> <span class="cn">NA</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="log-transformation" class="level2">
<h2 class="anchored" data-anchor-id="log-transformation">Log Transformation</h2>
<p>In the following, the gene expression data is log2 transformed. This method is commonly used for microarray data to handle the issue that higher expression levels tend to have higher variance and vice versa.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb82"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb82-1"><a href="#cb82-1" aria-hidden="true" tabindex="-1"></a>dat_expr <span class="ot">&lt;-</span> dat_expr <span class="sc">|&gt;</span> </span>
<span id="cb82-2"><a href="#cb82-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rownames_to_column</span>(<span class="at">var =</span> <span class="st">"identifier"</span>) <span class="sc">|&gt;</span> </span>
<span id="cb82-3"><a href="#cb82-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="fu">across</span>(<span class="fu">starts_with</span>(<span class="st">"ILMN"</span>), log2))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>The normalized and log2 transformed gene expression data and the meta-data are combined.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb83"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb83-1"><a href="#cb83-1" aria-hidden="true" tabindex="-1"></a>dat_aug <span class="ot">&lt;-</span> <span class="fu">inner_join</span>(<span class="at">x =</span> dat_meta,</span>
<span id="cb83-2"><a href="#cb83-2" aria-hidden="true" tabindex="-1"></a>                      <span class="at">y =</span> dat_expr,</span>
<span id="cb83-3"><a href="#cb83-3" aria-hidden="true" tabindex="-1"></a>                      <span class="at">by =</span> <span class="st">"identifier"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="write-data-to-files-1" class="level2">
<h2 class="anchored" data-anchor-id="write-data-to-files-1">Write Data to Files</h2>
<p>The data is saved as .tsv-files.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb84"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb84-1"><a href="#cb84-1" aria-hidden="true" tabindex="-1"></a>dat_aug <span class="sc">|&gt;</span> </span>
<span id="cb84-2"><a href="#cb84-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">write_tsv</span>(<span class="at">file =</span> <span class="st">"../data/04_dat_aug.tsv"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>This was just a way to briefly see the effect and is not expected to be included in the final code:</p>
</section>
<section id="effect-of-normalization" class="level2">
<h2 class="anchored" data-anchor-id="effect-of-normalization">Effect of Normalization</h2>
<p>To better understand the normalization, the distribution of the expression level is examined before normalization.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb85"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb85-1"><a href="#cb85-1" aria-hidden="true" tabindex="-1"></a>non_normalized_expr <span class="ot">&lt;-</span> dat_clean <span class="sc">|&gt;</span> </span>
<span id="cb85-2"><a href="#cb85-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">pivot_longer</span>(<span class="at">cols =</span> <span class="fu">starts_with</span>(<span class="st">"ILMN"</span>),</span>
<span id="cb85-3"><a href="#cb85-3" aria-hidden="true" tabindex="-1"></a>               <span class="at">names_to =</span> <span class="st">"gene_id"</span>,</span>
<span id="cb85-4"><a href="#cb85-4" aria-hidden="true" tabindex="-1"></a>               <span class="at">values_to =</span> <span class="st">"expression"</span>) <span class="sc">|&gt;</span> </span>
<span id="cb85-5"><a href="#cb85-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">log2_expression =</span> <span class="fu">log2</span>(expression)) <span class="sc">|&gt;</span> </span>
<span id="cb85-6"><a href="#cb85-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggplot</span>(<span class="at">mapping =</span> <span class="fu">aes</span>(<span class="at">x =</span> geo_accession,</span>
<span id="cb85-7"><a href="#cb85-7" aria-hidden="true" tabindex="-1"></a>                       <span class="at">y =</span> log2_expression)) <span class="sc">+</span></span>
<span id="cb85-8"><a href="#cb85-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_boxplot</span>(<span class="at">outlier.shape =</span> <span class="cn">NA</span>,</span>
<span id="cb85-9"><a href="#cb85-9" aria-hidden="true" tabindex="-1"></a>               <span class="at">color =</span> <span class="st">"royalblue"</span>,</span>
<span id="cb85-10"><a href="#cb85-10" aria-hidden="true" tabindex="-1"></a>               <span class="at">fill =</span> <span class="st">"royalblue"</span>,</span>
<span id="cb85-11"><a href="#cb85-11" aria-hidden="true" tabindex="-1"></a>               <span class="at">alpha =</span> <span class="fl">0.2</span>) <span class="sc">+</span></span>
<span id="cb85-12"><a href="#cb85-12" aria-hidden="true" tabindex="-1"></a>  <span class="fu">coord_cartesian</span>(<span class="at">ylim =</span>  <span class="fu">c</span>(<span class="dv">5</span>, <span class="dv">11</span>)) <span class="sc">+</span></span>
<span id="cb85-13"><a href="#cb85-13" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_minimal</span>(<span class="at">base_size =</span> <span class="dv">16</span>) <span class="sc">+</span> </span>
<span id="cb85-14"><a href="#cb85-14" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">x =</span> <span class="st">"Patient"</span>,</span>
<span id="cb85-15"><a href="#cb85-15" aria-hidden="true" tabindex="-1"></a>       <span class="at">y =</span> <span class="st">"Expression level (log2)"</span>,</span>
<span id="cb85-16"><a href="#cb85-16" aria-hidden="true" tabindex="-1"></a>       <span class="at">title =</span> <span class="st">"Non-normalized expression level"</span>,</span>
<span id="cb85-17"><a href="#cb85-17" aria-hidden="true" tabindex="-1"></a>       <span class="at">subtitle =</span> <span class="fu">str_c</span>(<span class="st">"Distribution of gene expression level for each patient"</span>,</span>
<span id="cb85-18"><a href="#cb85-18" aria-hidden="true" tabindex="-1"></a>                        <span class="st">"</span><span class="sc">\n</span><span class="st">"</span>,</span>
<span id="cb85-19"><a href="#cb85-19" aria-hidden="true" tabindex="-1"></a>                        <span class="st">"without normalization (most extreme values are removed)"</span>),</span>
<span id="cb85-20"><a href="#cb85-20" aria-hidden="true" tabindex="-1"></a>       <span class="at">caption =</span> <span class="st">"Data from Dawany et al. (2014)"</span>) <span class="sc">+</span></span>
<span id="cb85-21"><a href="#cb85-21" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">axis.text.x =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb85-22"><a href="#cb85-22" aria-hidden="true" tabindex="-1"></a>        <span class="at">axis.text.y =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb85-23"><a href="#cb85-23" aria-hidden="true" tabindex="-1"></a>        <span class="at">axis.title =</span> <span class="fu">element_text</span>(<span class="at">face=</span><span class="st">"bold"</span>),</span>
<span id="cb85-24"><a href="#cb85-24" aria-hidden="true" tabindex="-1"></a>        <span class="at">plot.caption =</span> <span class="fu">element_text</span>(<span class="at">face =</span> <span class="st">"italic"</span>),</span>
<span id="cb85-25"><a href="#cb85-25" aria-hidden="true" tabindex="-1"></a>        <span class="at">legend.text =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="dv">14</span>))</span>
<span id="cb85-26"><a href="#cb85-26" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb85-27"><a href="#cb85-27" aria-hidden="true" tabindex="-1"></a><span class="fu">ggsave</span>(<span class="st">"../results/non_normalized_expr.png"</span>, <span class="at">plot =</span> non_normalized_expr)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Saving 7 x 5 in image</code></pre>
</div>
<div class="sourceCode cell-code" id="cb87"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb87-1"><a href="#cb87-1" aria-hidden="true" tabindex="-1"></a><span class="fu">show</span>(non_normalized_expr)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<p><img src="00_all_files/figure-html/unnamed-chunk-45-1.png" class="img-fluid" width="672"></p>
</div>
</div>
<p>After normalization, the quantiles for the expression level is equal for each patient.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb88"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb88-1"><a href="#cb88-1" aria-hidden="true" tabindex="-1"></a>normalized_expr <span class="ot">&lt;-</span> dat_aug <span class="sc">|&gt;</span> </span>
<span id="cb88-2"><a href="#cb88-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">pivot_longer</span>(<span class="at">cols =</span> <span class="fu">starts_with</span>(<span class="st">"ILMN"</span>),</span>
<span id="cb88-3"><a href="#cb88-3" aria-hidden="true" tabindex="-1"></a>               <span class="at">names_to =</span> <span class="st">"gene_id"</span>,</span>
<span id="cb88-4"><a href="#cb88-4" aria-hidden="true" tabindex="-1"></a>               <span class="at">values_to =</span> <span class="st">"log2_norm_expression"</span>) <span class="sc">|&gt;</span> </span>
<span id="cb88-5"><a href="#cb88-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggplot</span>(<span class="at">mapping =</span> <span class="fu">aes</span>(<span class="at">x =</span> geo_accession,</span>
<span id="cb88-6"><a href="#cb88-6" aria-hidden="true" tabindex="-1"></a>                       <span class="at">y =</span> log2_norm_expression)) <span class="sc">+</span></span>
<span id="cb88-7"><a href="#cb88-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_boxplot</span>(<span class="at">outlier.shape =</span> <span class="cn">NA</span>,</span>
<span id="cb88-8"><a href="#cb88-8" aria-hidden="true" tabindex="-1"></a>               <span class="at">color =</span> <span class="st">"royalblue"</span>,</span>
<span id="cb88-9"><a href="#cb88-9" aria-hidden="true" tabindex="-1"></a>               <span class="at">fill =</span> <span class="st">"royalblue"</span>,</span>
<span id="cb88-10"><a href="#cb88-10" aria-hidden="true" tabindex="-1"></a>               <span class="at">alpha =</span> <span class="fl">0.2</span>) <span class="sc">+</span></span>
<span id="cb88-11"><a href="#cb88-11" aria-hidden="true" tabindex="-1"></a>  <span class="fu">coord_cartesian</span>(<span class="at">ylim =</span>  <span class="fu">c</span>(<span class="dv">6</span>, <span class="dv">11</span>)) <span class="sc">+</span></span>
<span id="cb88-12"><a href="#cb88-12" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_minimal</span>(<span class="at">base_size =</span> <span class="dv">16</span>) <span class="sc">+</span> </span>
<span id="cb88-13"><a href="#cb88-13" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">x =</span> <span class="st">"Patient"</span>,</span>
<span id="cb88-14"><a href="#cb88-14" aria-hidden="true" tabindex="-1"></a>       <span class="at">y =</span> <span class="st">"Expression level (log2)"</span>,</span>
<span id="cb88-15"><a href="#cb88-15" aria-hidden="true" tabindex="-1"></a>       <span class="at">title =</span> <span class="st">"Normalized expression level"</span>,</span>
<span id="cb88-16"><a href="#cb88-16" aria-hidden="true" tabindex="-1"></a>       <span class="at">subtitle =</span> <span class="fu">str_c</span>(<span class="st">"Distribution of gene expression level for each patient"</span>,</span>
<span id="cb88-17"><a href="#cb88-17" aria-hidden="true" tabindex="-1"></a>                        <span class="st">"</span><span class="sc">\n</span><span class="st">"</span>,</span>
<span id="cb88-18"><a href="#cb88-18" aria-hidden="true" tabindex="-1"></a>                        <span class="st">"with normalization (most extreme values are removed)"</span>),</span>
<span id="cb88-19"><a href="#cb88-19" aria-hidden="true" tabindex="-1"></a>       <span class="at">caption =</span> <span class="st">"Data from Dawany et al. (2014)"</span>) <span class="sc">+</span></span>
<span id="cb88-20"><a href="#cb88-20" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">axis.text.x =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb88-21"><a href="#cb88-21" aria-hidden="true" tabindex="-1"></a>        <span class="at">axis.text.y =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb88-22"><a href="#cb88-22" aria-hidden="true" tabindex="-1"></a>        <span class="at">axis.title =</span> <span class="fu">element_text</span>(<span class="at">face=</span><span class="st">"bold"</span>),</span>
<span id="cb88-23"><a href="#cb88-23" aria-hidden="true" tabindex="-1"></a>        <span class="at">plot.caption =</span> <span class="fu">element_text</span>(<span class="at">face =</span> <span class="st">"italic"</span>),</span>
<span id="cb88-24"><a href="#cb88-24" aria-hidden="true" tabindex="-1"></a>        <span class="at">legend.text =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="dv">14</span>))</span>
<span id="cb88-25"><a href="#cb88-25" aria-hidden="true" tabindex="-1"></a> </span>
<span id="cb88-26"><a href="#cb88-26" aria-hidden="true" tabindex="-1"></a><span class="fu">ggsave</span>(<span class="st">"../results/normalized_expr.png"</span>, <span class="at">plot =</span> normalized_expr)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Saving 7 x 5 in image</code></pre>
</div>
<div class="sourceCode cell-code" id="cb90"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb90-1"><a href="#cb90-1" aria-hidden="true" tabindex="-1"></a><span class="fu">show</span>(normalized_expr)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<p><img src="00_all_files/figure-html/unnamed-chunk-46-1.png" class="img-fluid" width="672"></p>
</div>
</div>
</section>
<section id="setup-4" class="level2">
<h2 class="anchored" data-anchor-id="setup-4">Setup</h2>
<p>Loading required packages.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb91"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb91-1"><a href="#cb91-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(<span class="st">'tidyverse'</span>)</span>
<span id="cb91-2"><a href="#cb91-2" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(<span class="st">"stringr"</span>)</span>
<span id="cb91-3"><a href="#cb91-3" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(<span class="st">'broom'</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>Load the augment data.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb92"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb92-1"><a href="#cb92-1" aria-hidden="true" tabindex="-1"></a>dat_aug <span class="ot">&lt;-</span> <span class="fu">read_tsv</span>(<span class="at">file =</span> <span class="st">"../data/04_dat_aug.tsv"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="principal-component-analysis" class="level2">
<h2 class="anchored" data-anchor-id="principal-component-analysis">Principal Component Analysis</h2>
<p>In the following a Principal Component Analysis (PCA) is carried out on the augment data.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb93"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb93-1"><a href="#cb93-1" aria-hidden="true" tabindex="-1"></a>pca_fit <span class="ot">&lt;-</span> dat_aug <span class="sc">|&gt;</span> </span>
<span id="cb93-2"><a href="#cb93-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(<span class="fu">starts_with</span>(<span class="st">"ILMN"</span>)) <span class="sc">|&gt;</span></span>
<span id="cb93-3"><a href="#cb93-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">prcomp</span>()</span>
<span id="cb93-4"><a href="#cb93-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb93-5"><a href="#cb93-5" aria-hidden="true" tabindex="-1"></a>pca_fit_aug <span class="ot">&lt;-</span> pca_fit <span class="sc">|&gt;</span> </span>
<span id="cb93-6"><a href="#cb93-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">augment</span>(dat_aug)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<section id="plot-principal-components" class="level3">
<h3 class="anchored" data-anchor-id="plot-principal-components">Plot Principal Components</h3>
<p>The Principal Components (PC) of the PCA is plotted against each other.</p>
<p>First, the first PC is plotted against the second PC.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb94"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb94-1"><a href="#cb94-1" aria-hidden="true" tabindex="-1"></a>scatter_pc1_pc2 <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(<span class="at">data =</span> pca_fit_aug,</span>
<span id="cb94-2"><a href="#cb94-2" aria-hidden="true" tabindex="-1"></a>                          <span class="at">mapping =</span> <span class="fu">aes</span>(<span class="at">x =</span> .fittedPC1, </span>
<span id="cb94-3"><a href="#cb94-3" aria-hidden="true" tabindex="-1"></a>                                        <span class="at">y =</span> .fittedPC2, </span>
<span id="cb94-4"><a href="#cb94-4" aria-hidden="true" tabindex="-1"></a>                                        <span class="at">color =</span> disease_state, </span>
<span id="cb94-5"><a href="#cb94-5" aria-hidden="true" tabindex="-1"></a>                                        <span class="at">shape =</span> gender)) <span class="sc">+</span> </span>
<span id="cb94-6"><a href="#cb94-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_point</span>(<span class="at">size =</span> <span class="dv">3</span>,</span>
<span id="cb94-7"><a href="#cb94-7" aria-hidden="true" tabindex="-1"></a>             <span class="at">alpha =</span> <span class="fl">0.8</span>) <span class="sc">+</span></span>
<span id="cb94-8"><a href="#cb94-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_color_manual</span>(</span>
<span id="cb94-9"><a href="#cb94-9" aria-hidden="true" tabindex="-1"></a>    <span class="at">values =</span> <span class="fu">c</span>(<span class="st">"HIV"</span> <span class="ot">=</span> <span class="st">"royalblue"</span>, </span>
<span id="cb94-10"><a href="#cb94-10" aria-hidden="true" tabindex="-1"></a>               <span class="st">"HIV/TB"</span> <span class="ot">=</span> <span class="st">"tomato"</span>)) <span class="sc">+</span></span>
<span id="cb94-11"><a href="#cb94-11" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_minimal</span>(<span class="at">base_size =</span> <span class="dv">16</span>) <span class="sc">+</span> </span>
<span id="cb94-12"><a href="#cb94-12" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">x =</span> <span class="st">"PC1"</span>,</span>
<span id="cb94-13"><a href="#cb94-13" aria-hidden="true" tabindex="-1"></a>       <span class="at">y =</span> <span class="st">"PC2"</span>,</span>
<span id="cb94-14"><a href="#cb94-14" aria-hidden="true" tabindex="-1"></a>       <span class="at">color =</span> <span class="st">"Disease state"</span>,</span>
<span id="cb94-15"><a href="#cb94-15" aria-hidden="true" tabindex="-1"></a>       <span class="at">shape =</span> <span class="st">"Gender"</span>,</span>
<span id="cb94-16"><a href="#cb94-16" aria-hidden="true" tabindex="-1"></a>       <span class="at">title =</span> <span class="st">"Principal Component Analysis"</span>,</span>
<span id="cb94-17"><a href="#cb94-17" aria-hidden="true" tabindex="-1"></a>       <span class="at">subtitle =</span> <span class="fu">str_c</span>(<span class="st">"Each patient represented by PC coordinates"</span>,</span>
<span id="cb94-18"><a href="#cb94-18" aria-hidden="true" tabindex="-1"></a>                        <span class="st">"</span><span class="sc">\n</span><span class="st">"</span>,</span>
<span id="cb94-19"><a href="#cb94-19" aria-hidden="true" tabindex="-1"></a>                        <span class="st">"colored by disease state"</span>),</span>
<span id="cb94-20"><a href="#cb94-20" aria-hidden="true" tabindex="-1"></a>       <span class="at">caption =</span> <span class="st">"Data from Dawany et al. (2014)"</span>) <span class="sc">+</span></span>
<span id="cb94-21"><a href="#cb94-21" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">axis.text.x =</span> <span class="fu">element_text</span>(<span class="at">color =</span> <span class="st">"black"</span>, </span>
<span id="cb94-22"><a href="#cb94-22" aria-hidden="true" tabindex="-1"></a>                                   <span class="at">size =</span> <span class="dv">14</span>),</span>
<span id="cb94-23"><a href="#cb94-23" aria-hidden="true" tabindex="-1"></a>        <span class="at">axis.text.y =</span> <span class="fu">element_text</span>(<span class="at">color =</span> <span class="st">"black"</span>, </span>
<span id="cb94-24"><a href="#cb94-24" aria-hidden="true" tabindex="-1"></a>                                   <span class="at">size =</span> <span class="dv">14</span>),</span>
<span id="cb94-25"><a href="#cb94-25" aria-hidden="true" tabindex="-1"></a>        <span class="at">axis.title =</span> <span class="fu">element_text</span>(<span class="at">face=</span><span class="st">"bold"</span>),</span>
<span id="cb94-26"><a href="#cb94-26" aria-hidden="true" tabindex="-1"></a>        <span class="at">plot.caption =</span> <span class="fu">element_text</span>(<span class="at">face =</span> <span class="st">"italic"</span>),</span>
<span id="cb94-27"><a href="#cb94-27" aria-hidden="true" tabindex="-1"></a>        <span class="at">legend.text =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="dv">14</span>))</span>
<span id="cb94-28"><a href="#cb94-28" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb94-29"><a href="#cb94-29" aria-hidden="true" tabindex="-1"></a><span class="fu">ggsave</span>(<span class="st">"../results/scatter_pc1_pc2.png"</span>, <span class="at">plot =</span> scatter_pc1_pc2)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Saving 7 x 5 in image</code></pre>
</div>
<div class="sourceCode cell-code" id="cb96"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb96-1"><a href="#cb96-1" aria-hidden="true" tabindex="-1"></a><span class="fu">show</span>(scatter_pc1_pc2)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<p><img src="00_all_files/figure-html/05_Plot%20PC1%20vs%20PC2-1.png" class="img-fluid" width="672"></p>
</div>
</div>
<p>From above it is seen, that PC1 to some extend is able to distinguis between HIV/TB co-infected and HIV mono-infected patients, while PC2 show no prominent pattern related to infection or gender.</p>
</section>
<section id="plot-the-variance-explained" class="level3">
<h3 class="anchored" data-anchor-id="plot-the-variance-explained">Plot the Variance Explained</h3>
<p>The variance explained by each PC is plotted.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb97"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb97-1"><a href="#cb97-1" aria-hidden="true" tabindex="-1"></a>pca_fit_eigen <span class="ot">&lt;-</span> pca_fit <span class="sc">|&gt;</span> </span>
<span id="cb97-2"><a href="#cb97-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">tidy</span>(<span class="at">matrix =</span> <span class="st">"eigenvalues"</span>)</span>
<span id="cb97-3"><a href="#cb97-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb97-4"><a href="#cb97-4" aria-hidden="true" tabindex="-1"></a>var_explained <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(<span class="at">data =</span> pca_fit_eigen,</span>
<span id="cb97-5"><a href="#cb97-5" aria-hidden="true" tabindex="-1"></a>                        <span class="at">mapping =</span> <span class="fu">aes</span>(<span class="at">x =</span> PC, </span>
<span id="cb97-6"><a href="#cb97-6" aria-hidden="true" tabindex="-1"></a>                                      <span class="at">y =</span> percent)) <span class="sc">+</span></span>
<span id="cb97-7"><a href="#cb97-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_col</span>(<span class="at">fill =</span> <span class="st">"royalblue"</span>) <span class="sc">+</span></span>
<span id="cb97-8"><a href="#cb97-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_smooth</span>(<span class="at">mapping =</span> <span class="fu">aes</span>(<span class="at">x =</span> PC, </span>
<span id="cb97-9"><a href="#cb97-9" aria-hidden="true" tabindex="-1"></a>                            <span class="at">y =</span> cumulative), </span>
<span id="cb97-10"><a href="#cb97-10" aria-hidden="true" tabindex="-1"></a>              <span class="at">method =</span> <span class="st">"loess"</span>, </span>
<span id="cb97-11"><a href="#cb97-11" aria-hidden="true" tabindex="-1"></a>              <span class="at">formula =</span> y <span class="sc">~</span> x, </span>
<span id="cb97-12"><a href="#cb97-12" aria-hidden="true" tabindex="-1"></a>              <span class="at">color =</span> <span class="st">"royalblue"</span>,</span>
<span id="cb97-13"><a href="#cb97-13" aria-hidden="true" tabindex="-1"></a>              <span class="at">se =</span> <span class="cn">FALSE</span>) <span class="sc">+</span></span>
<span id="cb97-14"><a href="#cb97-14" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_hline</span>(<span class="at">yintercept =</span> <span class="fl">0.90</span>) <span class="sc">+</span></span>
<span id="cb97-15"><a href="#cb97-15" aria-hidden="true" tabindex="-1"></a>  <span class="fu">annotate</span>(<span class="st">"text"</span>, </span>
<span id="cb97-16"><a href="#cb97-16" aria-hidden="true" tabindex="-1"></a>           <span class="at">x=</span><span class="dv">10</span>, </span>
<span id="cb97-17"><a href="#cb97-17" aria-hidden="true" tabindex="-1"></a>           <span class="at">y=</span><span class="fl">0.85</span>, </span>
<span id="cb97-18"><a href="#cb97-18" aria-hidden="true" tabindex="-1"></a>           <span class="at">label=</span><span class="st">"90% variance explained"</span>,</span>
<span id="cb97-19"><a href="#cb97-19" aria-hidden="true" tabindex="-1"></a>           <span class="at">size =</span> <span class="dv">4</span>) <span class="sc">+</span></span>
<span id="cb97-20"><a href="#cb97-20" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_y_continuous</span>(</span>
<span id="cb97-21"><a href="#cb97-21" aria-hidden="true" tabindex="-1"></a>    <span class="at">labels =</span> scales<span class="sc">::</span><span class="fu">percent_format</span>(),</span>
<span id="cb97-22"><a href="#cb97-22" aria-hidden="true" tabindex="-1"></a>    <span class="at">expand =</span> <span class="fu">expansion</span>(<span class="at">mult =</span> <span class="fu">c</span>(<span class="dv">0</span>, <span class="fl">0.01</span>))) <span class="sc">+</span></span>
<span id="cb97-23"><a href="#cb97-23" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_minimal</span>(<span class="at">base_size =</span> <span class="dv">16</span>) <span class="sc">+</span> </span>
<span id="cb97-24"><a href="#cb97-24" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">x =</span> <span class="st">"PCs"</span>,</span>
<span id="cb97-25"><a href="#cb97-25" aria-hidden="true" tabindex="-1"></a>       <span class="at">y =</span> <span class="st">"Variance explained"</span>,</span>
<span id="cb97-26"><a href="#cb97-26" aria-hidden="true" tabindex="-1"></a>       <span class="at">title =</span> <span class="st">"Explained Variance"</span>,</span>
<span id="cb97-27"><a href="#cb97-27" aria-hidden="true" tabindex="-1"></a>       <span class="at">subtitle =</span> <span class="st">"Variance explained by the first PCs"</span>,</span>
<span id="cb97-28"><a href="#cb97-28" aria-hidden="true" tabindex="-1"></a>       <span class="at">caption =</span> <span class="st">"Data from Dawany et al. (2014)"</span>) <span class="sc">+</span></span>
<span id="cb97-29"><a href="#cb97-29" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">axis.text.x =</span> <span class="fu">element_text</span>(<span class="at">color =</span> <span class="st">"black"</span>, </span>
<span id="cb97-30"><a href="#cb97-30" aria-hidden="true" tabindex="-1"></a>                                   <span class="at">size =</span> <span class="dv">14</span>),</span>
<span id="cb97-31"><a href="#cb97-31" aria-hidden="true" tabindex="-1"></a>        <span class="at">axis.text.y =</span> <span class="fu">element_text</span>(<span class="at">color =</span> <span class="st">"black"</span>, </span>
<span id="cb97-32"><a href="#cb97-32" aria-hidden="true" tabindex="-1"></a>                                   <span class="at">size =</span> <span class="dv">14</span>),</span>
<span id="cb97-33"><a href="#cb97-33" aria-hidden="true" tabindex="-1"></a>        <span class="at">axis.title =</span> <span class="fu">element_text</span>(<span class="at">face=</span><span class="st">"bold"</span>),</span>
<span id="cb97-34"><a href="#cb97-34" aria-hidden="true" tabindex="-1"></a>        <span class="at">plot.caption =</span> <span class="fu">element_text</span>(<span class="at">face =</span> <span class="st">"italic"</span>),</span>
<span id="cb97-35"><a href="#cb97-35" aria-hidden="true" tabindex="-1"></a>        <span class="at">legend.text =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="dv">14</span>))</span>
<span id="cb97-36"><a href="#cb97-36" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb97-37"><a href="#cb97-37" aria-hidden="true" tabindex="-1"></a><span class="fu">ggsave</span>(<span class="st">"../results/var_explained.png"</span>, <span class="at">plot =</span> var_explained)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Saving 7 x 5 in image</code></pre>
</div>
<div class="sourceCode cell-code" id="cb99"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb99-1"><a href="#cb99-1" aria-hidden="true" tabindex="-1"></a><span class="fu">show</span>(var_explained)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<p><img src="00_all_files/figure-html/05_Plot%20variance%20explained-1.png" class="img-fluid" width="672"></p>
</div>
</div>
<p>From the variance explained plot it is seen, that 35 PCs are required to explain 95% of the variance in the data.</p>
</section>
</section>
<section id="setup-5" class="level2">
<h2 class="anchored" data-anchor-id="setup-5">Setup</h2>
<p>Loading required packages.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb100"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb100-1"><a href="#cb100-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(<span class="st">'tidyverse'</span>)</span>
<span id="cb100-2"><a href="#cb100-2" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(<span class="st">'broom'</span>)</span>
<span id="cb100-3"><a href="#cb100-3" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(<span class="st">"ggrepel"</span>)</span>
<span id="cb100-4"><a href="#cb100-4" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(<span class="st">"illuminaHumanv4.db"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>Load the augment data.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb101"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb101-1"><a href="#cb101-1" aria-hidden="true" tabindex="-1"></a>dat_aug <span class="ot">&lt;-</span> <span class="fu">read_tsv</span>(<span class="at">file =</span> <span class="st">"../data/04_dat_aug.tsv"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="linear-regression" class="level2">
<h2 class="anchored" data-anchor-id="linear-regression">Linear Regression</h2>
<p>In the following a linear regression model is mapped to the data. In order to do so, the data need to be pivoted to long format and grouped by genes. Gene expression levels and disease label are nested pr. gene.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb102"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb102-1"><a href="#cb102-1" aria-hidden="true" tabindex="-1"></a>dat_nest <span class="ot">&lt;-</span> dat_aug <span class="sc">|&gt;</span></span>
<span id="cb102-2"><a href="#cb102-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">disease_state =</span> <span class="fu">case_when</span>(disease_state <span class="sc">==</span> <span class="st">"HIV"</span> <span class="sc">~</span> <span class="dv">0</span>,</span>
<span id="cb102-3"><a href="#cb102-3" aria-hidden="true" tabindex="-1"></a>                                   disease_state <span class="sc">==</span> <span class="st">"HIV/TB"</span> <span class="sc">~</span> <span class="dv">1</span>)) <span class="sc">|&gt;</span> </span>
<span id="cb102-4"><a href="#cb102-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">pivot_longer</span>(<span class="at">cols =</span> <span class="fu">matches</span>(<span class="st">"^ILMN_[0-9]+"</span>),</span>
<span id="cb102-5"><a href="#cb102-5" aria-hidden="true" tabindex="-1"></a>               <span class="at">names_to =</span> <span class="st">"gene"</span>, </span>
<span id="cb102-6"><a href="#cb102-6" aria-hidden="true" tabindex="-1"></a>               <span class="at">values_to =</span> <span class="st">"log2_expr_level"</span>) <span class="sc">|&gt;</span> </span>
<span id="cb102-7"><a href="#cb102-7" aria-hidden="true" tabindex="-1"></a>               <span class="fu">group_by</span>(gene) <span class="sc">|&gt;</span> </span>
<span id="cb102-8"><a href="#cb102-8" aria-hidden="true" tabindex="-1"></a>               <span class="fu">nest</span>()</span>
<span id="cb102-9"><a href="#cb102-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb102-10"><a href="#cb102-10" aria-hidden="true" tabindex="-1"></a>dat_nest</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code># A tibble: 47,231 × 2
# Groups:   gene [47,231]
   gene         data             
   &lt;chr&gt;        &lt;list&gt;           
 1 ILMN_1762337 &lt;tibble [43 × 7]&gt;
 2 ILMN_2055271 &lt;tibble [43 × 7]&gt;
 3 ILMN_1736007 &lt;tibble [43 × 7]&gt;
 4 ILMN_2383229 &lt;tibble [43 × 7]&gt;
 5 ILMN_1806310 &lt;tibble [43 × 7]&gt;
 6 ILMN_1779670 &lt;tibble [43 × 7]&gt;
 7 ILMN_1653355 &lt;tibble [43 × 7]&gt;
 8 ILMN_1717783 &lt;tibble [43 × 7]&gt;
 9 ILMN_1705025 &lt;tibble [43 × 7]&gt;
10 ILMN_1814316 &lt;tibble [43 × 7]&gt;
# ℹ 47,221 more rows</code></pre>
</div>
</div>
<p>A linear regression model is mapped to the nested data of each gene.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb104"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb104-1"><a href="#cb104-1" aria-hidden="true" tabindex="-1"></a>dat_nest <span class="ot">&lt;-</span> dat_nest <span class="sc">|&gt;</span></span>
<span id="cb104-2"><a href="#cb104-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">model_object =</span> <span class="fu">map</span>(<span class="at">.x =</span> data,</span>
<span id="cb104-3"><a href="#cb104-3" aria-hidden="true" tabindex="-1"></a>                            <span class="at">.f =</span> <span class="sc">~</span><span class="fu">lm</span>(<span class="at">formula =</span> log2_expr_level <span class="sc">~</span> disease_state,</span>
<span id="cb104-4"><a href="#cb104-4" aria-hidden="true" tabindex="-1"></a>                                     <span class="at">data =</span> .x)))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>Tidy statistics are mapped to the model of each gene and saved in ‘tidy_model_object’.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb105"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb105-1"><a href="#cb105-1" aria-hidden="true" tabindex="-1"></a>dat_nest <span class="ot">&lt;-</span> dat_nest <span class="sc">|&gt;</span></span>
<span id="cb105-2"><a href="#cb105-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">tidy_model_object =</span> <span class="fu">map</span>(<span class="at">.x =</span> model_object,</span>
<span id="cb105-3"><a href="#cb105-3" aria-hidden="true" tabindex="-1"></a>                                 <span class="at">.f =</span> <span class="sc">~</span><span class="fu">tidy</span>(<span class="at">conf.int =</span> <span class="cn">TRUE</span>,</span>
<span id="cb105-4"><a href="#cb105-4" aria-hidden="true" tabindex="-1"></a>                                            <span class="at">conf.level =</span> <span class="fl">0.95</span>,</span>
<span id="cb105-5"><a href="#cb105-5" aria-hidden="true" tabindex="-1"></a>                                            <span class="at">x =</span> .x)))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>The tidy model objects are unnested, only slope estimates are kept for each gene and selected together with p-values and CI values. Lastly, the data is being ungrouped.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb106"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb106-1"><a href="#cb106-1" aria-hidden="true" tabindex="-1"></a>dat_estimates <span class="ot">&lt;-</span> dat_nest <span class="sc">|&gt;</span> </span>
<span id="cb106-2"><a href="#cb106-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">unnest</span>(tidy_model_object) <span class="sc">|&gt;</span> </span>
<span id="cb106-3"><a href="#cb106-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(term <span class="sc">!=</span> <span class="st">"(Intercept)"</span>) <span class="sc">|&gt;</span> </span>
<span id="cb106-4"><a href="#cb106-4" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">select</span>(gene, </span>
<span id="cb106-5"><a href="#cb106-5" aria-hidden="true" tabindex="-1"></a>         p.value, </span>
<span id="cb106-6"><a href="#cb106-6" aria-hidden="true" tabindex="-1"></a>         estimate, </span>
<span id="cb106-7"><a href="#cb106-7" aria-hidden="true" tabindex="-1"></a>         conf.low, </span>
<span id="cb106-8"><a href="#cb106-8" aria-hidden="true" tabindex="-1"></a>         conf.high) <span class="sc">|&gt;</span> </span>
<span id="cb106-9"><a href="#cb106-9" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ungroup</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>An adjusted p-value is calculated and added to the data estimate table together with a “significant label” based on a 0.05%.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb107"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb107-1"><a href="#cb107-1" aria-hidden="true" tabindex="-1"></a>dat_estimates <span class="ot">&lt;-</span> dat_estimates <span class="sc">|&gt;</span></span>
<span id="cb107-2"><a href="#cb107-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">p_val_adjust =</span> <span class="fu">p.adjust</span>(p.value),</span>
<span id="cb107-3"><a href="#cb107-3" aria-hidden="true" tabindex="-1"></a>         <span class="at">significant =</span> <span class="fu">case_when</span>(p_val_adjust <span class="sc">&lt;=</span> <span class="fl">0.05</span> <span class="sc">~</span> <span class="st">"Yes"</span>, <span class="cn">TRUE</span> <span class="sc">~</span><span class="st">"No"</span>))</span>
<span id="cb107-4"><a href="#cb107-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb107-5"><a href="#cb107-5" aria-hidden="true" tabindex="-1"></a>dat_estimates <span class="sc">|&gt;</span> </span>
<span id="cb107-6"><a href="#cb107-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(significant <span class="sc">==</span> <span class="st">"Yes"</span>) <span class="sc">|&gt;</span> </span>
<span id="cb107-7"><a href="#cb107-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarise</span>(<span class="at">n =</span> <span class="fu">n</span>())</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code># A tibble: 1 × 1
      n
  &lt;int&gt;
1    40</code></pre>
</div>
</div>
<p>We convert the illumina probe names to the gene symbol.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb109"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb109-1"><a href="#cb109-1" aria-hidden="true" tabindex="-1"></a>dat_estimates <span class="ot">&lt;-</span> dat_estimates <span class="sc">|&gt;</span> </span>
<span id="cb109-2"><a href="#cb109-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">gene_symbol =</span> <span class="fu">map</span>(<span class="at">.x =</span> gene,</span>
<span id="cb109-3"><a href="#cb109-3" aria-hidden="true" tabindex="-1"></a>                           <span class="at">.f =</span> <span class="sc">~</span> <span class="fu">get</span>(<span class="at">x =</span> .x, </span>
<span id="cb109-4"><a href="#cb109-4" aria-hidden="true" tabindex="-1"></a>                                      <span class="at">envir =</span> illuminaHumanv4SYMBOL)),</span>
<span id="cb109-5"><a href="#cb109-5" aria-hidden="true" tabindex="-1"></a>         <span class="at">gene_symbol =</span> <span class="fu">unlist</span>(gene_symbol))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<section id="compare-most-significant-genes-with-list-of-diagnostic-genes-from-the-251-gene-signature-from-the-paper." class="level3">
<h3 class="anchored" data-anchor-id="compare-most-significant-genes-with-list-of-diagnostic-genes-from-the-251-gene-signature-from-the-paper.">Compare most significant genes with list of diagnostic genes from the 251 gene signature from the paper.</h3>
<p>Extract the significant genes with original gene ID</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb110"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb110-1"><a href="#cb110-1" aria-hidden="true" tabindex="-1"></a>significant_genes <span class="ot">&lt;-</span> dat_estimates <span class="sc">|&gt;</span> </span>
<span id="cb110-2"><a href="#cb110-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(significant <span class="sc">==</span> <span class="st">"Yes"</span>) <span class="sc">|&gt;</span></span>
<span id="cb110-3"><a href="#cb110-3" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">select</span>(gene_symbol) <span class="sc">|&gt;</span> </span>
<span id="cb110-4"><a href="#cb110-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">pull</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb111"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb111-1"><a href="#cb111-1" aria-hidden="true" tabindex="-1"></a>signature_251_str <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">"DHRS4</span></span>
<span id="cb111-2"><a href="#cb111-2" aria-hidden="true" tabindex="-1"></a><span class="st">EYA3</span></span>
<span id="cb111-3"><a href="#cb111-3" aria-hidden="true" tabindex="-1"></a><span class="st">CMIP</span></span>
<span id="cb111-4"><a href="#cb111-4" aria-hidden="true" tabindex="-1"></a><span class="st">UBTF</span></span>
<span id="cb111-5"><a href="#cb111-5" aria-hidden="true" tabindex="-1"></a><span class="st">TMEM164</span></span>
<span id="cb111-6"><a href="#cb111-6" aria-hidden="true" tabindex="-1"></a><span class="st">CDKN2B</span></span>
<span id="cb111-7"><a href="#cb111-7" aria-hidden="true" tabindex="-1"></a><span class="st">PTK2</span></span>
<span id="cb111-8"><a href="#cb111-8" aria-hidden="true" tabindex="-1"></a><span class="st">LOC648605</span></span>
<span id="cb111-9"><a href="#cb111-9" aria-hidden="true" tabindex="-1"></a><span class="st">NaN</span></span>
<span id="cb111-10"><a href="#cb111-10" aria-hidden="true" tabindex="-1"></a><span class="st">ACOX3</span></span>
<span id="cb111-11"><a href="#cb111-11" aria-hidden="true" tabindex="-1"></a><span class="st">MLKL</span></span>
<span id="cb111-12"><a href="#cb111-12" aria-hidden="true" tabindex="-1"></a><span class="st">ATP5I</span></span>
<span id="cb111-13"><a href="#cb111-13" aria-hidden="true" tabindex="-1"></a><span class="st">STX6</span></span>
<span id="cb111-14"><a href="#cb111-14" aria-hidden="true" tabindex="-1"></a><span class="st">WTAP</span></span>
<span id="cb111-15"><a href="#cb111-15" aria-hidden="true" tabindex="-1"></a><span class="st">MR1</span></span>
<span id="cb111-16"><a href="#cb111-16" aria-hidden="true" tabindex="-1"></a><span class="st">LOC151579</span></span>
<span id="cb111-17"><a href="#cb111-17" aria-hidden="true" tabindex="-1"></a><span class="st">SIK3</span></span>
<span id="cb111-18"><a href="#cb111-18" aria-hidden="true" tabindex="-1"></a><span class="st">USP41</span></span>
<span id="cb111-19"><a href="#cb111-19" aria-hidden="true" tabindex="-1"></a><span class="st">WDR63</span></span>
<span id="cb111-20"><a href="#cb111-20" aria-hidden="true" tabindex="-1"></a><span class="st">INTS3</span></span>
<span id="cb111-21"><a href="#cb111-21" aria-hidden="true" tabindex="-1"></a><span class="st">LOC653717</span></span>
<span id="cb111-22"><a href="#cb111-22" aria-hidden="true" tabindex="-1"></a><span class="st">TLN1</span></span>
<span id="cb111-23"><a href="#cb111-23" aria-hidden="true" tabindex="-1"></a><span class="st">NINJ2</span></span>
<span id="cb111-24"><a href="#cb111-24" aria-hidden="true" tabindex="-1"></a><span class="st">GRAMD3</span></span>
<span id="cb111-25"><a href="#cb111-25" aria-hidden="true" tabindex="-1"></a><span class="st">SCAND2</span></span>
<span id="cb111-26"><a href="#cb111-26" aria-hidden="true" tabindex="-1"></a><span class="st">RSPH3</span></span>
<span id="cb111-27"><a href="#cb111-27" aria-hidden="true" tabindex="-1"></a><span class="st">CREB3L2</span></span>
<span id="cb111-28"><a href="#cb111-28" aria-hidden="true" tabindex="-1"></a><span class="st">PARP2</span></span>
<span id="cb111-29"><a href="#cb111-29" aria-hidden="true" tabindex="-1"></a><span class="st">NENF</span></span>
<span id="cb111-30"><a href="#cb111-30" aria-hidden="true" tabindex="-1"></a><span class="st">TNFRSF14</span></span>
<span id="cb111-31"><a href="#cb111-31" aria-hidden="true" tabindex="-1"></a><span class="st">CSNK1G3</span></span>
<span id="cb111-32"><a href="#cb111-32" aria-hidden="true" tabindex="-1"></a><span class="st">AMY1B</span></span>
<span id="cb111-33"><a href="#cb111-33" aria-hidden="true" tabindex="-1"></a><span class="st">TIAL1</span></span>
<span id="cb111-34"><a href="#cb111-34" aria-hidden="true" tabindex="-1"></a><span class="st">MOBKL2C</span></span>
<span id="cb111-35"><a href="#cb111-35" aria-hidden="true" tabindex="-1"></a><span class="st">LOC728635</span></span>
<span id="cb111-36"><a href="#cb111-36" aria-hidden="true" tabindex="-1"></a><span class="st">LOC285053</span></span>
<span id="cb111-37"><a href="#cb111-37" aria-hidden="true" tabindex="-1"></a><span class="st">C3orf64</span></span>
<span id="cb111-38"><a href="#cb111-38" aria-hidden="true" tabindex="-1"></a><span class="st">USP34</span></span>
<span id="cb111-39"><a href="#cb111-39" aria-hidden="true" tabindex="-1"></a><span class="st">NaN</span></span>
<span id="cb111-40"><a href="#cb111-40" aria-hidden="true" tabindex="-1"></a><span class="st">RTN1</span></span>
<span id="cb111-41"><a href="#cb111-41" aria-hidden="true" tabindex="-1"></a><span class="st">EIF3J</span></span>
<span id="cb111-42"><a href="#cb111-42" aria-hidden="true" tabindex="-1"></a><span class="st">PPM1F</span></span>
<span id="cb111-43"><a href="#cb111-43" aria-hidden="true" tabindex="-1"></a><span class="st">REL</span></span>
<span id="cb111-44"><a href="#cb111-44" aria-hidden="true" tabindex="-1"></a><span class="st">C16orf48</span></span>
<span id="cb111-45"><a href="#cb111-45" aria-hidden="true" tabindex="-1"></a><span class="st">ASCL2</span></span>
<span id="cb111-46"><a href="#cb111-46" aria-hidden="true" tabindex="-1"></a><span class="st">STX6</span></span>
<span id="cb111-47"><a href="#cb111-47" aria-hidden="true" tabindex="-1"></a><span class="st">ACVR1</span></span>
<span id="cb111-48"><a href="#cb111-48" aria-hidden="true" tabindex="-1"></a><span class="st">ZNF91</span></span>
<span id="cb111-49"><a href="#cb111-49" aria-hidden="true" tabindex="-1"></a><span class="st">MICB</span></span>
<span id="cb111-50"><a href="#cb111-50" aria-hidden="true" tabindex="-1"></a><span class="st">MSRB2</span></span>
<span id="cb111-51"><a href="#cb111-51" aria-hidden="true" tabindex="-1"></a><span class="st">TCEA1</span></span>
<span id="cb111-52"><a href="#cb111-52" aria-hidden="true" tabindex="-1"></a><span class="st">BTBD1</span></span>
<span id="cb111-53"><a href="#cb111-53" aria-hidden="true" tabindex="-1"></a><span class="st">DHRS4L2</span></span>
<span id="cb111-54"><a href="#cb111-54" aria-hidden="true" tabindex="-1"></a><span class="st">LOC402057</span></span>
<span id="cb111-55"><a href="#cb111-55" aria-hidden="true" tabindex="-1"></a><span class="st">HPCAL1</span></span>
<span id="cb111-56"><a href="#cb111-56" aria-hidden="true" tabindex="-1"></a><span class="st">P2RY13</span></span>
<span id="cb111-57"><a href="#cb111-57" aria-hidden="true" tabindex="-1"></a><span class="st">LOC388122</span></span>
<span id="cb111-58"><a href="#cb111-58" aria-hidden="true" tabindex="-1"></a><span class="st">MATR3</span></span>
<span id="cb111-59"><a href="#cb111-59" aria-hidden="true" tabindex="-1"></a><span class="st">KIAA0947</span></span>
<span id="cb111-60"><a href="#cb111-60" aria-hidden="true" tabindex="-1"></a><span class="st">APOL6</span></span>
<span id="cb111-61"><a href="#cb111-61" aria-hidden="true" tabindex="-1"></a><span class="st">GK</span></span>
<span id="cb111-62"><a href="#cb111-62" aria-hidden="true" tabindex="-1"></a><span class="st">HLA-DMA</span></span>
<span id="cb111-63"><a href="#cb111-63" aria-hidden="true" tabindex="-1"></a><span class="st">FDFT1</span></span>
<span id="cb111-64"><a href="#cb111-64" aria-hidden="true" tabindex="-1"></a><span class="st">GBP2</span></span>
<span id="cb111-65"><a href="#cb111-65" aria-hidden="true" tabindex="-1"></a><span class="st">ST3GAL2</span></span>
<span id="cb111-66"><a href="#cb111-66" aria-hidden="true" tabindex="-1"></a><span class="st">GK</span></span>
<span id="cb111-67"><a href="#cb111-67" aria-hidden="true" tabindex="-1"></a><span class="st">BCCIP</span></span>
<span id="cb111-68"><a href="#cb111-68" aria-hidden="true" tabindex="-1"></a><span class="st">BRSK1</span></span>
<span id="cb111-69"><a href="#cb111-69" aria-hidden="true" tabindex="-1"></a><span class="st">MLKL</span></span>
<span id="cb111-70"><a href="#cb111-70" aria-hidden="true" tabindex="-1"></a><span class="st">MALT1</span></span>
<span id="cb111-71"><a href="#cb111-71" aria-hidden="true" tabindex="-1"></a><span class="st">CPEB3</span></span>
<span id="cb111-72"><a href="#cb111-72" aria-hidden="true" tabindex="-1"></a><span class="st">LOC648605</span></span>
<span id="cb111-73"><a href="#cb111-73" aria-hidden="true" tabindex="-1"></a><span class="st">TXNIP</span></span>
<span id="cb111-74"><a href="#cb111-74" aria-hidden="true" tabindex="-1"></a><span class="st">NaN</span></span>
<span id="cb111-75"><a href="#cb111-75" aria-hidden="true" tabindex="-1"></a><span class="st">TMEM51</span></span>
<span id="cb111-76"><a href="#cb111-76" aria-hidden="true" tabindex="-1"></a><span class="st">ETV6</span></span>
<span id="cb111-77"><a href="#cb111-77" aria-hidden="true" tabindex="-1"></a><span class="st">SUMO4</span></span>
<span id="cb111-78"><a href="#cb111-78" aria-hidden="true" tabindex="-1"></a><span class="st">MTDH</span></span>
<span id="cb111-79"><a href="#cb111-79" aria-hidden="true" tabindex="-1"></a><span class="st">RAB11FIP1</span></span>
<span id="cb111-80"><a href="#cb111-80" aria-hidden="true" tabindex="-1"></a><span class="st">DNAJC2</span></span>
<span id="cb111-81"><a href="#cb111-81" aria-hidden="true" tabindex="-1"></a><span class="st">LOC399715</span></span>
<span id="cb111-82"><a href="#cb111-82" aria-hidden="true" tabindex="-1"></a><span class="st">NCBP2</span></span>
<span id="cb111-83"><a href="#cb111-83" aria-hidden="true" tabindex="-1"></a><span class="st">EPB41L4A</span></span>
<span id="cb111-84"><a href="#cb111-84" aria-hidden="true" tabindex="-1"></a><span class="st">NaN</span></span>
<span id="cb111-85"><a href="#cb111-85" aria-hidden="true" tabindex="-1"></a><span class="st">ZNF516</span></span>
<span id="cb111-86"><a href="#cb111-86" aria-hidden="true" tabindex="-1"></a><span class="st">NaN</span></span>
<span id="cb111-87"><a href="#cb111-87" aria-hidden="true" tabindex="-1"></a><span class="st">LOC648210</span></span>
<span id="cb111-88"><a href="#cb111-88" aria-hidden="true" tabindex="-1"></a><span class="st">LOC285359</span></span>
<span id="cb111-89"><a href="#cb111-89" aria-hidden="true" tabindex="-1"></a><span class="st">FLVCR2</span></span>
<span id="cb111-90"><a href="#cb111-90" aria-hidden="true" tabindex="-1"></a><span class="st">RBBP7</span></span>
<span id="cb111-91"><a href="#cb111-91" aria-hidden="true" tabindex="-1"></a><span class="st">HSPD1</span></span>
<span id="cb111-92"><a href="#cb111-92" aria-hidden="true" tabindex="-1"></a><span class="st">UAP1</span></span>
<span id="cb111-93"><a href="#cb111-93" aria-hidden="true" tabindex="-1"></a><span class="st">ZWILCH</span></span>
<span id="cb111-94"><a href="#cb111-94" aria-hidden="true" tabindex="-1"></a><span class="st">NCL</span></span>
<span id="cb111-95"><a href="#cb111-95" aria-hidden="true" tabindex="-1"></a><span class="st">TRAFD1</span></span>
<span id="cb111-96"><a href="#cb111-96" aria-hidden="true" tabindex="-1"></a><span class="st">RNF181</span></span>
<span id="cb111-97"><a href="#cb111-97" aria-hidden="true" tabindex="-1"></a><span class="st">PACSIN2</span></span>
<span id="cb111-98"><a href="#cb111-98" aria-hidden="true" tabindex="-1"></a><span class="st">PDCD6</span></span>
<span id="cb111-99"><a href="#cb111-99" aria-hidden="true" tabindex="-1"></a><span class="st">FLJ38973</span></span>
<span id="cb111-100"><a href="#cb111-100" aria-hidden="true" tabindex="-1"></a><span class="st">WDR43</span></span>
<span id="cb111-101"><a href="#cb111-101" aria-hidden="true" tabindex="-1"></a><span class="st">PIK3R1</span></span>
<span id="cb111-102"><a href="#cb111-102" aria-hidden="true" tabindex="-1"></a><span class="st">WBP2</span></span>
<span id="cb111-103"><a href="#cb111-103" aria-hidden="true" tabindex="-1"></a><span class="st">RPRC1</span></span>
<span id="cb111-104"><a href="#cb111-104" aria-hidden="true" tabindex="-1"></a><span class="st">C1orf54</span></span>
<span id="cb111-105"><a href="#cb111-105" aria-hidden="true" tabindex="-1"></a><span class="st">C6orf47</span></span>
<span id="cb111-106"><a href="#cb111-106" aria-hidden="true" tabindex="-1"></a><span class="st">NaN</span></span>
<span id="cb111-107"><a href="#cb111-107" aria-hidden="true" tabindex="-1"></a><span class="st">DNAL4</span></span>
<span id="cb111-108"><a href="#cb111-108" aria-hidden="true" tabindex="-1"></a><span class="st">KIAA0319L</span></span>
<span id="cb111-109"><a href="#cb111-109" aria-hidden="true" tabindex="-1"></a><span class="st">SUB1</span></span>
<span id="cb111-110"><a href="#cb111-110" aria-hidden="true" tabindex="-1"></a><span class="st">C19orf56</span></span>
<span id="cb111-111"><a href="#cb111-111" aria-hidden="true" tabindex="-1"></a><span class="st">TNPO1</span></span>
<span id="cb111-112"><a href="#cb111-112" aria-hidden="true" tabindex="-1"></a><span class="st">G6PD</span></span>
<span id="cb111-113"><a href="#cb111-113" aria-hidden="true" tabindex="-1"></a><span class="st">LOC91431</span></span>
<span id="cb111-114"><a href="#cb111-114" aria-hidden="true" tabindex="-1"></a><span class="st">LOC648057</span></span>
<span id="cb111-115"><a href="#cb111-115" aria-hidden="true" tabindex="-1"></a><span class="st">SNORD36A</span></span>
<span id="cb111-116"><a href="#cb111-116" aria-hidden="true" tabindex="-1"></a><span class="st">CDKN2D</span></span>
<span id="cb111-117"><a href="#cb111-117" aria-hidden="true" tabindex="-1"></a><span class="st">LMO4</span></span>
<span id="cb111-118"><a href="#cb111-118" aria-hidden="true" tabindex="-1"></a><span class="st">ASCC3</span></span>
<span id="cb111-119"><a href="#cb111-119" aria-hidden="true" tabindex="-1"></a><span class="st">C7orf36</span></span>
<span id="cb111-120"><a href="#cb111-120" aria-hidden="true" tabindex="-1"></a><span class="st">MPZL2</span></span>
<span id="cb111-121"><a href="#cb111-121" aria-hidden="true" tabindex="-1"></a><span class="st">NPTN</span></span>
<span id="cb111-122"><a href="#cb111-122" aria-hidden="true" tabindex="-1"></a><span class="st">TGIF1</span></span>
<span id="cb111-123"><a href="#cb111-123" aria-hidden="true" tabindex="-1"></a><span class="st">UBE2Q2</span></span>
<span id="cb111-124"><a href="#cb111-124" aria-hidden="true" tabindex="-1"></a><span class="st">NaN</span></span>
<span id="cb111-125"><a href="#cb111-125" aria-hidden="true" tabindex="-1"></a><span class="st">CLK1</span></span>
<span id="cb111-126"><a href="#cb111-126" aria-hidden="true" tabindex="-1"></a><span class="st">KIAA0146</span></span>
<span id="cb111-127"><a href="#cb111-127" aria-hidden="true" tabindex="-1"></a><span class="st">RNF19A</span></span>
<span id="cb111-128"><a href="#cb111-128" aria-hidden="true" tabindex="-1"></a><span class="st">NEDD8</span></span>
<span id="cb111-129"><a href="#cb111-129" aria-hidden="true" tabindex="-1"></a><span class="st">LOC390354</span></span>
<span id="cb111-130"><a href="#cb111-130" aria-hidden="true" tabindex="-1"></a><span class="st">YME1L1</span></span>
<span id="cb111-131"><a href="#cb111-131" aria-hidden="true" tabindex="-1"></a><span class="st">LPCAT2</span></span>
<span id="cb111-132"><a href="#cb111-132" aria-hidden="true" tabindex="-1"></a><span class="st">NaN</span></span>
<span id="cb111-133"><a href="#cb111-133" aria-hidden="true" tabindex="-1"></a><span class="st">PCDHA1</span></span>
<span id="cb111-134"><a href="#cb111-134" aria-hidden="true" tabindex="-1"></a><span class="st">LZTFL1</span></span>
<span id="cb111-135"><a href="#cb111-135" aria-hidden="true" tabindex="-1"></a><span class="st">SEPT5</span></span>
<span id="cb111-136"><a href="#cb111-136" aria-hidden="true" tabindex="-1"></a><span class="st">CCDC99</span></span>
<span id="cb111-137"><a href="#cb111-137" aria-hidden="true" tabindex="-1"></a><span class="st">LOC285176</span></span>
<span id="cb111-138"><a href="#cb111-138" aria-hidden="true" tabindex="-1"></a><span class="st">KIAA1370</span></span>
<span id="cb111-139"><a href="#cb111-139" aria-hidden="true" tabindex="-1"></a><span class="st">MRPL47</span></span>
<span id="cb111-140"><a href="#cb111-140" aria-hidden="true" tabindex="-1"></a><span class="st">C9orf91</span></span>
<span id="cb111-141"><a href="#cb111-141" aria-hidden="true" tabindex="-1"></a><span class="st">KIAA1333</span></span>
<span id="cb111-142"><a href="#cb111-142" aria-hidden="true" tabindex="-1"></a><span class="st">FAM120A</span></span>
<span id="cb111-143"><a href="#cb111-143" aria-hidden="true" tabindex="-1"></a><span class="st">LOC644380</span></span>
<span id="cb111-144"><a href="#cb111-144" aria-hidden="true" tabindex="-1"></a><span class="st">MYL6</span></span>
<span id="cb111-145"><a href="#cb111-145" aria-hidden="true" tabindex="-1"></a><span class="st">BZW1</span></span>
<span id="cb111-146"><a href="#cb111-146" aria-hidden="true" tabindex="-1"></a><span class="st">ASPHD2</span></span>
<span id="cb111-147"><a href="#cb111-147" aria-hidden="true" tabindex="-1"></a><span class="st">ABCE1</span></span>
<span id="cb111-148"><a href="#cb111-148" aria-hidden="true" tabindex="-1"></a><span class="st">PCGF6</span></span>
<span id="cb111-149"><a href="#cb111-149" aria-hidden="true" tabindex="-1"></a><span class="st">TSPYL2</span></span>
<span id="cb111-150"><a href="#cb111-150" aria-hidden="true" tabindex="-1"></a><span class="st">NCOA4</span></span>
<span id="cb111-151"><a href="#cb111-151" aria-hidden="true" tabindex="-1"></a><span class="st">SFXN1</span></span>
<span id="cb111-152"><a href="#cb111-152" aria-hidden="true" tabindex="-1"></a><span class="st">IRF1</span></span>
<span id="cb111-153"><a href="#cb111-153" aria-hidden="true" tabindex="-1"></a><span class="st">COASY</span></span>
<span id="cb111-154"><a href="#cb111-154" aria-hidden="true" tabindex="-1"></a><span class="st">TAOK2</span></span>
<span id="cb111-155"><a href="#cb111-155" aria-hidden="true" tabindex="-1"></a><span class="st">RAB24</span></span>
<span id="cb111-156"><a href="#cb111-156" aria-hidden="true" tabindex="-1"></a><span class="st">AEBP2</span></span>
<span id="cb111-157"><a href="#cb111-157" aria-hidden="true" tabindex="-1"></a><span class="st">PPP2CB</span></span>
<span id="cb111-158"><a href="#cb111-158" aria-hidden="true" tabindex="-1"></a><span class="st">HLA-DMB</span></span>
<span id="cb111-159"><a href="#cb111-159" aria-hidden="true" tabindex="-1"></a><span class="st">SFRS2IP</span></span>
<span id="cb111-160"><a href="#cb111-160" aria-hidden="true" tabindex="-1"></a><span class="st">NaN</span></span>
<span id="cb111-161"><a href="#cb111-161" aria-hidden="true" tabindex="-1"></a><span class="st">USP16</span></span>
<span id="cb111-162"><a href="#cb111-162" aria-hidden="true" tabindex="-1"></a><span class="st">KPNA3</span></span>
<span id="cb111-163"><a href="#cb111-163" aria-hidden="true" tabindex="-1"></a><span class="st">MAN1A1</span></span>
<span id="cb111-164"><a href="#cb111-164" aria-hidden="true" tabindex="-1"></a><span class="st">SDHAP3</span></span>
<span id="cb111-165"><a href="#cb111-165" aria-hidden="true" tabindex="-1"></a><span class="st">HIF1AN</span></span>
<span id="cb111-166"><a href="#cb111-166" aria-hidden="true" tabindex="-1"></a><span class="st">SLC25A3</span></span>
<span id="cb111-167"><a href="#cb111-167" aria-hidden="true" tabindex="-1"></a><span class="st">PCMT1</span></span>
<span id="cb111-168"><a href="#cb111-168" aria-hidden="true" tabindex="-1"></a><span class="st">BZW1</span></span>
<span id="cb111-169"><a href="#cb111-169" aria-hidden="true" tabindex="-1"></a><span class="st">ZNF124</span></span>
<span id="cb111-170"><a href="#cb111-170" aria-hidden="true" tabindex="-1"></a><span class="st">ZNF679</span></span>
<span id="cb111-171"><a href="#cb111-171" aria-hidden="true" tabindex="-1"></a><span class="st">S100Z</span></span>
<span id="cb111-172"><a href="#cb111-172" aria-hidden="true" tabindex="-1"></a><span class="st">STAT2</span></span>
<span id="cb111-173"><a href="#cb111-173" aria-hidden="true" tabindex="-1"></a><span class="st">WAS</span></span>
<span id="cb111-174"><a href="#cb111-174" aria-hidden="true" tabindex="-1"></a><span class="st">DHRS4</span></span>
<span id="cb111-175"><a href="#cb111-175" aria-hidden="true" tabindex="-1"></a><span class="st">TRIM23</span></span>
<span id="cb111-176"><a href="#cb111-176" aria-hidden="true" tabindex="-1"></a><span class="st">NaN</span></span>
<span id="cb111-177"><a href="#cb111-177" aria-hidden="true" tabindex="-1"></a><span class="st">MEFV</span></span>
<span id="cb111-178"><a href="#cb111-178" aria-hidden="true" tabindex="-1"></a><span class="st">R3HDM1</span></span>
<span id="cb111-179"><a href="#cb111-179" aria-hidden="true" tabindex="-1"></a><span class="st">DHRS9</span></span>
<span id="cb111-180"><a href="#cb111-180" aria-hidden="true" tabindex="-1"></a><span class="st">LOC653610</span></span>
<span id="cb111-181"><a href="#cb111-181" aria-hidden="true" tabindex="-1"></a><span class="st">DGAT1</span></span>
<span id="cb111-182"><a href="#cb111-182" aria-hidden="true" tabindex="-1"></a><span class="st">DNAJB6</span></span>
<span id="cb111-183"><a href="#cb111-183" aria-hidden="true" tabindex="-1"></a><span class="st">SFRS3</span></span>
<span id="cb111-184"><a href="#cb111-184" aria-hidden="true" tabindex="-1"></a><span class="st">EED</span></span>
<span id="cb111-185"><a href="#cb111-185" aria-hidden="true" tabindex="-1"></a><span class="st">NaN</span></span>
<span id="cb111-186"><a href="#cb111-186" aria-hidden="true" tabindex="-1"></a><span class="st">ZNF254</span></span>
<span id="cb111-187"><a href="#cb111-187" aria-hidden="true" tabindex="-1"></a><span class="st">ANKRD44</span></span>
<span id="cb111-188"><a href="#cb111-188" aria-hidden="true" tabindex="-1"></a><span class="st">CDK6</span></span>
<span id="cb111-189"><a href="#cb111-189" aria-hidden="true" tabindex="-1"></a><span class="st">ZNF451</span></span>
<span id="cb111-190"><a href="#cb111-190" aria-hidden="true" tabindex="-1"></a><span class="st">NAE1</span></span>
<span id="cb111-191"><a href="#cb111-191" aria-hidden="true" tabindex="-1"></a><span class="st">REV3L</span></span>
<span id="cb111-192"><a href="#cb111-192" aria-hidden="true" tabindex="-1"></a><span class="st">ITGA4</span></span>
<span id="cb111-193"><a href="#cb111-193" aria-hidden="true" tabindex="-1"></a><span class="st">UBA2</span></span>
<span id="cb111-194"><a href="#cb111-194" aria-hidden="true" tabindex="-1"></a><span class="st">PGCP</span></span>
<span id="cb111-195"><a href="#cb111-195" aria-hidden="true" tabindex="-1"></a><span class="st">GOLT1B</span></span>
<span id="cb111-196"><a href="#cb111-196" aria-hidden="true" tabindex="-1"></a><span class="st">NARG2</span></span>
<span id="cb111-197"><a href="#cb111-197" aria-hidden="true" tabindex="-1"></a><span class="st">PRCP</span></span>
<span id="cb111-198"><a href="#cb111-198" aria-hidden="true" tabindex="-1"></a><span class="st">TIPIN</span></span>
<span id="cb111-199"><a href="#cb111-199" aria-hidden="true" tabindex="-1"></a><span class="st">SNORA25</span></span>
<span id="cb111-200"><a href="#cb111-200" aria-hidden="true" tabindex="-1"></a><span class="st">CASP4</span></span>
<span id="cb111-201"><a href="#cb111-201" aria-hidden="true" tabindex="-1"></a><span class="st">PCF11</span></span>
<span id="cb111-202"><a href="#cb111-202" aria-hidden="true" tabindex="-1"></a><span class="st">LOC649555</span></span>
<span id="cb111-203"><a href="#cb111-203" aria-hidden="true" tabindex="-1"></a><span class="st">TPMT</span></span>
<span id="cb111-204"><a href="#cb111-204" aria-hidden="true" tabindex="-1"></a><span class="st">RANBP2</span></span>
<span id="cb111-205"><a href="#cb111-205" aria-hidden="true" tabindex="-1"></a><span class="st">TCP11L1</span></span>
<span id="cb111-206"><a href="#cb111-206" aria-hidden="true" tabindex="-1"></a><span class="st">NCL</span></span>
<span id="cb111-207"><a href="#cb111-207" aria-hidden="true" tabindex="-1"></a><span class="st">LOC654189</span></span>
<span id="cb111-208"><a href="#cb111-208" aria-hidden="true" tabindex="-1"></a><span class="st">ZC3H15</span></span>
<span id="cb111-209"><a href="#cb111-209" aria-hidden="true" tabindex="-1"></a><span class="st">G6PD</span></span>
<span id="cb111-210"><a href="#cb111-210" aria-hidden="true" tabindex="-1"></a><span class="st">NAGK</span></span>
<span id="cb111-211"><a href="#cb111-211" aria-hidden="true" tabindex="-1"></a><span class="st">SUMF1</span></span>
<span id="cb111-212"><a href="#cb111-212" aria-hidden="true" tabindex="-1"></a><span class="st">CARM1</span></span>
<span id="cb111-213"><a href="#cb111-213" aria-hidden="true" tabindex="-1"></a><span class="st">PPBP</span></span>
<span id="cb111-214"><a href="#cb111-214" aria-hidden="true" tabindex="-1"></a><span class="st">FASTKD2</span></span>
<span id="cb111-215"><a href="#cb111-215" aria-hidden="true" tabindex="-1"></a><span class="st">C10orf54</span></span>
<span id="cb111-216"><a href="#cb111-216" aria-hidden="true" tabindex="-1"></a><span class="st">PDE4D</span></span>
<span id="cb111-217"><a href="#cb111-217" aria-hidden="true" tabindex="-1"></a><span class="st">NaN</span></span>
<span id="cb111-218"><a href="#cb111-218" aria-hidden="true" tabindex="-1"></a><span class="st">KRAS</span></span>
<span id="cb111-219"><a href="#cb111-219" aria-hidden="true" tabindex="-1"></a><span class="st">ARSA</span></span>
<span id="cb111-220"><a href="#cb111-220" aria-hidden="true" tabindex="-1"></a><span class="st">MRPL35</span></span>
<span id="cb111-221"><a href="#cb111-221" aria-hidden="true" tabindex="-1"></a><span class="st">ARL4C</span></span>
<span id="cb111-222"><a href="#cb111-222" aria-hidden="true" tabindex="-1"></a><span class="st">SMAD7</span></span>
<span id="cb111-223"><a href="#cb111-223" aria-hidden="true" tabindex="-1"></a><span class="st">SNORA32</span></span>
<span id="cb111-224"><a href="#cb111-224" aria-hidden="true" tabindex="-1"></a><span class="st">CCNC</span></span>
<span id="cb111-225"><a href="#cb111-225" aria-hidden="true" tabindex="-1"></a><span class="st">UBE2L6</span></span>
<span id="cb111-226"><a href="#cb111-226" aria-hidden="true" tabindex="-1"></a><span class="st">C2orf79</span></span>
<span id="cb111-227"><a href="#cb111-227" aria-hidden="true" tabindex="-1"></a><span class="st">TMLHE</span></span>
<span id="cb111-228"><a href="#cb111-228" aria-hidden="true" tabindex="-1"></a><span class="st">LOC90120</span></span>
<span id="cb111-229"><a href="#cb111-229" aria-hidden="true" tabindex="-1"></a><span class="st">MALT1</span></span>
<span id="cb111-230"><a href="#cb111-230" aria-hidden="true" tabindex="-1"></a><span class="st">IFNAR2</span></span>
<span id="cb111-231"><a href="#cb111-231" aria-hidden="true" tabindex="-1"></a><span class="st">FLJ10996</span></span>
<span id="cb111-232"><a href="#cb111-232" aria-hidden="true" tabindex="-1"></a><span class="st">LOC647436</span></span>
<span id="cb111-233"><a href="#cb111-233" aria-hidden="true" tabindex="-1"></a><span class="st">TAF9</span></span>
<span id="cb111-234"><a href="#cb111-234" aria-hidden="true" tabindex="-1"></a><span class="st">NPIP</span></span>
<span id="cb111-235"><a href="#cb111-235" aria-hidden="true" tabindex="-1"></a><span class="st">YME1L1</span></span>
<span id="cb111-236"><a href="#cb111-236" aria-hidden="true" tabindex="-1"></a><span class="st">EED</span></span>
<span id="cb111-237"><a href="#cb111-237" aria-hidden="true" tabindex="-1"></a><span class="st">ZNF624</span></span>
<span id="cb111-238"><a href="#cb111-238" aria-hidden="true" tabindex="-1"></a><span class="st">ANAPC1</span></span>
<span id="cb111-239"><a href="#cb111-239" aria-hidden="true" tabindex="-1"></a><span class="st">ZNF669</span></span>
<span id="cb111-240"><a href="#cb111-240" aria-hidden="true" tabindex="-1"></a><span class="st">STAT1</span></span>
<span id="cb111-241"><a href="#cb111-241" aria-hidden="true" tabindex="-1"></a><span class="st">SNX27</span></span>
<span id="cb111-242"><a href="#cb111-242" aria-hidden="true" tabindex="-1"></a><span class="st">MOBKL2A</span></span>
<span id="cb111-243"><a href="#cb111-243" aria-hidden="true" tabindex="-1"></a><span class="st">HSP90AA1</span></span>
<span id="cb111-244"><a href="#cb111-244" aria-hidden="true" tabindex="-1"></a><span class="st">ZNF644</span></span>
<span id="cb111-245"><a href="#cb111-245" aria-hidden="true" tabindex="-1"></a><span class="st">KIAA0020</span></span>
<span id="cb111-246"><a href="#cb111-246" aria-hidden="true" tabindex="-1"></a><span class="st">LOC728643</span></span>
<span id="cb111-247"><a href="#cb111-247" aria-hidden="true" tabindex="-1"></a><span class="st">LOC440145</span></span>
<span id="cb111-248"><a href="#cb111-248" aria-hidden="true" tabindex="-1"></a><span class="st">GEMIN6</span></span>
<span id="cb111-249"><a href="#cb111-249" aria-hidden="true" tabindex="-1"></a><span class="st">DDHD1</span></span>
<span id="cb111-250"><a href="#cb111-250" aria-hidden="true" tabindex="-1"></a><span class="st">BCLAF1</span></span>
<span id="cb111-251"><a href="#cb111-251" aria-hidden="true" tabindex="-1"></a><span class="st">CLTB</span></span>
<span id="cb111-252"><a href="#cb111-252" aria-hidden="true" tabindex="-1"></a><span class="st">"</span>)</span>
<span id="cb111-253"><a href="#cb111-253" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb111-254"><a href="#cb111-254" aria-hidden="true" tabindex="-1"></a>signature_251_list <span class="ot">&lt;-</span> <span class="fu">str_split</span>(signature_251_str, <span class="st">"</span><span class="sc">\n</span><span class="st">"</span>)[[<span class="dv">1</span>]]</span>
<span id="cb111-255"><a href="#cb111-255" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb111-256"><a href="#cb111-256" aria-hidden="true" tabindex="-1"></a>intersect_genes <span class="ot">&lt;-</span> <span class="fu">intersect</span>(significant_genes, signature_251_list)</span>
<span id="cb111-257"><a href="#cb111-257" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb111-258"><a href="#cb111-258" aria-hidden="true" tabindex="-1"></a>dat_estimates <span class="ot">&lt;-</span> dat_estimates <span class="sc">|&gt;</span> </span>
<span id="cb111-259"><a href="#cb111-259" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">signature_251 =</span> <span class="fu">case_when</span>(gene_symbol <span class="sc">%in%</span> intersect_genes <span class="sc">~</span> <span class="st">"Yes"</span>, <span class="cn">TRUE</span> <span class="sc">~</span> <span class="st">"No"</span>))</span>
<span id="cb111-260"><a href="#cb111-260" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb111-261"><a href="#cb111-261" aria-hidden="true" tabindex="-1"></a>intersect_genes</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code> [1] "CMIP"    "DNAJC2"  "HPCAL1"  "BZW1"    "MRPL47"  "NAE1"    "PIK3R1" 
 [8] "RBBP7"   "TMEM164" "USP16"  </code></pre>
</div>
</div>
<p>We find 10 overlapping genes between the 40 significant genes from the linear analysis and the 251 signature genes found by a SVM algorithm from the reference article.</p>
</section>
<section id="forest-plot" class="level3">
<h3 class="anchored" data-anchor-id="forest-plot">Forest Plot</h3>
<p>A forest plot is generated showing the genes that are significantly up- or down-regulated in tuberculosis patients (TB).</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb113"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb113-1"><a href="#cb113-1" aria-hidden="true" tabindex="-1"></a>dat_forest_plot <span class="ot">&lt;-</span> dat_estimates <span class="sc">|&gt;</span> </span>
<span id="cb113-2"><a href="#cb113-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(significant <span class="sc">==</span> <span class="st">"Yes"</span>) </span>
<span id="cb113-3"><a href="#cb113-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb113-4"><a href="#cb113-4" aria-hidden="true" tabindex="-1"></a>forest_plot <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(<span class="at">data =</span> dat_forest_plot, </span>
<span id="cb113-5"><a href="#cb113-5" aria-hidden="true" tabindex="-1"></a>                      <span class="fu">aes</span>(<span class="at">x=</span>estimate, </span>
<span id="cb113-6"><a href="#cb113-6" aria-hidden="true" tabindex="-1"></a>                          <span class="at">y=</span><span class="fu">fct_reorder</span>(gene_symbol, estimate), </span>
<span id="cb113-7"><a href="#cb113-7" aria-hidden="true" tabindex="-1"></a>                          <span class="at">xmin =</span> conf.low, </span>
<span id="cb113-8"><a href="#cb113-8" aria-hidden="true" tabindex="-1"></a>                          <span class="at">xmax =</span> conf.high,</span>
<span id="cb113-9"><a href="#cb113-9" aria-hidden="true" tabindex="-1"></a>                          <span class="at">color =</span> signature_251)) <span class="sc">+</span></span>
<span id="cb113-10"><a href="#cb113-10" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_errorbarh</span>() <span class="sc">+</span></span>
<span id="cb113-11"><a href="#cb113-11" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_point</span>() <span class="sc">+</span></span>
<span id="cb113-12"><a href="#cb113-12" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_color_manual</span>(<span class="at">values =</span> <span class="fu">c</span>(<span class="st">"Yes"</span> <span class="ot">=</span> <span class="st">"royalblue"</span>, </span>
<span id="cb113-13"><a href="#cb113-13" aria-hidden="true" tabindex="-1"></a>                                <span class="st">"No"</span> <span class="ot">=</span> <span class="st">"gray"</span>)) <span class="sc">+</span></span>
<span id="cb113-14"><a href="#cb113-14" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">x =</span> <span class="st">"Estimates (95% CIs)"</span>, </span>
<span id="cb113-15"><a href="#cb113-15" aria-hidden="true" tabindex="-1"></a>       <span class="at">y =</span> <span class="st">""</span>, </span>
<span id="cb113-16"><a href="#cb113-16" aria-hidden="true" tabindex="-1"></a>       <span class="at">title =</span> <span class="st">"Genes Associated with Tuberculosis in HIV patients"</span>) <span class="sc">+</span></span>
<span id="cb113-17"><a href="#cb113-17" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_minimal</span>(<span class="at">base_size =</span> <span class="dv">14</span>) <span class="sc">+</span> </span>
<span id="cb113-18"><a href="#cb113-18" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">x =</span> <span class="st">"Estimates (95% CIs)"</span>,</span>
<span id="cb113-19"><a href="#cb113-19" aria-hidden="true" tabindex="-1"></a>       <span class="at">y =</span> <span class="st">"Genes"</span>,</span>
<span id="cb113-20"><a href="#cb113-20" aria-hidden="true" tabindex="-1"></a>       <span class="at">title =</span> <span class="st">"TB Associated Genes"</span>,</span>
<span id="cb113-21"><a href="#cb113-21" aria-hidden="true" tabindex="-1"></a>       <span class="at">subtitle =</span> <span class="fu">str_c</span>(<span class="st">"Genes Significantly Associated with Tuberculosis in HIV patients"</span>,</span>
<span id="cb113-22"><a href="#cb113-22" aria-hidden="true" tabindex="-1"></a>                        <span class="st">"</span><span class="sc">\n</span><span class="st">"</span>,</span>
<span id="cb113-23"><a href="#cb113-23" aria-hidden="true" tabindex="-1"></a>                        <span class="st">""</span>),</span>
<span id="cb113-24"><a href="#cb113-24" aria-hidden="true" tabindex="-1"></a>       <span class="at">color =</span> <span class="st">"Genes in SVM 251 Signature"</span>,</span>
<span id="cb113-25"><a href="#cb113-25" aria-hidden="true" tabindex="-1"></a>       <span class="at">caption =</span> <span class="st">"Data from Dawany et al. (2014)"</span>) <span class="sc">+</span></span>
<span id="cb113-26"><a href="#cb113-26" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">axis.text.x =</span> <span class="fu">element_text</span>(<span class="at">color =</span> <span class="st">"black"</span>, </span>
<span id="cb113-27"><a href="#cb113-27" aria-hidden="true" tabindex="-1"></a>                                   <span class="at">size =</span> <span class="dv">14</span>),</span>
<span id="cb113-28"><a href="#cb113-28" aria-hidden="true" tabindex="-1"></a>        <span class="at">axis.text.y =</span> <span class="fu">element_text</span>(<span class="at">color =</span> <span class="st">"black"</span>, </span>
<span id="cb113-29"><a href="#cb113-29" aria-hidden="true" tabindex="-1"></a>                                   <span class="at">size =</span> <span class="dv">8</span>),</span>
<span id="cb113-30"><a href="#cb113-30" aria-hidden="true" tabindex="-1"></a>        <span class="at">axis.title =</span> <span class="fu">element_text</span>(<span class="at">face=</span><span class="st">"bold"</span>),</span>
<span id="cb113-31"><a href="#cb113-31" aria-hidden="true" tabindex="-1"></a>        <span class="at">plot.caption =</span> <span class="fu">element_text</span>(<span class="at">face =</span> <span class="st">"italic"</span>),</span>
<span id="cb113-32"><a href="#cb113-32" aria-hidden="true" tabindex="-1"></a>        <span class="at">legend.text =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="dv">14</span>),</span>
<span id="cb113-33"><a href="#cb113-33" aria-hidden="true" tabindex="-1"></a>        <span class="at">legend.position =</span> <span class="st">"bottom"</span>)</span>
<span id="cb113-34"><a href="#cb113-34" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb113-35"><a href="#cb113-35" aria-hidden="true" tabindex="-1"></a><span class="fu">ggsave</span>(<span class="st">"../results/forest_plot.png"</span>, <span class="at">plot =</span> forest_plot) </span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Saving 7 x 5 in image</code></pre>
</div>
<div class="sourceCode cell-code" id="cb115"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb115-1"><a href="#cb115-1" aria-hidden="true" tabindex="-1"></a><span class="fu">show</span>(forest_plot)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<p><img src="00_all_files/figure-html/06_Forrest%20plot-1.png" class="img-fluid" width="672"></p>
</div>
</div>
<p>We see that most genes are down-regulated.</p>
</section>
<section id="volcano-plot" class="level3">
<h3 class="anchored" data-anchor-id="volcano-plot">Volcano Plot</h3>
<p>A volcano plot is generated to show significance of the genes as a function of the gene expression estimates.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb116"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb116-1"><a href="#cb116-1" aria-hidden="true" tabindex="-1"></a>dat_volcano_plot <span class="ot">&lt;-</span> dat_estimates <span class="sc">|&gt;</span> </span>
<span id="cb116-2"><a href="#cb116-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">gene_labels =</span> <span class="fu">case_when</span>(</span>
<span id="cb116-3"><a href="#cb116-3" aria-hidden="true" tabindex="-1"></a>    significant <span class="sc">==</span> <span class="st">"Yes"</span> <span class="sc">&amp;</span> signature_251 <span class="sc">==</span> <span class="st">"Yes"</span> <span class="sc">~</span> gene_symbol, </span>
<span id="cb116-4"><a href="#cb116-4" aria-hidden="true" tabindex="-1"></a>                                 <span class="cn">TRUE</span> <span class="sc">~</span> <span class="st">""</span>)) </span>
<span id="cb116-5"><a href="#cb116-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb116-6"><a href="#cb116-6" aria-hidden="true" tabindex="-1"></a>volcano_plot <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(<span class="at">data =</span> dat_volcano_plot, </span>
<span id="cb116-7"><a href="#cb116-7" aria-hidden="true" tabindex="-1"></a>                       <span class="fu">aes</span>(<span class="at">x =</span> estimate,</span>
<span id="cb116-8"><a href="#cb116-8" aria-hidden="true" tabindex="-1"></a>                           <span class="at">y =</span> <span class="sc">-</span><span class="fu">log10</span>(p.value), </span>
<span id="cb116-9"><a href="#cb116-9" aria-hidden="true" tabindex="-1"></a>                           <span class="at">color =</span> significant,</span>
<span id="cb116-10"><a href="#cb116-10" aria-hidden="true" tabindex="-1"></a>                           <span class="at">label =</span> gene_labels</span>
<span id="cb116-11"><a href="#cb116-11" aria-hidden="true" tabindex="-1"></a>                           )) <span class="sc">+</span></span>
<span id="cb116-12"><a href="#cb116-12" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_point</span>(<span class="fu">aes</span>(<span class="at">alpha =</span> <span class="fl">0.7</span>)) <span class="sc">+</span></span>
<span id="cb116-13"><a href="#cb116-13" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_point</span>(<span class="at">data=</span>dat_volcano_plot <span class="sc">|&gt;</span> </span>
<span id="cb116-14"><a href="#cb116-14" aria-hidden="true" tabindex="-1"></a>               <span class="fu">filter</span>(signature_251 <span class="sc">==</span> <span class="st">"Yes"</span>),</span>
<span id="cb116-15"><a href="#cb116-15" aria-hidden="true" tabindex="-1"></a>             <span class="at">pch=</span><span class="dv">21</span>,</span>
<span id="cb116-16"><a href="#cb116-16" aria-hidden="true" tabindex="-1"></a>             <span class="at">size=</span><span class="fl">2.5</span>,</span>
<span id="cb116-17"><a href="#cb116-17" aria-hidden="true" tabindex="-1"></a>             <span class="at">colour=</span><span class="st">"black"</span>) <span class="sc">+</span></span>
<span id="cb116-18"><a href="#cb116-18" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_color_manual</span>(<span class="at">values =</span> <span class="fu">c</span>(<span class="st">"Yes"</span> <span class="ot">=</span> <span class="st">"royalblue"</span>, </span>
<span id="cb116-19"><a href="#cb116-19" aria-hidden="true" tabindex="-1"></a>                                <span class="st">"No"</span> <span class="ot">=</span> <span class="st">"gray"</span>)) <span class="sc">+</span></span>
<span id="cb116-20"><a href="#cb116-20" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_minimal</span>(<span class="at">base_size =</span> <span class="dv">16</span>) <span class="sc">+</span> </span>
<span id="cb116-21"><a href="#cb116-21" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">x =</span> <span class="st">"Estimates"</span>,</span>
<span id="cb116-22"><a href="#cb116-22" aria-hidden="true" tabindex="-1"></a>       <span class="at">y =</span> <span class="st">"-log10(p)"</span>,</span>
<span id="cb116-23"><a href="#cb116-23" aria-hidden="true" tabindex="-1"></a>       <span class="at">title =</span> <span class="st">"Genes Associated with Tuberculosis"</span>,</span>
<span id="cb116-24"><a href="#cb116-24" aria-hidden="true" tabindex="-1"></a>       <span class="at">subtitle =</span> <span class="fu">str_c</span>(<span class="st">"Genes highlighted in blue were significant after multiple"</span>,</span>
<span id="cb116-25"><a href="#cb116-25" aria-hidden="true" tabindex="-1"></a>                        <span class="st">"</span><span class="sc">\n</span><span class="st">"</span>,</span>
<span id="cb116-26"><a href="#cb116-26" aria-hidden="true" tabindex="-1"></a>                        <span class="st">"testing correction."</span>),</span>
<span id="cb116-27"><a href="#cb116-27" aria-hidden="true" tabindex="-1"></a>       <span class="at">caption =</span> <span class="st">"Data from Dawany et al. (2014)"</span>) <span class="sc">+</span></span>
<span id="cb116-28"><a href="#cb116-28" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">legend.position =</span> <span class="st">"none"</span>,</span>
<span id="cb116-29"><a href="#cb116-29" aria-hidden="true" tabindex="-1"></a>        <span class="at">axis.text.x =</span> <span class="fu">element_text</span>(<span class="at">color =</span> <span class="st">"black"</span>, </span>
<span id="cb116-30"><a href="#cb116-30" aria-hidden="true" tabindex="-1"></a>                                   <span class="at">size =</span> <span class="dv">14</span>),</span>
<span id="cb116-31"><a href="#cb116-31" aria-hidden="true" tabindex="-1"></a>        <span class="at">axis.text.y =</span> <span class="fu">element_text</span>(<span class="at">color =</span> <span class="st">"black"</span>, </span>
<span id="cb116-32"><a href="#cb116-32" aria-hidden="true" tabindex="-1"></a>                                   <span class="at">size =</span> <span class="dv">14</span>),</span>
<span id="cb116-33"><a href="#cb116-33" aria-hidden="true" tabindex="-1"></a>        <span class="at">axis.title =</span> <span class="fu">element_text</span>(<span class="at">face=</span><span class="st">"bold"</span>),</span>
<span id="cb116-34"><a href="#cb116-34" aria-hidden="true" tabindex="-1"></a>        <span class="at">plot.caption =</span> <span class="fu">element_text</span>(<span class="at">face =</span> <span class="st">"italic"</span>),</span>
<span id="cb116-35"><a href="#cb116-35" aria-hidden="true" tabindex="-1"></a>        <span class="at">legend.text =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="dv">14</span>)) <span class="sc">+</span></span>
<span id="cb116-36"><a href="#cb116-36" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_text_repel</span>(<span class="at">size =</span> <span class="dv">2</span>, </span>
<span id="cb116-37"><a href="#cb116-37" aria-hidden="true" tabindex="-1"></a>                  <span class="at">max.overlaps =</span> <span class="dv">30</span>) </span>
<span id="cb116-38"><a href="#cb116-38" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb116-39"><a href="#cb116-39" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb116-40"><a href="#cb116-40" aria-hidden="true" tabindex="-1"></a><span class="fu">ggsave</span>(<span class="st">"../results/volcano_plot.png"</span>,<span class="at">plot =</span> volcano_plot)  </span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Saving 7 x 5 in image</code></pre>
</div>
</div>
<p>Consider if we want to rename the final html file to something else??</p>
</section>
</section>

</main>
<!-- /main column -->
<script id="quarto-html-after-body" type="application/javascript">
window.document.addEventListener("DOMContentLoaded", function (event) {
  const toggleBodyColorMode = (bsSheetEl) => {
    const mode = bsSheetEl.getAttribute("data-mode");
    const bodyEl = window.document.querySelector("body");
    if (mode === "dark") {
      bodyEl.classList.add("quarto-dark");
      bodyEl.classList.remove("quarto-light");
    } else {
      bodyEl.classList.add("quarto-light");
      bodyEl.classList.remove("quarto-dark");
    }
  }
  const toggleBodyColorPrimary = () => {
    const bsSheetEl = window.document.querySelector("link#quarto-bootstrap");
    if (bsSheetEl) {
      toggleBodyColorMode(bsSheetEl);
    }
  }
  toggleBodyColorPrimary();  
  const icon = "";
  const anchorJS = new window.AnchorJS();
  anchorJS.options = {
    placement: 'right',
    icon: icon
  };
  anchorJS.add('.anchored');
  const isCodeAnnotation = (el) => {
    for (const clz of el.classList) {
      if (clz.startsWith('code-annotation-')) {                     
        return true;
      }
    }
    return false;
  }
  const clipboard = new window.ClipboardJS('.code-copy-button', {
    text: function(trigger) {
      const codeEl = trigger.previousElementSibling.cloneNode(true);
      for (const childEl of codeEl.children) {
        if (isCodeAnnotation(childEl)) {
          childEl.remove();
        }
      }
      return codeEl.innerText;
    }
  });
  clipboard.on('success', function(e) {
    // button target
    const button = e.trigger;
    // don't keep focus
    button.blur();
    // flash "checked"
    button.classList.add('code-copy-button-checked');
    var currentTitle = button.getAttribute("title");
    button.setAttribute("title", "Copied!");
    let tooltip;
    if (window.bootstrap) {
      button.setAttribute("data-bs-toggle", "tooltip");
      button.setAttribute("data-bs-placement", "left");
      button.setAttribute("data-bs-title", "Copied!");
      tooltip = new bootstrap.Tooltip(button, 
        { trigger: "manual", 
          customClass: "code-copy-button-tooltip",
          offset: [0, -8]});
      tooltip.show();    
    }
    setTimeout(function() {
      if (tooltip) {
        tooltip.hide();
        button.removeAttribute("data-bs-title");
        button.removeAttribute("data-bs-toggle");
        button.removeAttribute("data-bs-placement");
      }
      button.setAttribute("title", currentTitle);
      button.classList.remove('code-copy-button-checked');
    }, 1000);
    // clear code selection
    e.clearSelection();
  });
  function tippyHover(el, contentFn) {
    const config = {
      allowHTML: true,
      content: contentFn,
      maxWidth: 500,
      delay: 100,
      arrow: false,
      appendTo: function(el) {
          return el.parentElement;
      },
      interactive: true,
      interactiveBorder: 10,
      theme: 'quarto',
      placement: 'bottom-start'
    };
    window.tippy(el, config); 
  }
  const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
  for (var i=0; i<noterefs.length; i++) {
    const ref = noterefs[i];
    tippyHover(ref, function() {
      // use id or data attribute instead here
      let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
      try { href = new URL(href).hash; } catch {}
      const id = href.replace(/^#\/?/, "");
      const note = window.document.getElementById(id);
      return note.innerHTML;
    });
  }
      let selectedAnnoteEl;
      const selectorForAnnotation = ( cell, annotation) => {
        let cellAttr = 'data-code-cell="' + cell + '"';
        let lineAttr = 'data-code-annotation="' +  annotation + '"';
        const selector = 'span[' + cellAttr + '][' + lineAttr + ']';
        return selector;
      }
      const selectCodeLines = (annoteEl) => {
        const doc = window.document;
        const targetCell = annoteEl.getAttribute("data-target-cell");
        const targetAnnotation = annoteEl.getAttribute("data-target-annotation");
        const annoteSpan = window.document.querySelector(selectorForAnnotation(targetCell, targetAnnotation));
        const lines = annoteSpan.getAttribute("data-code-lines").split(",");
        const lineIds = lines.map((line) => {
          return targetCell + "-" + line;
        })
        let top = null;
        let height = null;
        let parent = null;
        if (lineIds.length > 0) {
            //compute the position of the single el (top and bottom and make a div)
            const el = window.document.getElementById(lineIds[0]);
            top = el.offsetTop;
            height = el.offsetHeight;
            parent = el.parentElement.parentElement;
          if (lineIds.length > 1) {
            const lastEl = window.document.getElementById(lineIds[lineIds.length - 1]);
            const bottom = lastEl.offsetTop + lastEl.offsetHeight;
            height = bottom - top;
          }
          if (top !== null && height !== null && parent !== null) {
            // cook up a div (if necessary) and position it 
            let div = window.document.getElementById("code-annotation-line-highlight");
            if (div === null) {
              div = window.document.createElement("div");
              div.setAttribute("id", "code-annotation-line-highlight");
              div.style.position = 'absolute';
              parent.appendChild(div);
            }
            div.style.top = top - 2 + "px";
            div.style.height = height + 4 + "px";
            let gutterDiv = window.document.getElementById("code-annotation-line-highlight-gutter");
            if (gutterDiv === null) {
              gutterDiv = window.document.createElement("div");
              gutterDiv.setAttribute("id", "code-annotation-line-highlight-gutter");
              gutterDiv.style.position = 'absolute';
              const codeCell = window.document.getElementById(targetCell);
              const gutter = codeCell.querySelector('.code-annotation-gutter');
              gutter.appendChild(gutterDiv);
            }
            gutterDiv.style.top = top - 2 + "px";
            gutterDiv.style.height = height + 4 + "px";
          }
          selectedAnnoteEl = annoteEl;
        }
      };
      const unselectCodeLines = () => {
        const elementsIds = ["code-annotation-line-highlight", "code-annotation-line-highlight-gutter"];
        elementsIds.forEach((elId) => {
          const div = window.document.getElementById(elId);
          if (div) {
            div.remove();
          }
        });
        selectedAnnoteEl = undefined;
      };
      // Attach click handler to the DT
      const annoteDls = window.document.querySelectorAll('dt[data-target-cell]');
      for (const annoteDlNode of annoteDls) {
        annoteDlNode.addEventListener('click', (event) => {
          const clickedEl = event.target;
          if (clickedEl !== selectedAnnoteEl) {
            unselectCodeLines();
            const activeEl = window.document.querySelector('dt[data-target-cell].code-annotation-active');
            if (activeEl) {
              activeEl.classList.remove('code-annotation-active');
            }
            selectCodeLines(clickedEl);
            clickedEl.classList.add('code-annotation-active');
          } else {
            // Unselect the line
            unselectCodeLines();
            clickedEl.classList.remove('code-annotation-active');
          }
        });
      }
  const findCites = (el) => {
    const parentEl = el.parentElement;
    if (parentEl) {
      const cites = parentEl.dataset.cites;
      if (cites) {
        return {
          el,
          cites: cites.split(' ')
        };
      } else {
        return findCites(el.parentElement)
      }
    } else {
      return undefined;
    }
  };
  var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
  for (var i=0; i<bibliorefs.length; i++) {
    const ref = bibliorefs[i];
    const citeInfo = findCites(ref);
    if (citeInfo) {
      tippyHover(citeInfo.el, function() {
        var popup = window.document.createElement('div');
        citeInfo.cites.forEach(function(cite) {
          var citeDiv = window.document.createElement('div');
          citeDiv.classList.add('hanging-indent');
          citeDiv.classList.add('csl-entry');
          var biblioDiv = window.document.getElementById('ref-' + cite);
          if (biblioDiv) {
            citeDiv.innerHTML = biblioDiv.innerHTML;
          }
          popup.appendChild(citeDiv);
        });
        return popup.innerHTML;
      });
    }
  }
});
</script>
</div> <!-- /content -->



</body></html>