<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en"><head>

<meta charset="utf-8">
<meta name="generator" content="quarto-1.3.433">

<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">


<title>Data Clean</title>
<style>
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
div.columns{display: flex; gap: min(4vw, 1.5em);}
div.column{flex: auto; overflow-x: auto;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
ul.task-list li input[type="checkbox"] {
  width: 0.8em;
  margin: 0 0.8em 0.2em -1em; /* quarto-specific, see https://github.com/quarto-dev/quarto-cli/issues/4556 */ 
  vertical-align: middle;
}
/* CSS for syntax highlighting */
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
  }
pre.numberSource { margin-left: 3em;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
</style>


<script src="00_all_new_files/libs/clipboard/clipboard.min.js"></script>
<script src="00_all_new_files/libs/quarto-html/quarto.js"></script>
<script src="00_all_new_files/libs/quarto-html/popper.min.js"></script>
<script src="00_all_new_files/libs/quarto-html/tippy.umd.min.js"></script>
<script src="00_all_new_files/libs/quarto-html/anchor.min.js"></script>
<link href="00_all_new_files/libs/quarto-html/tippy.css" rel="stylesheet">
<link href="00_all_new_files/libs/quarto-html/quarto-syntax-highlighting.css" rel="stylesheet" id="quarto-text-highlighting-styles">
<script src="00_all_new_files/libs/bootstrap/bootstrap.min.js"></script>
<link href="00_all_new_files/libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="00_all_new_files/libs/bootstrap/bootstrap.min.css" rel="stylesheet" id="quarto-bootstrap" data-mode="light">


</head>

<body class="fullcontent">

<div id="quarto-content" class="page-columns page-rows-contents page-layout-article">

<main class="content" id="quarto-document-content">

<header id="title-block-header" class="quarto-title-block default">
<div class="quarto-title">
<h1 class="title">Data Clean</h1>
</div>



<div class="quarto-title-meta">

    
  
    
  </div>
  

</header>

<section id="data-origin" class="level2">
<h2 class="anchored" data-anchor-id="data-origin">Data Origin</h2>
<p>The gene expression data used in this project is available in the Gene Expression Omnibus (GEO) under the accession number GSE50834. The data was used for the following paper:</p>
<p>Dawany N, Showe LC, Kossenkov AV, Chang C et al.&nbsp;Identification of a 251 gene expression signature that can accurately detect M. tuberculosis in patients with and without HIV co-infection. PLoS One 2014;9(2):e89925. PMID: 24587128</p>
</section>
<section id="summary" class="level2">
<h2 class="anchored" data-anchor-id="summary">Summary</h2>
<p>The data comes from total RNA isolated from peripheral blood mononuclear cell (PBMC) from a South African cohort. Microarry analysis was performed to compare gene expression in patients either infected with HIV or co-infected with HIV/TB. Diagnosis of TB, especially in the presence of an HIV co-infection, can be challenging when using conventional diagnostic methods. Therefore, the data was used to identify a TB-specific gene signature.</p>
</section>
<section id="setup" class="level2">
<h2 class="anchored" data-anchor-id="setup">Setup</h2>
<p>Loading required packages.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb1"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb1-1"><a href="#cb1-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(<span class="st">"tidyverse"</span>)</span>
<span id="cb1-2"><a href="#cb1-2" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(<span class="st">"stringr"</span>)</span>
<span id="cb1-3"><a href="#cb1-3" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(<span class="st">"GEOquery"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>Note: Instruction of how to install the package GEOquery can be found here <em>https://bioconductor.org/packages/release/bioc/html/GEOquery.html</em></p>
</section>
<section id="load-data" class="level2">
<h2 class="anchored" data-anchor-id="load-data">Load Data</h2>
<p>This document will load the data from Gene Expression Omnibus. Furthermore, the file will export the data to a tsv-file. In the following, the data directory is defined along with the location of the original data.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb2"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb2-1"><a href="#cb2-1" aria-hidden="true" tabindex="-1"></a>dest_dir <span class="ot">&lt;-</span> <span class="st">"../data/"</span></span>
<span id="cb2-2"><a href="#cb2-2" aria-hidden="true" tabindex="-1"></a>raw_dir <span class="ot">&lt;-</span> <span class="fu">str_c</span>(dest_dir, <span class="st">"_raw/"</span>)</span>
<span id="cb2-3"><a href="#cb2-3" aria-hidden="true" tabindex="-1"></a>expr_file <span class="ot">&lt;-</span> <span class="st">"GSE50834_non-normalized.txt.gz"</span></span>
<span id="cb2-4"><a href="#cb2-4" aria-hidden="true" tabindex="-1"></a>meta_file <span class="ot">&lt;-</span> <span class="st">"GSE50834_series_matrix.txt.gz"</span></span>
<span id="cb2-5"><a href="#cb2-5" aria-hidden="true" tabindex="-1"></a>geo <span class="ot">&lt;-</span> <span class="st">"GSE50834"</span></span>
<span id="cb2-6"><a href="#cb2-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-7"><a href="#cb2-7" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span>( <span class="sc">!</span><span class="fu">dir.exists</span>(raw_dir) ){</span>
<span id="cb2-8"><a href="#cb2-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">dir.create</span>(<span class="at">path =</span> raw_dir,</span>
<span id="cb2-9"><a href="#cb2-9" aria-hidden="true" tabindex="-1"></a>             <span class="at">recursive =</span> <span class="cn">TRUE</span>)</span>
<span id="cb2-10"><a href="#cb2-10" aria-hidden="true" tabindex="-1"></a>}</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>The gene expression data, containing patient-ID’s and gene expression, is downloaded and stored in raw-data directory using the GEOquery library. GEOquery can download datas ets from the NCBI Gene Expression Omnibus (GEO), using a unique accession number.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb3"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb3-1"><a href="#cb3-1" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span>( <span class="sc">!</span><span class="fu">file.exists</span>(<span class="fu">str_c</span>(raw_dir, expr_file)) ){</span>
<span id="cb3-2"><a href="#cb3-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">getGEOSuppFiles</span>(<span class="at">GEO =</span> geo,</span>
<span id="cb3-3"><a href="#cb3-3" aria-hidden="true" tabindex="-1"></a>                  <span class="at">makeDirectory =</span> <span class="cn">FALSE</span>,</span>
<span id="cb3-4"><a href="#cb3-4" aria-hidden="true" tabindex="-1"></a>                  <span class="at">baseDir =</span> raw_dir,</span>
<span id="cb3-5"><a href="#cb3-5" aria-hidden="true" tabindex="-1"></a>                  <span class="at">fetch_files =</span> <span class="cn">TRUE</span>,</span>
<span id="cb3-6"><a href="#cb3-6" aria-hidden="true" tabindex="-1"></a>                  <span class="at">filter_regex =</span> <span class="st">"txt"</span>)</span>
<span id="cb3-7"><a href="#cb3-7" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb3-8"><a href="#cb3-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-9"><a href="#cb3-9" aria-hidden="true" tabindex="-1"></a>dat_expr <span class="ot">&lt;-</span> <span class="fu">read.table</span>(<span class="at">file =</span> <span class="fu">str_c</span>(raw_dir, expr_file),</span>
<span id="cb3-10"><a href="#cb3-10" aria-hidden="true" tabindex="-1"></a>                         <span class="at">sep =</span> <span class="st">"</span><span class="sc">\t</span><span class="st">"</span>,</span>
<span id="cb3-11"><a href="#cb3-11" aria-hidden="true" tabindex="-1"></a>                         <span class="at">header =</span> <span class="cn">TRUE</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>The meta data, containing patient-ID’s patient information, is downloaded and stored in raw-data directory.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb4"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb4-1"><a href="#cb4-1" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span> (<span class="sc">!</span><span class="fu">file.exists</span>(<span class="fu">str_c</span>(raw_dir, meta_file))){</span>
<span id="cb4-2"><a href="#cb4-2" aria-hidden="true" tabindex="-1"></a>  dat_meta <span class="ot">&lt;-</span> <span class="fu">getGEO</span>(<span class="at">GEO =</span> geo)</span>
<span id="cb4-3"><a href="#cb4-3" aria-hidden="true" tabindex="-1"></a>  dat_meta <span class="ot">&lt;-</span> <span class="fu">pData</span>(<span class="fu">phenoData</span>(dat_meta[[<span class="dv">1</span>]]))</span>
<span id="cb4-4"><a href="#cb4-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">write.table</span>(dat_meta, <span class="at">file =</span> <span class="fu">str_c</span>(raw_dir, meta_file), </span>
<span id="cb4-5"><a href="#cb4-5" aria-hidden="true" tabindex="-1"></a>             <span class="at">sep =</span> <span class="st">"</span><span class="sc">\t</span><span class="st">"</span>)</span>
<span id="cb4-6"><a href="#cb4-6" aria-hidden="true" tabindex="-1"></a>  dat_meta <span class="ot">&lt;-</span> <span class="fu">read.table</span>(<span class="fu">str_c</span>(raw_dir, meta_file), </span>
<span id="cb4-7"><a href="#cb4-7" aria-hidden="true" tabindex="-1"></a>                        <span class="at">sep =</span> <span class="st">"</span><span class="sc">\t</span><span class="st">"</span>,</span>
<span id="cb4-8"><a href="#cb4-8" aria-hidden="true" tabindex="-1"></a>                        <span class="at">header =</span> <span class="cn">TRUE</span>)</span>
<span id="cb4-9"><a href="#cb4-9" aria-hidden="true" tabindex="-1"></a>} <span class="cf">else</span> {</span>
<span id="cb4-10"><a href="#cb4-10" aria-hidden="true" tabindex="-1"></a>  dat_meta <span class="ot">&lt;-</span> <span class="fu">read.table</span>(<span class="fu">str_c</span>(raw_dir, meta_file), </span>
<span id="cb4-11"><a href="#cb4-11" aria-hidden="true" tabindex="-1"></a>                          <span class="at">sep =</span> <span class="st">"</span><span class="sc">\t</span><span class="st">"</span>,</span>
<span id="cb4-12"><a href="#cb4-12" aria-hidden="true" tabindex="-1"></a>                          <span class="at">header =</span> <span class="cn">TRUE</span>)</span>
<span id="cb4-13"><a href="#cb4-13" aria-hidden="true" tabindex="-1"></a>}</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="write-data-to-files" class="level2">
<h2 class="anchored" data-anchor-id="write-data-to-files">Write Data to Files</h2>
<p>The gene expression data and the meta data are saved as .tsv-files</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb5"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb5-1"><a href="#cb5-1" aria-hidden="true" tabindex="-1"></a>dat_expr <span class="sc">|&gt;</span> </span>
<span id="cb5-2"><a href="#cb5-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">write_tsv</span>(<span class="at">file =</span> <span class="fu">str_c</span>(dest_dir, <span class="st">"01_dat_load_expr.tsv"</span>))</span>
<span id="cb5-3"><a href="#cb5-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-4"><a href="#cb5-4" aria-hidden="true" tabindex="-1"></a>dat_meta <span class="sc">|&gt;</span> </span>
<span id="cb5-5"><a href="#cb5-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">write_tsv</span>(<span class="at">file =</span> <span class="fu">str_c</span>(dest_dir, <span class="st">"01_dat_load_meta.tsv"</span>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="setup-1" class="level2">
<h2 class="anchored" data-anchor-id="setup-1">Setup</h2>
<p>Load required packages.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb6"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb6-1"><a href="#cb6-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(<span class="st">"tidyverse"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>Load gene expression and meta data.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb7"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb7-1"><a href="#cb7-1" aria-hidden="true" tabindex="-1"></a>dat_expr <span class="ot">&lt;-</span> <span class="fu">read_tsv</span>(<span class="st">"../data/01_dat_load_expr.tsv"</span>)</span>
<span id="cb7-2"><a href="#cb7-2" aria-hidden="true" tabindex="-1"></a>dat_meta <span class="ot">&lt;-</span> <span class="fu">read_tsv</span>(<span class="st">"../data/01_dat_load_meta.tsv"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="data-wrangling-i" class="level2">
<h2 class="anchored" data-anchor-id="data-wrangling-i">Data Wrangling I</h2>
<p>In this section, variable of gene expression data and meta data are variables are cleaned seperately.</p>
<section id="gene-expression-data" class="level3">
<h3 class="anchored" data-anchor-id="gene-expression-data">Gene Expression Data</h3>
<p>In the following the gene expression data is inspected, cleaned and simplified.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb8"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb8-1"><a href="#cb8-1" aria-hidden="true" tabindex="-1"></a>dat_expr <span class="sc">|&gt;</span> </span>
<span id="cb8-2"><a href="#cb8-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">sample_n</span>(<span class="dv">10</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code># A tibble: 10 × 89
   ID_REF       AAA0VKQN.12_18.Mar.2009 Detection.Pval AAA0W22R.11_16.Apr.2009
   &lt;chr&gt;                          &lt;dbl&gt;          &lt;dbl&gt;                   &lt;dbl&gt;
 1 ILMN_1658034                    231.         0.664                    140. 
 2 ILMN_1808217                    203.         0.718                     87.1
 3 ILMN_1680330                    149.         0.881                     75.0
 4 ILMN_1712184                    133.         0.968                     89.7
 5 ILMN_1727984                    310.         0.155                    133. 
 6 ILMN_1750664                    284.         0.348                    144. 
 7 ILMN_1687857                    366.         0.0130                   247. 
 8 ILMN_3200018                    279.         0.386                    128. 
 9 ILMN_1690122                   4884.         0                       2567. 
10 ILMN_1651936                    712.         0                        283. 
# ℹ 85 more variables: Detection.Pval.1 &lt;dbl&gt;, AAA0W22R.14_16.Apr.2009 &lt;dbl&gt;,
#   Detection.Pval.2 &lt;dbl&gt;, BAA0V9SR.15_25.Feb.2009 &lt;dbl&gt;,
#   Detection.Pval.3 &lt;dbl&gt;, BAA0W3BM.11_20.Apr.2009 &lt;dbl&gt;,
#   Detection.Pval.4 &lt;dbl&gt;, BAA0W69B.10_29.Apr.2009 &lt;dbl&gt;,
#   Detection.Pval.5 &lt;dbl&gt;, BAA0WVPG.10_04.Jun.2009 &lt;dbl&gt;,
#   Detection.Pval.6 &lt;dbl&gt;, CAA0Q3CK.13_24.Apr.2008 &lt;dbl&gt;,
#   Detection.Pval.7 &lt;dbl&gt;, CAA0W4LJ.15_23.Apr.2009 &lt;dbl&gt;, …</code></pre>
</div>
</div>
<p>From the inspection, it is seen that the data contain both gene expressions and p values. The detected p values are removed such that the gene expression data only consists of the gene expressions.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb10"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb10-1"><a href="#cb10-1" aria-hidden="true" tabindex="-1"></a>dat_expr <span class="ot">&lt;-</span> dat_expr <span class="sc">|&gt;</span> </span>
<span id="cb10-2"><a href="#cb10-2" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">select</span>(<span class="sc">-</span><span class="fu">matches</span>(<span class="at">match =</span> <span class="st">"Detection.Pval"</span>))</span>
<span id="cb10-3"><a href="#cb10-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-4"><a href="#cb10-4" aria-hidden="true" tabindex="-1"></a>dat_expr <span class="sc">|&gt;</span> </span>
<span id="cb10-5"><a href="#cb10-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">sample_n</span>(<span class="dv">10</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code># A tibble: 10 × 45
   ID_REF   AAA0VKQN.12_18.Mar.2…¹ AAA0W22R.11_16.Apr.2…² AAA0W22R.14_16.Apr.2…³
   &lt;chr&gt;                     &lt;dbl&gt;                  &lt;dbl&gt;                  &lt;dbl&gt;
 1 ILMN_17…                   343.                  173.                    279.
 2 ILMN_18…                   342.                  177.                    297.
 3 ILMN_17…                   152.                  155.                    246.
 4 ILMN_17…                   131.                   69.0                   119.
 5 ILMN_32…                   264.                  132.                    213.
 6 ILMN_22…                   398.                  206.                    340.
 7 ILMN_17…                   188.                  110.                    158.
 8 ILMN_21…                   320.                  191.                    270.
 9 ILMN_17…                  2632.                 1084.                   1873.
10 ILMN_16…                   391.                  148.                    313.
# ℹ abbreviated names: ¹​AAA0VKQN.12_18.Mar.2009, ²​AAA0W22R.11_16.Apr.2009,
#   ³​AAA0W22R.14_16.Apr.2009
# ℹ 41 more variables: BAA0V9SR.15_25.Feb.2009 &lt;dbl&gt;,
#   BAA0W3BM.11_20.Apr.2009 &lt;dbl&gt;, BAA0W69B.10_29.Apr.2009 &lt;dbl&gt;,
#   BAA0WVPG.10_04.Jun.2009 &lt;dbl&gt;, CAA0Q3CK.13_24.Apr.2008 &lt;dbl&gt;,
#   CAA0W4LJ.15_23.Apr.2009 &lt;dbl&gt;, CAA0W4LT.13_23.Apr.2009 &lt;dbl&gt;,
#   DAA0PFCT.11_17.Jan.2008 &lt;dbl&gt;, DAA0V9RP.16_25.Feb.2009 &lt;dbl&gt;, …</code></pre>
</div>
</div>
<p>Column names are simplified as the date is saved in the meta data.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb12"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb12-1"><a href="#cb12-1" aria-hidden="true" tabindex="-1"></a>dat_expr <span class="sc">|&gt;</span> </span>
<span id="cb12-2"><a href="#cb12-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">colnames</span>() <span class="ot">=</span> <span class="fu">str_extract</span>(<span class="at">string =</span> <span class="fu">colnames</span>(dat_expr),</span>
<span id="cb12-3"><a href="#cb12-3" aria-hidden="true" tabindex="-1"></a>                           <span class="at">pattern =</span> <span class="st">"^[A-Z0-9]+</span><span class="sc">\\</span><span class="st">.?[0-9]*"</span>)</span>
<span id="cb12-4"><a href="#cb12-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb12-5"><a href="#cb12-5" aria-hidden="true" tabindex="-1"></a>dat_expr <span class="sc">|&gt;</span> </span>
<span id="cb12-6"><a href="#cb12-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">sample_n</span>(<span class="dv">10</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code># A tibble: 10 × 45
   ID    AAA0VKQN.12 AAA0W22R.11 AAA0W22R.14 BAA0V9SR.15 BAA0W3BM.11 BAA0W69B.10
   &lt;chr&gt;       &lt;dbl&gt;       &lt;dbl&gt;       &lt;dbl&gt;       &lt;dbl&gt;       &lt;dbl&gt;       &lt;dbl&gt;
 1 ILMN…        231.       127.         208.        292.        231.       108. 
 2 ILMN…        281.       151.         254.        283.        258.       115. 
 3 ILMN…        304.       161.         311.        277.        309.       152. 
 4 ILMN…        308.       170.         264.        230.        304.       135. 
 5 ILMN…        584.       316.         646.        936.        496.       485. 
 6 ILMN…        178.        86.1        170.        206.        176.        94.8
 7 ILMN…        332.       169.         301.        332.        355.       140. 
 8 ILMN…        368.       177.         313.        327.        294.       129. 
 9 ILMN…        317.       155.         292.        281.        279.       165. 
10 ILMN…        242.       126.         278.        264.        287.       133. 
# ℹ 38 more variables: BAA0WVPG.10 &lt;dbl&gt;, CAA0Q3CK.13 &lt;dbl&gt;, CAA0W4LJ.15 &lt;dbl&gt;,
#   CAA0W4LT.13 &lt;dbl&gt;, DAA0PFCT.11 &lt;dbl&gt;, DAA0V9RP.16 &lt;dbl&gt;, DAA0VD27.11 &lt;dbl&gt;,
#   FAA0W22D.13 &lt;dbl&gt;, HAA0W3B8.14 &lt;dbl&gt;, HAA0W3R9.13 &lt;dbl&gt;, JAA0P0DQ.11 &lt;dbl&gt;,
#   JAA0W223.10 &lt;dbl&gt;, KAA0PBTZ.17 &lt;dbl&gt;, KAA0Q3QH.14 &lt;dbl&gt;, AAA0WVKX.11 &lt;dbl&gt;,
#   EAA0VKQY.17 &lt;dbl&gt;, JAA0WVMP.13 &lt;dbl&gt;, AAA0KHPN.10 &lt;dbl&gt;, AAA0RWJ0.14 &lt;dbl&gt;,
#   AAA0Z4D1.15 &lt;dbl&gt;, AAA0ZSJX.10 &lt;dbl&gt;, BAA0YXVB.12 &lt;dbl&gt;, BAA0ZC4J.13 &lt;dbl&gt;,
#   BAA0ZXTF.11 &lt;dbl&gt;, CAA0ZN6L.12 &lt;dbl&gt;, DAA0Z4CR.10 &lt;dbl&gt;, …</code></pre>
</div>
</div>
</section>
<section id="meta-data" class="level3">
<h3 class="anchored" data-anchor-id="meta-data">Meta Data</h3>
<p>In the following the meta data is inspected, relevant columns are extracted, renamed and simplified</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb14"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb14-1"><a href="#cb14-1" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(dat_meta)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code> [1] "title"                   "geo_accession"          
 [3] "status"                  "submission_date"        
 [5] "last_update_date"        "type"                   
 [7] "channel_count"           "source_name_ch1"        
 [9] "organism_ch1"            "characteristics_ch1"    
[11] "characteristics_ch1.1"   "characteristics_ch1.2"  
[13] "molecule_ch1"            "extract_protocol_ch1"   
[15] "label_ch1"               "label_protocol_ch1"     
[17] "taxid_ch1"               "hyb_protocol"           
[19] "scan_protocol"           "description"            
[21] "description.1"           "data_processing"        
[23] "platform_id"             "contact_name"           
[25] "contact_email"           "contact_phone"          
[27] "contact_laboratory"      "contact_institute"      
[29] "contact_address"         "contact_city"           
[31] "contact_state"           "contact_zip.postal_code"
[33] "contact_country"         "supplementary_file"     
[35] "data_row_count"          "disease.state.ch1"      
[37] "gender.ch1"              "sample.type.ch1"        </code></pre>
</div>
</div>
<p>We are interested in keeping ‘title’, ‘geo_accession’, and ‘description.1’. The last mentioned column is the identifier refering to each sample found in the gene expression data. Furthermore, we want to extract information about gender and disease state described by ‘characteristics_ch1’ and ‘characteristics_ch1.2’. This information is also found in the variables ‘gender.ch1’ and ‘disease.state.ch1’ in a nicer format. We chose to work with the first mentioned columns for educational purposes.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb16"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb16-1"><a href="#cb16-1" aria-hidden="true" tabindex="-1"></a>dat_meta <span class="ot">&lt;-</span> dat_meta <span class="sc">|&gt;</span></span>
<span id="cb16-2"><a href="#cb16-2" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">select</span>(title,</span>
<span id="cb16-3"><a href="#cb16-3" aria-hidden="true" tabindex="-1"></a>                geo_accession,</span>
<span id="cb16-4"><a href="#cb16-4" aria-hidden="true" tabindex="-1"></a>                characteristics_ch1,</span>
<span id="cb16-5"><a href="#cb16-5" aria-hidden="true" tabindex="-1"></a>                characteristics_ch1<span class="fl">.2</span>,</span>
<span id="cb16-6"><a href="#cb16-6" aria-hidden="true" tabindex="-1"></a>                description<span class="fl">.1</span>) <span class="sc">|&gt;</span> </span>
<span id="cb16-7"><a href="#cb16-7" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">rename</span>(<span class="st">"patient_n"</span> <span class="ot">=</span> title,</span>
<span id="cb16-8"><a href="#cb16-8" aria-hidden="true" tabindex="-1"></a>                <span class="st">"gender"</span> <span class="ot">=</span> characteristics_ch1,</span>
<span id="cb16-9"><a href="#cb16-9" aria-hidden="true" tabindex="-1"></a>                <span class="st">"disease_state"</span> <span class="ot">=</span> characteristics_ch1<span class="fl">.2</span>,</span>
<span id="cb16-10"><a href="#cb16-10" aria-hidden="true" tabindex="-1"></a>                <span class="st">"identifier"</span> <span class="ot">=</span> description<span class="fl">.1</span>)</span>
<span id="cb16-11"><a href="#cb16-11" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb16-12"><a href="#cb16-12" aria-hidden="true" tabindex="-1"></a>dat_meta <span class="sc">|&gt;</span> </span>
<span id="cb16-13"><a href="#cb16-13" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(patient_n, gender, disease_state) <span class="sc">|&gt;</span></span>
<span id="cb16-14"><a href="#cb16-14" aria-hidden="true" tabindex="-1"></a>  <span class="fu">sample_n</span>(<span class="dv">10</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code># A tibble: 10 × 3
   patient_n                     gender         disease_state          
   &lt;chr&gt;                         &lt;chr&gt;          &lt;chr&gt;                  
 1 HIV/TB co-infected patient 18 gender: Female disease state: HIV/TB  
 2 HIV infected patient 16       gender: Female disease state: HIV only
 3 HIV infected patient 22       gender: Female disease state: HIV only
 4 HIV/TB co-infected patient 20 gender: Male   disease state: HIV/TB  
 5 HIV infected patient 20       gender: Female disease state: HIV only
 6 HIV infected patient 4        gender: Female disease state: HIV only
 7 HIV infected patient 5        gender: Female disease state: HIV only
 8 HIV/TB co-infected patient 10 gender: Male   disease state: HIV/TB  
 9 HIV infected patient 2_2      gender: Female disease state: HIV only
10 HIV infected patient 9        gender: Female disease state: HIV only</code></pre>
</div>
</div>
<p>Simplify the columns ‘patient’, ‘gender’ and ‘disease state’.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb18"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb18-1"><a href="#cb18-1" aria-hidden="true" tabindex="-1"></a>dat_meta <span class="ot">&lt;-</span> dat_meta <span class="sc">|&gt;</span> </span>
<span id="cb18-2"><a href="#cb18-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">patient_n =</span> <span class="fu">str_extract</span>(<span class="at">string =</span> patient_n,</span>
<span id="cb18-3"><a href="#cb18-3" aria-hidden="true" tabindex="-1"></a>                                 <span class="at">pattern =</span> <span class="st">"[0-9]+_?[0-9]?"</span>),</span>
<span id="cb18-4"><a href="#cb18-4" aria-hidden="true" tabindex="-1"></a>         <span class="at">gender =</span> <span class="fu">str_extract</span>(<span class="at">string =</span> gender,</span>
<span id="cb18-5"><a href="#cb18-5" aria-hidden="true" tabindex="-1"></a>                              <span class="at">pattern =</span> <span class="st">"Female|Male"</span>),</span>
<span id="cb18-6"><a href="#cb18-6" aria-hidden="true" tabindex="-1"></a>         <span class="at">disease_state =</span> <span class="fu">str_extract</span>(<span class="at">string =</span> disease_state,</span>
<span id="cb18-7"><a href="#cb18-7" aria-hidden="true" tabindex="-1"></a>                              <span class="at">pattern =</span> <span class="st">"HIV/TB|HIV"</span>))</span>
<span id="cb18-8"><a href="#cb18-8" aria-hidden="true" tabindex="-1"></a>dat_meta <span class="sc">|&gt;</span> </span>
<span id="cb18-9"><a href="#cb18-9" aria-hidden="true" tabindex="-1"></a>  <span class="fu">sample_n</span>(<span class="dv">10</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code># A tibble: 10 × 5
   patient_n geo_accession gender disease_state identifier             
   &lt;chr&gt;     &lt;chr&gt;         &lt;chr&gt;  &lt;chr&gt;         &lt;chr&gt;                  
 1 3         GSM1230927    Male   HIV/TB        AAA0Z4D1-15_25-Nov-2009
 2 6         GSM1230930    Male   HIV/TB        BAA0ZC4J-13_07-Dec-2009
 3 8         GSM1230932    Male   HIV/TB        CAA0ZN6L-12_13-Jan-2010
 4 8         GSM1230910    Male   HIV           CAA0W4LJ-15_23-Apr-2009
 5 3         GSM1230905    Male   HIV           BAA0V9SR-15_25-Feb-2009
 6 18        GSM1230942    Female HIV/TB        KAA0Z0GV-10_18-Nov-2009
 7 2_1       GSM1230903    Female HIV           AAA0W22R-11_16-Apr-2009
 8 18        GSM1230920    Female HIV           KAA0PBTZ-17_06-Mar-2008
 9 21        GSM1230923    Male   HIV           EAA0VKQY-17_18-Mar-2009
10 11        GSM1230913    Female HIV           DAA0V9RP-16_25-Feb-2009</code></pre>
</div>
</div>
<p><strong>Identifier</strong></p>
<p>The column ‘identifier’ in the meta data contain the names of the genes described in the expression data. In the following, the ‘identifier’ is corrected to mimic the genes of the expression data.</p>
<p>The ‘identifier’ in the meta data is split into two columns to simplify the identifier but still keep the information about the dates.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb20"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb20-1"><a href="#cb20-1" aria-hidden="true" tabindex="-1"></a>dat_meta <span class="ot">&lt;-</span> dat_meta <span class="sc">|&gt;</span> </span>
<span id="cb20-2"><a href="#cb20-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">separate</span>(<span class="at">col =</span> identifier,</span>
<span id="cb20-3"><a href="#cb20-3" aria-hidden="true" tabindex="-1"></a>           <span class="at">into =</span> <span class="fu">c</span>(<span class="st">"identifier"</span>, <span class="st">"date"</span>),</span>
<span id="cb20-4"><a href="#cb20-4" aria-hidden="true" tabindex="-1"></a>           <span class="at">sep =</span> <span class="st">"_"</span>)</span>
<span id="cb20-5"><a href="#cb20-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb20-6"><a href="#cb20-6" aria-hidden="true" tabindex="-1"></a>dat_meta <span class="sc">|&gt;</span> </span>
<span id="cb20-7"><a href="#cb20-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">sample_n</span>(<span class="dv">10</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code># A tibble: 10 × 6
   patient_n geo_accession gender disease_state identifier  date       
   &lt;chr&gt;     &lt;chr&gt;         &lt;chr&gt;  &lt;chr&gt;         &lt;chr&gt;       &lt;chr&gt;      
 1 11        GSM1230935    Male   HIV/TB        FAA0ZVHT-14 02-Feb-2010
 2 2_2       GSM1230904    Female HIV           AAA0W22R-14 16-Apr-2009
 3 1         GSM1230925    Female HIV/TB        AAA0KHPN-10 14-Jun-2007
 4 2         GSM1230926    Female HIV/TB        AAA0RWJ0-14 02-Sep-2008
 5 17        GSM1230941    Female HIV/TB        JAA1063Y-11 03-Mar-2010
 6 21        GSM1230945    Male   HIV/TB        EAA0ZNH3-13 13-Jan-2010
 7 3         GSM1230927    Male   HIV/TB        AAA0Z4D1-15 25-Nov-2009
 8 20        GSM1230922    Female HIV           AAA0WVKX-11 04-Jun-2009
 9 4         GSM1230906    Female HIV           BAA0W3BM-11 20-Apr-2009
10 13        GSM1230915    Male   HIV           FAA0W22D-13 16-Apr-2009</code></pre>
</div>
</div>
<p>In the gene expression data the gene names contain “-” while the ‘identifier’ in the meta data contain “.”. Replace the “-” in the meta data to streamline the two data sets</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb22"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb22-1"><a href="#cb22-1" aria-hidden="true" tabindex="-1"></a>dat_meta <span class="ot">&lt;-</span> dat_meta <span class="sc">|&gt;</span> </span>
<span id="cb22-2"><a href="#cb22-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">identifier =</span> <span class="fu">str_replace_all</span>(<span class="at">string =</span> identifier,</span>
<span id="cb22-3"><a href="#cb22-3" aria-hidden="true" tabindex="-1"></a>                                      <span class="at">pattern =</span> <span class="st">"-"</span>,</span>
<span id="cb22-4"><a href="#cb22-4" aria-hidden="true" tabindex="-1"></a>                                      <span class="at">replacement =</span> <span class="st">"."</span>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>Check if there is consensus between the ‘identifiers’ in the meta data and the column names (gene names) of the gene expression data.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb23"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb23-1"><a href="#cb23-1" aria-hidden="true" tabindex="-1"></a>meta_identifier <span class="ot">&lt;-</span> dat_meta <span class="sc">|&gt;</span></span>
<span id="cb23-2"><a href="#cb23-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">pull</span>(identifier)</span>
<span id="cb23-3"><a href="#cb23-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb23-4"><a href="#cb23-4" aria-hidden="true" tabindex="-1"></a><span class="fu">setdiff</span>(<span class="fu">colnames</span>(dat_expr),meta_identifier)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] "ID"</code></pre>
</div>
</div>
<p>The only difference between ‘identifier’ in meta data and column names of the gene expression data is the ‘ID’ column for the genes as expected.</p>
</section>
</section>
<section id="data-wrangling-ii" class="level2">
<h2 class="anchored" data-anchor-id="data-wrangling-ii">Data Wrangling II</h2>
<p>In the following steps, the format of the data will be changed. Furthermore, we wish to join the gene expression data and the meta data.</p>
<section id="data-pivoting" class="level3">
<h3 class="anchored" data-anchor-id="data-pivoting">Data Pivoting</h3>
<p>The gene expression data is transposed by pivoting sample ‘identifier’ to long format and gene ‘ID’ to wide format.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb25"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb25-1"><a href="#cb25-1" aria-hidden="true" tabindex="-1"></a>dat_expr <span class="ot">&lt;-</span> dat_expr <span class="sc">|&gt;</span> </span>
<span id="cb25-2"><a href="#cb25-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">pivot_longer</span>(<span class="at">cols =</span> <span class="sc">!</span>ID,</span>
<span id="cb25-3"><a href="#cb25-3" aria-hidden="true" tabindex="-1"></a>               <span class="at">names_to =</span> <span class="st">"identifier"</span>,</span>
<span id="cb25-4"><a href="#cb25-4" aria-hidden="true" tabindex="-1"></a>               <span class="at">values_to =</span> <span class="st">"count"</span>) <span class="sc">|&gt;</span> </span>
<span id="cb25-5"><a href="#cb25-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">pivot_wider</span>(<span class="at">names_from =</span> ID,</span>
<span id="cb25-6"><a href="#cb25-6" aria-hidden="true" tabindex="-1"></a>              <span class="at">values_from =</span> count)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="data-joining" class="level3">
<h3 class="anchored" data-anchor-id="data-joining">Data Joining</h3>
<p>The meta data and the count data should be merge such that all data about a patient is contained in one tibble. Meta data should be in the first columns followed by gene variables.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb26"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb26-1"><a href="#cb26-1" aria-hidden="true" tabindex="-1"></a>dat_clean <span class="ot">&lt;-</span> <span class="fu">inner_join</span>(<span class="at">x =</span> dat_expr,</span>
<span id="cb26-2"><a href="#cb26-2" aria-hidden="true" tabindex="-1"></a>                        <span class="at">y =</span> dat_meta,</span>
<span id="cb26-3"><a href="#cb26-3" aria-hidden="true" tabindex="-1"></a>                        <span class="at">by =</span> <span class="st">"identifier"</span>) <span class="sc">|&gt;</span> </span>
<span id="cb26-4"><a href="#cb26-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">relocate</span>(<span class="fu">colnames</span>(dat_meta))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="handling-replicates" class="level3">
<h3 class="anchored" data-anchor-id="handling-replicates">Handling Replicates</h3>
<p>There are two technical replicates available for one patient, namely patient 2_1 and 2_2 indicated by the ‘patient_n’ column. We wish to averaged the measurements for this patient for further analysis.</p>
<p>First, the gene expression values for the replicates are extracted.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb27"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb27-1"><a href="#cb27-1" aria-hidden="true" tabindex="-1"></a>dat_rep <span class="ot">&lt;-</span> dat_clean <span class="sc">|&gt;</span></span>
<span id="cb27-2"><a href="#cb27-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(<span class="fu">str_detect</span>(patient_n, <span class="st">"_"</span>)) <span class="sc">|&gt;</span></span>
<span id="cb27-3"><a href="#cb27-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">patient_n =</span> <span class="fu">str_extract</span>(<span class="at">string =</span> patient_n, </span>
<span id="cb27-4"><a href="#cb27-4" aria-hidden="true" tabindex="-1"></a>                                 <span class="at">pattern =</span> <span class="st">"[0-9]+"</span>)) </span>
<span id="cb27-5"><a href="#cb27-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-6"><a href="#cb27-6" aria-hidden="true" tabindex="-1"></a>dat_rep </span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code># A tibble: 2 × 47,237
  patient_n geo_accession gender disease_state identifier  date     ILMN_1762337
  &lt;chr&gt;     &lt;chr&gt;         &lt;chr&gt;  &lt;chr&gt;         &lt;chr&gt;       &lt;chr&gt;           &lt;dbl&gt;
1 2         GSM1230903    Female HIV           AAA0W22R.11 16-Apr-…         158.
2 2         GSM1230904    Female HIV           AAA0W22R.14 16-Apr-…         300.
# ℹ 47,230 more variables: ILMN_2055271 &lt;dbl&gt;, ILMN_1736007 &lt;dbl&gt;,
#   ILMN_2383229 &lt;dbl&gt;, ILMN_1806310 &lt;dbl&gt;, ILMN_1779670 &lt;dbl&gt;,
#   ILMN_1653355 &lt;dbl&gt;, ILMN_1717783 &lt;dbl&gt;, ILMN_1705025 &lt;dbl&gt;,
#   ILMN_1814316 &lt;dbl&gt;, ILMN_2359168 &lt;dbl&gt;, ILMN_1731507 &lt;dbl&gt;,
#   ILMN_1787689 &lt;dbl&gt;, ILMN_3241953 &lt;dbl&gt;, ILMN_1745607 &lt;dbl&gt;,
#   ILMN_2136495 &lt;dbl&gt;, ILMN_1668111 &lt;dbl&gt;, ILMN_2295559 &lt;dbl&gt;,
#   ILMN_1735045 &lt;dbl&gt;, ILMN_1680754 &lt;dbl&gt;, ILMN_2375184 &lt;dbl&gt;, …</code></pre>
</div>
</div>
<p>The mean gene expression is calculated for each replicated patient.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb29"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb29-1"><a href="#cb29-1" aria-hidden="true" tabindex="-1"></a>dat_rep_mean <span class="ot">&lt;-</span> dat_rep <span class="sc">|&gt;</span> </span>
<span id="cb29-2"><a href="#cb29-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(patient_n) <span class="sc">|&gt;</span></span>
<span id="cb29-3"><a href="#cb29-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarise</span>(<span class="fu">across</span>(<span class="fu">starts_with</span>(<span class="st">"ILMN"</span>), </span>
<span id="cb29-4"><a href="#cb29-4" aria-hidden="true" tabindex="-1"></a>                   mean, <span class="at">na.rm =</span> <span class="cn">TRUE</span>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: There was 1 warning in `summarise()`.
ℹ In argument: `across(starts_with("ILMN"), mean, na.rm = TRUE)`.
ℹ In group 1: `patient_n = "2"`.
Caused by warning:
! The `...` argument of `across()` is deprecated as of dplyr 1.1.0.
Supply arguments directly to `.fns` through an anonymous function instead.

  # Previously
  across(a:b, mean, na.rm = TRUE)

  # Now
  across(a:b, \(x) mean(x, na.rm = TRUE))</code></pre>
</div>
<div class="sourceCode cell-code" id="cb31"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb31-1"><a href="#cb31-1" aria-hidden="true" tabindex="-1"></a>dat_rep_mean</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code># A tibble: 1 × 47,232
  patient_n ILMN_1762337 ILMN_2055271 ILMN_1736007 ILMN_2383229 ILMN_1806310
  &lt;chr&gt;            &lt;dbl&gt;        &lt;dbl&gt;        &lt;dbl&gt;        &lt;dbl&gt;        &lt;dbl&gt;
1 2                 229.         209.         200.         192.         189.
# ℹ 47,226 more variables: ILMN_1779670 &lt;dbl&gt;, ILMN_1653355 &lt;dbl&gt;,
#   ILMN_1717783 &lt;dbl&gt;, ILMN_1705025 &lt;dbl&gt;, ILMN_1814316 &lt;dbl&gt;,
#   ILMN_2359168 &lt;dbl&gt;, ILMN_1731507 &lt;dbl&gt;, ILMN_1787689 &lt;dbl&gt;,
#   ILMN_3241953 &lt;dbl&gt;, ILMN_1745607 &lt;dbl&gt;, ILMN_2136495 &lt;dbl&gt;,
#   ILMN_1668111 &lt;dbl&gt;, ILMN_2295559 &lt;dbl&gt;, ILMN_1735045 &lt;dbl&gt;,
#   ILMN_1680754 &lt;dbl&gt;, ILMN_2375184 &lt;dbl&gt;, ILMN_1659452 &lt;dbl&gt;,
#   ILMN_1767388 &lt;dbl&gt;, ILMN_1675204 &lt;dbl&gt;, ILMN_1673870 &lt;dbl&gt;, …</code></pre>
</div>
</div>
<p>The metadata for each replicated patient is extracted. The ‘geo accession’ and ‘identifer’ is collapsed, meaning that the new ‘geo accession’ and ‘identifer’ for the replicated patient is a combination of the originals.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb33"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb33-1"><a href="#cb33-1" aria-hidden="true" tabindex="-1"></a>cols_rep_meta_collapesed <span class="ot">&lt;-</span> dat_rep <span class="sc">|&gt;</span></span>
<span id="cb33-2"><a href="#cb33-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(patient_n) <span class="sc">|&gt;</span> </span>
<span id="cb33-3"><a href="#cb33-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarise</span>(<span class="at">geo_accession =</span> <span class="fu">str_c</span>(geo_accession, <span class="at">collapse =</span> <span class="st">"_"</span>),</span>
<span id="cb33-4"><a href="#cb33-4" aria-hidden="true" tabindex="-1"></a>            <span class="at">identifier =</span> <span class="fu">str_c</span>(identifier, <span class="at">collapse =</span> <span class="st">"_"</span>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>A new column called ‘modified’ is added to the meta data. The column describes if the patient has been modified (eg. the gene expression represent a mean of multiple gene expressions for one patient).</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb34"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb34-1"><a href="#cb34-1" aria-hidden="true" tabindex="-1"></a>cols_rep_meta <span class="ot">&lt;-</span> dat_rep <span class="sc">|&gt;</span></span>
<span id="cb34-2"><a href="#cb34-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(patient_n, gender, disease_state, date) <span class="sc">|&gt;</span> </span>
<span id="cb34-3"><a href="#cb34-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">distinct</span>() <span class="sc">|&gt;</span> </span>
<span id="cb34-4"><a href="#cb34-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">add_column</span>(cols_rep_meta_collapesed <span class="sc">|&gt;</span> <span class="fu">select</span>(geo_accession, identifier),</span>
<span id="cb34-5"><a href="#cb34-5" aria-hidden="true" tabindex="-1"></a>             <span class="at">modified =</span> <span class="dv">1</span>)</span>
<span id="cb34-6"><a href="#cb34-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb34-7"><a href="#cb34-7" aria-hidden="true" tabindex="-1"></a>cols_rep_meta</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code># A tibble: 1 × 7
  patient_n gender disease_state date        geo_accession   identifier modified
  &lt;chr&gt;     &lt;chr&gt;  &lt;chr&gt;         &lt;chr&gt;       &lt;chr&gt;           &lt;chr&gt;         &lt;dbl&gt;
1 2         Female HIV           16-Apr-2009 GSM1230903_GSM… AAA0W22R.…        1</code></pre>
</div>
<div class="sourceCode cell-code" id="cb36"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb36-1"><a href="#cb36-1" aria-hidden="true" tabindex="-1"></a>dat_clean <span class="ot">&lt;-</span> dat_clean <span class="sc">|&gt;</span></span>
<span id="cb36-2"><a href="#cb36-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">add_column</span>(<span class="at">modified =</span> <span class="dv">0</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>Combine the new meta data with the mean gene expression for the each replicated patient.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb37"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb37-1"><a href="#cb37-1" aria-hidden="true" tabindex="-1"></a>dat_rep_clean <span class="ot">&lt;-</span> <span class="fu">inner_join</span>(<span class="at">x =</span> cols_rep_meta, </span>
<span id="cb37-2"><a href="#cb37-2" aria-hidden="true" tabindex="-1"></a>                          <span class="at">y =</span> dat_rep_mean,</span>
<span id="cb37-3"><a href="#cb37-3" aria-hidden="true" tabindex="-1"></a>                          <span class="at">by =</span> <span class="st">"patient_n"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>Finally, the new data for th replicated patients are combines with the data for the non-replicated patients to form a clean data set without replicates.</p>
<p>As the patient_n variable is no longer informative, this column is removed for all patients.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb38"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb38-1"><a href="#cb38-1" aria-hidden="true" tabindex="-1"></a>dat_clean <span class="ot">&lt;-</span> <span class="fu">bind_rows</span>(dat_clean, dat_rep_clean) <span class="sc">|&gt;</span></span>
<span id="cb38-2"><a href="#cb38-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">relocate</span>(modified, <span class="at">.after =</span> date) <span class="sc">|&gt;</span></span>
<span id="cb38-3"><a href="#cb38-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(<span class="sc">!</span><span class="fu">str_detect</span>(patient_n, <span class="st">"_"</span>)) <span class="sc">|&gt;</span>  </span>
<span id="cb38-4"><a href="#cb38-4" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">select</span>(<span class="sc">-</span>patient_n) <span class="sc">|&gt;</span> </span>
<span id="cb38-5"><a href="#cb38-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">arrange</span>(disease_state) </span>
<span id="cb38-6"><a href="#cb38-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb38-7"><a href="#cb38-7" aria-hidden="true" tabindex="-1"></a>dat_clean <span class="sc">|&gt;</span> </span>
<span id="cb38-8"><a href="#cb38-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">sample_n</span>(<span class="dv">10</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code># A tibble: 10 × 47,237
   geo_accession gender disease_state identifier  date     modified ILMN_1762337
   &lt;chr&gt;         &lt;chr&gt;  &lt;chr&gt;         &lt;chr&gt;       &lt;chr&gt;       &lt;dbl&gt;        &lt;dbl&gt;
 1 GSM1230907    Female HIV           BAA0W69B.10 29-Apr-…        0         158.
 2 GSM1230920    Female HIV           KAA0PBTZ.17 06-Mar-…        0         216.
 3 GSM1230912    Female HIV           DAA0PFCT.11 17-Jan-…        0         287.
 4 GSM1230923    Male   HIV           EAA0VKQY.17 18-Mar-…        0         318.
 5 GSM1230941    Female HIV/TB        JAA1063Y.11 03-Mar-…        0         122.
 6 GSM1230932    Male   HIV/TB        CAA0ZN6L.12 13-Jan-…        0         342.
 7 GSM1230915    Male   HIV           FAA0W22D.13 16-Apr-…        0         154.
 8 GSM1230905    Male   HIV           BAA0V9SR.15 25-Feb-…        0         326.
 9 GSM1230934    Male   HIV/TB        FAA0Z9CZ.10 03-Dec-…        0         300.
10 GSM1230927    Male   HIV/TB        AAA0Z4D1.15 25-Nov-…        0         318.
# ℹ 47,230 more variables: ILMN_2055271 &lt;dbl&gt;, ILMN_1736007 &lt;dbl&gt;,
#   ILMN_2383229 &lt;dbl&gt;, ILMN_1806310 &lt;dbl&gt;, ILMN_1779670 &lt;dbl&gt;,
#   ILMN_1653355 &lt;dbl&gt;, ILMN_1717783 &lt;dbl&gt;, ILMN_1705025 &lt;dbl&gt;,
#   ILMN_1814316 &lt;dbl&gt;, ILMN_2359168 &lt;dbl&gt;, ILMN_1731507 &lt;dbl&gt;,
#   ILMN_1787689 &lt;dbl&gt;, ILMN_3241953 &lt;dbl&gt;, ILMN_1745607 &lt;dbl&gt;,
#   ILMN_2136495 &lt;dbl&gt;, ILMN_1668111 &lt;dbl&gt;, ILMN_2295559 &lt;dbl&gt;,
#   ILMN_1735045 &lt;dbl&gt;, ILMN_1680754 &lt;dbl&gt;, ILMN_2375184 &lt;dbl&gt;, …</code></pre>
</div>
</div>
</section>
</section>
<section id="write-data-to-file" class="level2">
<h2 class="anchored" data-anchor-id="write-data-to-file">Write Data to File</h2>
<p>The cleaned data is written as a tsv file.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb40"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb40-1"><a href="#cb40-1" aria-hidden="true" tabindex="-1"></a>dat_clean <span class="sc">|&gt;</span> </span>
<span id="cb40-2"><a href="#cb40-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">write_tsv</span>(<span class="at">file =</span> <span class="st">"../data/02_dat_clean.tsv"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>

</main>
<!-- /main column -->
<script id="quarto-html-after-body" type="application/javascript">
window.document.addEventListener("DOMContentLoaded", function (event) {
  const toggleBodyColorMode = (bsSheetEl) => {
    const mode = bsSheetEl.getAttribute("data-mode");
    const bodyEl = window.document.querySelector("body");
    if (mode === "dark") {
      bodyEl.classList.add("quarto-dark");
      bodyEl.classList.remove("quarto-light");
    } else {
      bodyEl.classList.add("quarto-light");
      bodyEl.classList.remove("quarto-dark");
    }
  }
  const toggleBodyColorPrimary = () => {
    const bsSheetEl = window.document.querySelector("link#quarto-bootstrap");
    if (bsSheetEl) {
      toggleBodyColorMode(bsSheetEl);
    }
  }
  toggleBodyColorPrimary();  
  const icon = "";
  const anchorJS = new window.AnchorJS();
  anchorJS.options = {
    placement: 'right',
    icon: icon
  };
  anchorJS.add('.anchored');
  const isCodeAnnotation = (el) => {
    for (const clz of el.classList) {
      if (clz.startsWith('code-annotation-')) {                     
        return true;
      }
    }
    return false;
  }
  const clipboard = new window.ClipboardJS('.code-copy-button', {
    text: function(trigger) {
      const codeEl = trigger.previousElementSibling.cloneNode(true);
      for (const childEl of codeEl.children) {
        if (isCodeAnnotation(childEl)) {
          childEl.remove();
        }
      }
      return codeEl.innerText;
    }
  });
  clipboard.on('success', function(e) {
    // button target
    const button = e.trigger;
    // don't keep focus
    button.blur();
    // flash "checked"
    button.classList.add('code-copy-button-checked');
    var currentTitle = button.getAttribute("title");
    button.setAttribute("title", "Copied!");
    let tooltip;
    if (window.bootstrap) {
      button.setAttribute("data-bs-toggle", "tooltip");
      button.setAttribute("data-bs-placement", "left");
      button.setAttribute("data-bs-title", "Copied!");
      tooltip = new bootstrap.Tooltip(button, 
        { trigger: "manual", 
          customClass: "code-copy-button-tooltip",
          offset: [0, -8]});
      tooltip.show();    
    }
    setTimeout(function() {
      if (tooltip) {
        tooltip.hide();
        button.removeAttribute("data-bs-title");
        button.removeAttribute("data-bs-toggle");
        button.removeAttribute("data-bs-placement");
      }
      button.setAttribute("title", currentTitle);
      button.classList.remove('code-copy-button-checked');
    }, 1000);
    // clear code selection
    e.clearSelection();
  });
  function tippyHover(el, contentFn) {
    const config = {
      allowHTML: true,
      content: contentFn,
      maxWidth: 500,
      delay: 100,
      arrow: false,
      appendTo: function(el) {
          return el.parentElement;
      },
      interactive: true,
      interactiveBorder: 10,
      theme: 'quarto',
      placement: 'bottom-start'
    };
    window.tippy(el, config); 
  }
  const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
  for (var i=0; i<noterefs.length; i++) {
    const ref = noterefs[i];
    tippyHover(ref, function() {
      // use id or data attribute instead here
      let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
      try { href = new URL(href).hash; } catch {}
      const id = href.replace(/^#\/?/, "");
      const note = window.document.getElementById(id);
      return note.innerHTML;
    });
  }
      let selectedAnnoteEl;
      const selectorForAnnotation = ( cell, annotation) => {
        let cellAttr = 'data-code-cell="' + cell + '"';
        let lineAttr = 'data-code-annotation="' +  annotation + '"';
        const selector = 'span[' + cellAttr + '][' + lineAttr + ']';
        return selector;
      }
      const selectCodeLines = (annoteEl) => {
        const doc = window.document;
        const targetCell = annoteEl.getAttribute("data-target-cell");
        const targetAnnotation = annoteEl.getAttribute("data-target-annotation");
        const annoteSpan = window.document.querySelector(selectorForAnnotation(targetCell, targetAnnotation));
        const lines = annoteSpan.getAttribute("data-code-lines").split(",");
        const lineIds = lines.map((line) => {
          return targetCell + "-" + line;
        })
        let top = null;
        let height = null;
        let parent = null;
        if (lineIds.length > 0) {
            //compute the position of the single el (top and bottom and make a div)
            const el = window.document.getElementById(lineIds[0]);
            top = el.offsetTop;
            height = el.offsetHeight;
            parent = el.parentElement.parentElement;
          if (lineIds.length > 1) {
            const lastEl = window.document.getElementById(lineIds[lineIds.length - 1]);
            const bottom = lastEl.offsetTop + lastEl.offsetHeight;
            height = bottom - top;
          }
          if (top !== null && height !== null && parent !== null) {
            // cook up a div (if necessary) and position it 
            let div = window.document.getElementById("code-annotation-line-highlight");
            if (div === null) {
              div = window.document.createElement("div");
              div.setAttribute("id", "code-annotation-line-highlight");
              div.style.position = 'absolute';
              parent.appendChild(div);
            }
            div.style.top = top - 2 + "px";
            div.style.height = height + 4 + "px";
            let gutterDiv = window.document.getElementById("code-annotation-line-highlight-gutter");
            if (gutterDiv === null) {
              gutterDiv = window.document.createElement("div");
              gutterDiv.setAttribute("id", "code-annotation-line-highlight-gutter");
              gutterDiv.style.position = 'absolute';
              const codeCell = window.document.getElementById(targetCell);
              const gutter = codeCell.querySelector('.code-annotation-gutter');
              gutter.appendChild(gutterDiv);
            }
            gutterDiv.style.top = top - 2 + "px";
            gutterDiv.style.height = height + 4 + "px";
          }
          selectedAnnoteEl = annoteEl;
        }
      };
      const unselectCodeLines = () => {
        const elementsIds = ["code-annotation-line-highlight", "code-annotation-line-highlight-gutter"];
        elementsIds.forEach((elId) => {
          const div = window.document.getElementById(elId);
          if (div) {
            div.remove();
          }
        });
        selectedAnnoteEl = undefined;
      };
      // Attach click handler to the DT
      const annoteDls = window.document.querySelectorAll('dt[data-target-cell]');
      for (const annoteDlNode of annoteDls) {
        annoteDlNode.addEventListener('click', (event) => {
          const clickedEl = event.target;
          if (clickedEl !== selectedAnnoteEl) {
            unselectCodeLines();
            const activeEl = window.document.querySelector('dt[data-target-cell].code-annotation-active');
            if (activeEl) {
              activeEl.classList.remove('code-annotation-active');
            }
            selectCodeLines(clickedEl);
            clickedEl.classList.add('code-annotation-active');
          } else {
            // Unselect the line
            unselectCodeLines();
            clickedEl.classList.remove('code-annotation-active');
          }
        });
      }
  const findCites = (el) => {
    const parentEl = el.parentElement;
    if (parentEl) {
      const cites = parentEl.dataset.cites;
      if (cites) {
        return {
          el,
          cites: cites.split(' ')
        };
      } else {
        return findCites(el.parentElement)
      }
    } else {
      return undefined;
    }
  };
  var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
  for (var i=0; i<bibliorefs.length; i++) {
    const ref = bibliorefs[i];
    const citeInfo = findCites(ref);
    if (citeInfo) {
      tippyHover(citeInfo.el, function() {
        var popup = window.document.createElement('div');
        citeInfo.cites.forEach(function(cite) {
          var citeDiv = window.document.createElement('div');
          citeDiv.classList.add('hanging-indent');
          citeDiv.classList.add('csl-entry');
          var biblioDiv = window.document.getElementById('ref-' + cite);
          if (biblioDiv) {
            citeDiv.innerHTML = biblioDiv.innerHTML;
          }
          popup.appendChild(citeDiv);
        });
        return popup.innerHTML;
      });
    }
  }
});
</script>
</div> <!-- /content -->



</body></html>